# I Loading

## Packages

```{r Packages ,echo=FALSE, message=FALSE, warning = F}
# BiocManager::install("msa") 
easypackages::libraries(c("dplyr","readr","FactoMineR","RColorBrewer","magrittr","ggplot2",           "forcats","TITAN2","phyloseq","phylosmith","DESeq2","microeco","microViz","speedyseq",           "stringr","factoextra","ggpubr","picante","msa","phangorn","reshape2","data.table",           "patchwork","scales","tidyr","ggConvexHull","ggvegan","mixOmics","purrr","hillR","ggcorrplot",
                          "flextable","igraph","tibble","ggnetwork"))
```

```{r Packages ,echo=FALSE, message=FALSE}
# if (!requireNamespace("BiocManager", quietly = TRUE)) install.packages("BiocManager") 
# BiocManager::install(c("microbiome", "ComplexHeatmap","mixOmics"), update = FALSE) 
# devtools::install_github("david-barnett/microViz") 
# devtools::install_github("mikemc/speedyseq") 
# devtools::install_github("gavinsimpson/ggvegan")
# devtools::install_github("cmartin/ggConvexHull")
# devtools::install_github("davidsjoberg/ggsankey")
# remotes::install_github("metabaRfactory/metabaR")
library(speedyseq);library(microbiome); library(microViz); library(metabaR); library(ggsankey); library(ggvegan); library(ggConvexHull); library(mixOmics)
```

## functions

```{r}
source("./articlez/gen1/script/pairwise_letterZ.R")
```


```{r}
# A function to aggregate counts at a given taxonomic level
agg.table.taxo <- function(tab, tax.lvl="genus", tax.table) {
  tax.table <- tax.table[match(rownames(tab),
                               rownames(tax.table)),]
  message(paste('Table aggregation to the', tax.lvl, "level."))
  #message('Please be sure that the ASV/OTU table and the taxonomy table are ordered the same way')
  if(nrow(tab) != nrow(tax.table)) stop("The ASV/OTU table and the taxonomy table do not have the same number of rows")
  tax <- tax.table[,grep(tax.lvl, colnames(tax.table), ignore.case = T)]
  tax[is.na(tax)] <- "Unknown"
  tab <- aggregate(tab, by=list("taxo"=tax), FUN=sum)
  rownames(tab) <- tab[,1]  
  tab <- tab[,-1]
  return(tab)
}

# A function to fit arrows from ordination to ggplot graphs
arrowMul <- function(arrows, data, at = c(0, 0), fill = 0.75) {
  u <- c(range(data[,1], range(data[,2])))
  u <- u - rep(at, each = 2)
  r <- c(range(arrows[, 1], na.rm = TRUE), range(arrows[, 2], na.rm = TRUE))
  rev <- sign(diff(u))[-2]
  if (rev[1] < 0)
    u[1:2] <- u[2:1]
  if (rev[2] < 0)
    u[3:4] <- u[4:3]
  u <- u/r
  u <- u[is.finite(u) & u > 0]
  fill * min(u)
}

# A function to ad compact letter significance display on a ggplot boxplot graph
cld_bxplot_wilcox <-
  function(data, x, y, significance = 0.05, plot, nudge = 0.05){
    if (kruskal.test(y,x)$p.value<0.05){
      box_rslt <- with(data,graphics::boxplot(y~x ,plot = F))
      
      loc_vec <- box_rslt$stats[nrow(box_rslt$stats),]
      
      wilcx_rslt <- with(data,pairwise.wilcox.test(x = y, g = x,p.adjust.method = 'holm'))
      temp <- wilcx_rslt$p.value
      
      # Make a square matrix to populate with the alpha values.
      n <- nrow(temp)
      mat.names <- c(colnames(temp), rownames(temp)[n])
      my.mat <- matrix(data = NA, nrow = n+1, ncol = n+1)
      colnames(my.mat) <- mat.names
      rownames(my.mat) <- mat.names
      
      # Add diagonal.
      for (i in 1:nrow(my.mat)) {
        my.mat[ i, i] <- 0
      }
      
      # Get vector of p.values
      stat <- na.exclude(as.vector(wilcx_rslt$p.value))
      
      # Add other cells to square matrix.
      k=1
      for (j in 1:(nrow(my.mat)-1)) {
        for (i in ((j+1):nrow(my.mat))) {
          my.mat[i,j] <-  my.mat[j,i] <- stat[k]
          k=k+1
        }
      }
      cld_vec <- multcompView::multcompLetters(my.mat, threshold=significance)
      
      #df with the letters
      x_df <- c(1:length(cld_vec$Letters))
      y_df <- box_rslt$stats[nrow(box_rslt$stats),]
      cbd <- cld_vec$Letters
      ltr_df <- data.frame(x_df, y_df, cbd)
      
      #get coord of the graph
      plt1 <- ggplot2::ggplot_build(plot)
      xmin = plt1 $layout$panel_params[[1]]$x.range[1]
      xmax = plt1 $layout$panel_params[[1]]$x.range[2]
      ymin = plt1 $layout$panel_params[[1]]$y.range[1]
      ymax = plt1 $layout$panel_params[[1]]$y.range[2]
      lmts <- list(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax)
      y.range <- lmts$ymax - lmts$ymin
      y.nudge <- nudge * y.range
      
      cld <- list(ltr_df = ltr_df, y.nudge = y.nudge)
      
      return(plot +
               annotate(geom="text", x=cld$ltr_df$x_df, y=cld$ltr_df$y_df+cld$y.nudge, label=cld$ltr_df$cbd))
    }
    else {return(plot)}
  }
```


## data

### Soil Chi

```{r}
soil_chi <- read.csv2("./data/chemistry/Chimie_sols.csv") #load
colnames(soil_chi) <- c("ID","Mass","N_pct","C_pct","N_mg","C_mg","ConN","P_mgkg","pH_h2o") #rename
soil_chi$trmt <- c(rep("P",3),rep("I",3),rep("R",3)) #set treatments label
soil_chi%<>% mutate(across(2:9, as.numeric)) # transform to numeric
soil_chi %<>% mutate(trmt = fct_relevel(trmt,"P","I","R")) # set factor levels order (poor to rich)
```

### Microbiota

Read clean microbiota data.

```{r}
load("./articlez/gen1/data/cleaned_microtable_ff.RData") # load final
```

```{r}
# rename all datasets and keep those of interest
names(to_save) <- c("meco_mo16s_samsg",
                    "meco_mo16s_seqsg",
                    "meco_moits_samsg",
                    "meco_moits_seqsg",
                    "meco_mo16s_samsg_wrmed",
                    "meco_mo16s_seqsg_wrmed",
                    "meco_moits_samsg_wrmed",
                    "meco_moits_seqsg_wrmed",
                    "meco_mo16s_samsg_raref",
                    "meco_mo16s_seqsg_raref",
                    "meco_moits_samsg_raref",
                    "meco_moits_seqsg_raref",
                    "meco_mo16s_samsg_wrmed_raref",
                    "meco_mo16s_seqsg_wrmed_raref",
                    "meco_moits_samsg_wrmed_raref",
                    "meco_moits_seqsg_wrmed_raref")

for(i in c(10,12)){
  assign(names(to_save)[[i]],to_save[[i]])
}
rm(to_save)
gc()

# same for the "datasets" variable that will be use to loop on
datasets <- c("meco_mo16s_samsg",
              "meco_mo16s_seqsg",
              "meco_moits_samsg",
              "meco_moits_seqsg",
              "meco_mo16s_samsg_wrmed",
              "meco_mo16s_seqsg_wrmed",
              "meco_moits_samsg_wrmed",
              "meco_moits_seqsg_wrmed",
              "meco_mo16s_samsg_raref",
              "meco_mo16s_seqsg_raref",
              "meco_moits_samsg_raref",
              "meco_moits_seqsg_raref",
              "meco_mo16s_samsg_wrmed_raref",
              "meco_mo16s_seqsg_wrmed_raref",
              "meco_moits_samsg_wrmed_raref",
              "meco_moits_seqsg_wrmed_raref")
datasets <- datasets[c(10,12)]
```

#### addspecies

```{r}
# fasta <- paste0(paste0(">",rownames(meco_mo16s_seqsg_raref$tax_table),"\n",gsub("s__","",meco_mo16s_seqsg_raref$ori_tax[,"sequence"])))
# write(fasta,paste0("./articlez/gen1/resultats/final/fasta4dada/fasta_16s.fasta"))
```


### Leaf traits

```{r import_data, echo=FALSE,warning=FALSE}
data_leaf <- read.csv("./articlez/gen1/data/data_art1.csv") #load
data_leaf$Trtmt <- as.factor(data_leaf$Trtmt) # treatment as factor
levels(data_leaf$Trtmt) <- c("R-D","I-D","P-D","R-W","I-W","P-W") # change treatment code
data_leaf %<>% mutate(Trtmt = fct_relevel(Trtmt,"P-D","I-D","R-D","P-W","I-W","R-W")) # change level order to go from harsh to lenient env: DRY > poor/int/rich -> Wet > poor/int/rich

mo_col <- c("P-D"="#FFCC99","I-D"="#993300","R-D"="#330000","P-W"="#5697f5","I-W"="#1d61c2","R-W"="#052d66") # set treatments colors
var_cat_col <- c("biomass"="burlywood3","nutrient"="darkcyan","photosynthesis"="darkolivegreen4","structural"="chocolate3","water"="darkslateblue","reproduction"="yellow3") # set variable categories colors
```

Variables:
-   *FM*, *DM* and *TM* : *Fresh*, *Dry* and *Turgid* mass respectively (g)
-   *mu_thick* : leaf thickness (cm)
-   *LDMC* : Leaf dry mass content (g.g\^-1)
-   *RWC* : Relative Water Content (%)
-   *WHC* : Water Holding Capacity (gH20.m\^-2)
-   *Nb_leaves* : Number of Leaves
-   *Pl_H* : Plant height (cm)
-   *LL_H* : Youngest mature leaf length (cm)
-   *LL_W* : Youngest mature leaf width (cm)
-   *chloro* : Leaf chlorophyll content (µg.cm\^-2)
-   *FvFm* : PSII yield
-   *ETRmax* : maximal electron transfert rate (µmol photon.m^-2.s^-1)
-   *Water_pot* : water potential (MPa?)
-   *C.N_leaf* : Leaf C/N ratio
-   *SS_leaf* : Leaf soluble sugars (µg.mg\^-1)
-   *Starch_leaf* : Leaf starch (µg.mg\^-1)
-   *mean_trich_diam_inf* : abaxial face trichome diameter (mm)
-   *mean_trich_diam_sup* : adaxial face trichome diameter (mm)
-   *inf_trich_dens* : abaxial face trichome density (nb.mm\^-2)
-   *sup_trich_dens* : adaxial face trichome density (nb.mm\^-2)
-   *sto_dens* : stomatal density (nb.mm\^-2), only abaxial face, hypostomatous species
-   *LMA* : Leaf Mass Area (g.m\^-2)
-   *C-*, *N-* and *P-mg* : Carbon, Nitrogen and Phosphorous content (mg.mg\^-1)

```{r}
trait_list_glob <- c("Nb_leaves", # list of used traits
                     "Pl_H",
                     "LL_H",
                     "LL_W",
                     "LMA",
                     "LDMC",
                     "mu_thick",
                     "sto_dens",
                     "chloro",
                     "FvFm",
                     "ETRmax",
                     "Water_pot",
                     "RWC",
                     "C.N_leaf",
                     "SS_leaf",
                     # "Starch_leaf",  #Remove starch 'cause detection limit
                     "Cmg",
                     "Nmg",
                     "Pmg",
                     "Nb_Fr")
Nbcol <- match(trait_list_glob,colnames(data_leaf)) # get corresponding column indices

trait_df_name <- data.frame(rawnames=trait_list_glob,cleannames=c('Number of leaves','Plant height','Leaf length','Leaf width', 'LMA', 'LDMC', 'Thickness','Stomata','Chlorophyll','Fv/Fm','ETRmax','Water potential','RWC','C/N','Soluble sugar','Carbon','Nitrogen','Phosphorous',"Number of Flowers")) # create clean names used for plots

trait_df_name$unit <- c(bquote(""), # write units used for plots
                        bquote("(cm)"),
                        bquote("(cm)"),
                        bquote("(cm)"),
                        bquote("(g.m"^-2~")"),
                        bquote("(g.g"^1~")"),
                        bquote("(mm)"),
                        bquote("(mm"^2~")"),
                        bquote("(mg.g"^-1~")"),
                        bquote(""),
                        bquote("(µmol.m"^-2~".s"^-1~")"),
                        bquote("(MPa)"),
                        bquote("(%)"),
                        bquote(""),
                        bquote("(µg.mg"^-1~")"),
                        bquote("(mg.g"^-1~")"),
                        bquote("(mg.g"^-1~")"),
                        bquote("(mg.g"^-1~")"),
                        bquote(""))

var_cat <- c("biomass", #set variable categories
             "biomass",
             "biomass",
             "biomass",
             "structural",
             "structural",
             "structural",
             "structural",
             "photosynthesis",
             "photosynthesis",
             "photosynthesis",
             "water",
             "water",
             "nutrient",
             "nutrient",
             "nutrient",
             "nutrient",
             'nutrient',
             'reproduction')

trait_df_name %<>% mutate(var_cat=as.factor(var_cat)) # as factor

df_col_var <- data.frame(var_cat=c("biomass", # also store colors
                                   "nutrient",
                                   "photosynthesis",
                                   "structural",
                                   "water",
                                   "reproduction",
                                   "otus"),
                         colorz=c("burlywood3",
                                  "darkcyan",
                                  "darkolivegreen4",
                                  "chocolate3",
                                  "darkslateblue",
                                  "yellow3",
                                  "darkred"))
trait_df_name%<>%left_join(df_col_var)
```

# II Make Graphs

## Soil chemistry

```{r}
# Soil Nitrogen
N <- ggplot(soil_chi,aes(x=trmt,y=N_mg/Mass, color=trmt))+ 
  geom_point(size=3)+
  scale_color_manual(values=c("#ffe1b0","#6b5e49","#0f0d0a"))+
  theme_classic2()+
  labs(color="Soil treatments (n=3)",
       y=bquote("Nitrogen (mg.mg "[DM]~")"),
       x="")

# Soil Carbon
C <- ggplot(soil_chi,aes(x=trmt,y=C_mg/Mass, color=trmt))+ 
  geom_point(size=3)+
  scale_color_manual(values=c("#ffe1b0","#6b5e49","#0f0d0a"))+
  theme_classic2()+
  labs(color="Soil treatments",
       y=bquote("Carbon (mg.mg "[DM]~")"),
       x="")+
  theme(legend.position="none")

# Soil Phosphorous
P <- ggplot(soil_chi,aes(x=trmt,y=P_mgkg/1000, color=trmt))+
  geom_point(size=3)+
  scale_color_manual(values=c("#ffe1b0","#6b5e49","#0f0d0a"))+
  theme_classic2()+
  labs(color="Soil treatments",
       y=bquote("Phosphorous (mg.mg "[DM]~")"),
       x="")+
  theme(legend.position="none")

leg <- EcophyCofog::xtract_legend(N) #Extract legend

soil_chi_plot <- (C|N+theme(legend.position="none"))/(P|leg)
ggsave("./articlez/gen1/resultats/final/soil/soil_chi.png",soil_chi_plot, device="png",height = 7,width=5)
```

## POV traits

```{r}
# Traits partition of variance
pov_list <- NULL #to store
for(traits in trait_df_name$rawnames){ #for all traits
  col <- data_leaf[,match(traits,colnames(data_leaf))] # get the trait values
  data_temp <- data.frame(trait=col, drought=data_leaf$drought,substrate=data_leaf$substrate) # add treatment info
  pov_list[[traits]] <- POV::POV(trait~drought*substrate,data_temp) # variance partitioning
}
pov_list %<>% # transform to ggplot compatible dataframe
  purrr::map(~mutate(.,var_source=rownames(.)))%>%
  purrr::map(~reshape2::melt(.))

pov_merged <- bind_rows(pov_list, .id = "trait") # merge into one dataframe

new_pov <- pov_merged%>% # rename variable and order factors
  filter(variable=='% of total')%>%
  arrange(value)%>%
  mutate(var_source=as.factor(var_source),
         rawnames=as.factor(trait))%>%
  mutate(var_source=forcats::fct_relevel(var_source,
                                         "Common",
                                         "Between drought",
                                         "Between substrate",
                                         "Within drought",
                                         "Within substrate",
                                         "Between drought:substrate",
                                         "Within drought:substrate"))

var_order_common <- filter(new_pov,var_source=="Common")%>%
  arrange(desc(value))%>%
  mutate(order_com=1:nrow(.))%>%
  dplyr::select(trait,order_com)
var_order_drought <- filter(new_pov,var_source=="Between drought")%>%
  arrange(desc(value))%>%
  mutate(order_drought=1:nrow(.))%>%
  dplyr::select(trait,order_drought)
var_order_substrate <- filter(new_pov,var_source=="Between substrate")%>%
  arrange(desc(value))%>%
  mutate(orde_sbustrater=1:nrow(.))%>%
  dplyr::select(trait,orde_sbustrater)

new_pov <- plyr::join_all(list(new_pov,var_order_common,var_order_drought,var_order_substrate),by="trait")
new_pov <- left_join(new_pov,trait_df_name)


bar_traits_var <- new_pov%>% #build the graph
  mutate(cleannames = fct_reorder(cleannames,var_cat))%>% # reorder levels
  mutate(var_source = fct_relevel(var_source,"Common", after = Inf ))%>% # reorder levels
  ggplot( aes(x=cleannames,y=value , fill= var_source)) + # plot
  geom_bar(position =position_stack(reverse=T),stat="identity")+
  scale_fill_brewer(palette='Set2', name="")+
  theme_classic2()+
  xlab("")+
  ylab("% of variance")+ # theme
  theme(axis.text.x = element_text(margin=margin(t=5),hjust=1))+
  scale_y_continuous(expand=c(0,0))+
  theme(legend.position = "bottom",
        legend.justification = c("left"),
        legend.box.just = "left",
        legend.margin=margin(c(1,1,1,1)),
        legend.box.margin=margin(1,1,1,1),
        legend.box.spacing = unit(0, "pt"),
        legend.key.size = unit(0.4, 'cm'),
        legend.text = element_text(size=8),
        legend.spacing.x = unit(1, "mm"),
        legend.spacing.y = unit(1, "mm"))+
  coord_flip()

bar_trait <- (bar_traits_var+theme(axis.text = element_text(face="bold"), # transform to bold text
                                   axis.title.x = element_text(face="bold"),
                                   legend.text = element_text(face='bold')) | 
                trait_df_name%>% # add variable categories color
                mutate(cleannames = fct_reorder(cleannames,var_cat))%>%
                mutate(var_cat=fct_relevel(var_cat,c("biomass",'nutrient','photosynthesis','structural','water','reproduction')))%>%
                ggplot(aes(x=1,y=cleannames,fill=var_cat))+
                geom_tile()+
                theme_void()+
                scale_fill_manual(values = var_cat_col, 
                                  name="Variable categories")+
                theme(legend.text = element_text(face="bold"),
                      legend.title = element_text(face='bold')))+
  plot_layout(widths = c(15,1))
```

```{r}
ggsave("./articlez/gen1/resultats/final/traits/pov.png",bar_trait,width = 10,height = 5)
```

## Leaf trait range

```{r}
pdens <- NULL
for(i in Nbcol){ # for all variable in Nbcol  
  
  unitofvar <- trait_df_name$unit[match(names(data_leaf)[i],trait_df_name$rawnames)][[1]] # get units
  var <-trait_df_name$cleannames[match(names(data_leaf)[i],trait_df_name$rawnames)] # get the variable
  
  plot <- local({
    i<-i
      ggplot(data_leaf,aes(x=data_leaf[,i],color=Trtmt),alpha=.7)+ # plot the data
      geom_density(key_glyph = "timeseries")+ # draw density
      scale_color_manual(values=mo_col,name="Treatments")+
      theme_classic2()+
      ylab("")+
      xlab(bquote(~.(var)~.(unitofvar)))+
      theme(axis.text.x = element_text(angle = 90),
            axis.title.x = element_text(face="bold",color=arrange(trait_df_name,var_cat)[which(arrange(trait_df_name,var_cat)$rawnames==names(data_leaf)[i]),"colorz"]), # color X title according to variable category
            axis.text.y = element_blank(),
            axis.ticks.y = element_blank())})
  ggsave(paste0("./articlez/gen1/resultats/final/range_leaf_traits/density_",names(data_leaf)[i],".png"),plot,device="png",width=6,height = 2)
  pdens[[names(data_leaf)[i]]] <- plot
}
```

```{r}
# Build huge plot
ptrait_range <-pdens$Nb_leaves+pdens$Pl_H+pdens$LL_H+pdens$LL_W+pdens$LMA+pdens$LDMC+pdens$mu_thick+pdens$mu_thick+pdens$sto_dens+pdens$chloro+pdens$FvFm+pdens$ETRmax+pdens$ETRmax+pdens$Water_pot+pdens$RWC+pdens$C.N_leaf+pdens$SS_leaf+pdens$Cmg+pdens$Nmg+pdens$Pmg+pdens$Nb_Fr+plot_layout(guides="collect",widths = c(1,1,1))&theme(axis.title = element_text(face="bold"),axis.text = element_text(face="bold"),legend.text = element_text(face = "bold"))
ggsave("./articlez/gen1/resultats/final/range_leaf_traits/combined.png",ptrait_range,height = 13,width = 9)
```

## PCA


```{r}
# Run PCA on leaf traits but on C/N ratio as its redundant with C and N
PCA_plant_trait <- PCA(data_leaf[,Nbcol[-c(14,33)]], scale.unit = T,  graph  = F ,ncp=length(Nbcol)-1)
```


```{r}
#PCA biplot
plot_pca<-fviz_pca_biplot(PCA_plant_trait,
                          geom="point",
                          habillage = data_leaf$Trtmt,## color by groups
                          addEllipses = TRUE, # add ellipses
                          legend.title = "Treatment",
                          ellipse.type="convex",
                          pointsize=2,
                          palette = mo_col,
                          col.var="black"
)+
  scale_shape_manual(values = c(17,17,17,19,19,19))+
  theme_classic2()+
  theme(panel.background = element_rect(fill = "transparent"),
        plot.background = element_rect(fill = "transparent", color = NA), 
        legend.background = element_rect(fill = "transparent"),
        legend.key.size = unit(.7, 'in'), 
        legend.key.height = unit(.7, 'in'), 
        legend.key.width = unit(.7, 'in'), 
        legend.title = element_text(size=22),
        legend.text = element_text(size=20), 
        axis.title = element_text(size = 16),
        plot.margin = margin(2, 2, 2, 2),
        title = element_blank())

ggsave(plot=plot_pca,
       filename = "./articlez/gen1/resultats/final/pcaleaf/pca_plant.svg", # save to svg for aesthetic edition with Inkscape: https://inkscape.org/
       device = "svg",
       width = 9, 
       height = 7,
       units = "in")
```

```{r}
# Plot variable contribution to 2 first components and correlogram with farther components
rownames(PCA_plant_trait$var$coord) <- trait_df_name$cleannames[match(rownames(PCA_plant_trait$var$coord),trait_df_name$rawnames)] # set rownames as clean variable names

# on dim1
contrib_dim1 <- fviz_contrib(PCA_plant_trait,'var',fill='lightgrey',color='grey28')+
  theme_classic2()+
  scale_y_continuous(expand=c(0,0))+
  labs(x="")+
  theme(axis.text.x = element_text(angle=45,margin=margin(t=3),hjust=1,vjust=1))+
  ggtitle("First dimension")
# on dim2
contrib_dim2 <- fviz_contrib(PCA_plant_trait,'var',fill='lightgrey',color='grey28',axes=2)+
  theme_classic2()+
  scale_y_continuous(expand=c(0,0))+
  labs(x="")+
  theme(axis.text.x = element_text(angle=45,margin=margin(t=3),hjust=1,vjust=1))+
  ggtitle("Second dimension")
# combine
contrib_var <- contrib_dim1/contrib_dim2

# correlogram
rownames(PCA_plant_trait$var$cos2) <- trait_df_name$cleannames[match(rownames(PCA_plant_trait$var$cos2),
                                                                     trait_df_name$rawnames)]
corrplot_pca_traits <- ggcorrplot(PCA_plant_trait$var$cos2, method = "square", tl.col = "black") 

ggsave("./articlez/gen1/resultats/final/pcaleaf/corrplot_pca_traits.png",corrplot_pca_traits,device="png",height = 7,width=8)
ggsave("./articlez/gen1/resultats/final/pcaleaf/contrib_var.png",contrib_var,device="png", height=7,width=8 )
```

## Alphadiv

### Composition plot

```{r}
# create trans_abund object
comp_barplots <- NULL
for(i in datasets){ # for both datasets (its and 16s)
  tmp <- get(i) # get the data
  trans_tmp <- trans_abund$new(dataset = tmp, taxrank = "Class_conf", ntaxa = 12) # abundance @class level. The top 12 most abundant are kept, others are grouped
  p_class <- trans_tmp$plot_bar(others_color = "grey70",facet = "Trtmt", xtext_keep = FALSE, legend_text_italic = FALSE) # plot it
  
  trans_tmp <- trans_abund$new(dataset = tmp, taxrank = "Order_conf", ntaxa = 12) # @order level
  p_order <- trans_tmp$plot_bar(others_color = "grey70",facet = "Trtmt", xtext_keep = FALSE, legend_text_italic = FALSE)
  
  trans_tmp <- trans_abund$new(dataset = tmp, taxrank = "Family_conf", ntaxa = 12) # @family
  p_family <- trans_tmp$plot_bar(others_color = "grey70",facet = "Trtmt", xtext_keep = FALSE, legend_text_italic = FALSE)
  
  trans_tmp <- trans_abund$new(dataset = tmp, taxrank = "Genus_conf", ntaxa = 12) # @genus
  p_genus <- trans_tmp$plot_bar(others_color = "grey70",facet = "Trtmt", xtext_keep = FALSE, legend_text_italic = FALSE)
  
  plot_comb <- (p_class|p_order)/(p_family|p_genus) # merge
  comp_barplots[[i]] <- plot_comb # store
  ggsave(paste0("./articlez/gen1/resultats/final/relab/relab_",i,".png"),plot_comb, device="png", width=15,height = 10) # save
}


# Make specific graph @genus level

# set facet strip colors as treatments colors and bold title
strips <- ggh4x::strip_themed(background_x = ggh4x::elem_list_rect(fill=mo_col),
                              text_x = ggh4x::elem_list_text(color=c("white"),
                                                             face="bold"))

# set labels as [treatment]([sample size])
trtmt_labels <- paste0(levels(meco_mo16s_seqsg_raref$sample_table$Trtmt)," (",c(12,8,6,13,8,10),")")
names(trtmt_labels) <- levels(meco_mo16s_seqsg_raref$sample_table$Trtmt)

# redo plot with these informations
bac_genus_complot <- comp_barplots$meco_mo16s_seqsg_raref[[2]][[2]]$data%>% # get the relative abundance data from previous plot
  ggplot(aes(y=Sample,x=Abundance,fill=Taxonomy))+ # plot
  geom_bar(stat="identity",position = "stack",width = 1)+
  ggh4x::facet_wrap2(~Trtmt,scales = "free_y",nrow = 6,strip = strips,labeller = labeller(Trtmt=trtmt_labels))+ # facet by treatments with custom facet looks
  scale_fill_manual(values=c("grey60",rev(colorRampPalette(brewer.pal(8, "Dark2"))(12))),name="Bacterial genus")+ # expand Dark2 palette to encompass 12 genera
  scale_x_continuous(expand = c(0,0))+
  theme_classic2()+
  theme(axis.text.y = element_blank(),axis.title.y = element_blank(),axis.ticks.y = element_blank())+
  xlab("Relative abundance (%)")+
  theme(legend.position = "bottom",
        legend.direction = "horizontal",
        legend.title = element_text(hjust = .5,face = "bold"),
        legend.title.position = "top")+
  guides(fill = guide_legend(override.aes = list(size = 0.25),ncol = 5))+
  theme(legend.title = element_text(size = 7), 
        legend.text = element_text(size = 7),
        legend.key.size = unit(.5, "cm"),
        legend.key.spacing = unit(0,"cm"))


# same for fungi
trtmt_labels <- paste0(levels(meco_moits_seqsg_raref$sample_table$Trtmt)," (",c(14,15,7,15,13,15),")")
names(trtmt_labels) <- levels(meco_moits_seqsg_raref$sample_table$Trtmt)

fung_genus_complot <- comp_barplots$meco_moits_seqsg_raref[[2]][[2]]$data%>%
  ggplot(aes(y=Sample,x=Abundance,fill=Taxonomy))+
  geom_bar(stat="identity",position = "stack",width = 1)+
  ggh4x::facet_wrap2(~Trtmt,scales = "free_y",nrow = 6,strip = strips,labeller = labeller(Trtmt=trtmt_labels))+
  scale_fill_manual(values=c("grey60",rev(colorRampPalette(paletteer::paletteer_d("ggthemes::Classic_Green_Orange_12") )(12))),name="Fungal genus")+
  scale_x_continuous(expand = c(0,0))+
  theme_classic2()+
  theme(axis.text.y = element_blank(),axis.title.y = element_blank(),axis.ticks.y = element_blank())+
  xlab("Relative abundance (%)")+
  theme(legend.position = "bottom",
        legend.direction = "horizontal",
        legend.title = element_text(hjust = .5,face = "bold"),
        legend.title.position = "top") +
  guides(fill = guide_legend(override.aes = list(size = 0.25),ncol = 5))+
  theme(legend.title = element_text(size = 7), 
        legend.text = element_text(size = 7),
        legend.key.size = unit(.5, "cm"),
        legend.key.spacing = unit(0,"cm"))

# combine and save these plots
comp <- bac_genus_complot+fung_genus_complot 
ggsave("./articlez/gen1/resultats/final/relab/comp_final.png",comp,width = 13,height = 10)

# same plot but tighter and with legend on the sides
comp2 <- bac_genus_complot+theme(legend.position = "left",legend.direction = "vertical",legend.title = element_text(hjust = 1,face = "bold"))+
  guides(fill=guide_legend(ncol=1,label.position="left"))+
  fung_genus_complot+theme(legend.position = "right",legend.direction = "vertical",legend.title = element_text(hjust = 0,face = "bold"))+
  guides(fill=guide_legend(ncol=1))
ggsave("./articlez/gen1/resultats/final/relab/comp_final2.png",comp2,width = 8,height = 6)
```

### Taxa smry

```{r}
otu_info <- NULL
for (i in datasets){
  data_tmp <- get(i)
  data_tmp <- clone(data_tmp)
  d <- data_tmp$tax_table[,2:6]
  tab <- data_tmp$otu_table 
  
  tax.lvl <- "Order_conf"
  order_tot <-agg.table.taxo(tab = tab, # aggregate my data at the targeted tax level
                             tax.lvl = tax.lvl,
                             tax.table = data_tmp$tax_table) %>%
    apply(1,sum) %>% # sum the number of reads
    sort %>% # sort in ascending order
    identity
  tax.lvl <- "Class_conf"
  class_tot <-agg.table.taxo(tab = tab, # aggregate my data at the targeted tax level
                             tax.lvl = tax.lvl,
                             tax.table = data_tmp$tax_table) %>%
    apply(1,sum) %>% # sum the number of reads
    sort %>% # sort in ascending order
    identity
  tax.lvl <- "Phylum_conf"
  phylum_tot <-agg.table.taxo(tab = tab, # aggregate my data at the targeted tax level
                              tax.lvl = tax.lvl,
                              tax.table = data_tmp$tax_table) %>%
    apply(1,sum) %>% # sum the number of reads
    sort %>% # sort in ascending order
    identity
  
  tax.lvl <- "Family_conf"
  family_tot <-agg.table.taxo(tab = tab, # aggregate my data at the targeted tax level
                              tax.lvl = tax.lvl,
                              tax.table = data_tmp$tax_table) %>%
    apply(1,sum) %>% # sum the number of reads
    sort %>% # sort in ascending order
    identity
  
  
  tax.lvl <- "Genus_conf"
  genus_tot <-agg.table.taxo(tab = tab, # aggregate my data at the targeted tax level
                             tax.lvl = tax.lvl,
                             tax.table = data_tmp$tax_table) %>%
    apply(1,sum) %>% # sum the number of reads
    sort %>% # sort in ascending order
    identity
  
  top10_order <- names(order_tot[(length(order_tot)-ifelse(length(order_tot)<10,length(order_tot),10)):length(order_tot)])
  top10_order <- top10_order[which(top10_order!="o__")]
  top10_class <- names(class_tot[(length(class_tot)-ifelse(length(class_tot)<10,length(class_tot),10)):length(class_tot)])
  top10_class <- top10_class[which(top10_class!="c__")]
  top10_phylum <- names(phylum_tot[(length(phylum_tot)-ifelse(length(phylum_tot)<10,length(phylum_tot),10)):length(phylum_tot)])
  top10_phylum <- top10_phylum[which(top10_phylum!="p__")]
  top10_genus <- names(genus_tot[(length(genus_tot)-ifelse(length(genus_tot)<10,length(genus_tot),10)):length(genus_tot)])
  top10_genus <- top10_genus[which(top10_genus!="g__")]
  top10_family <- names(family_tot[(length(family_tot)-ifelse(length(family_tot)<10,length(family_tot),10)):length(family_tot)])
  top10_family <- top10_family[which(top10_family!="f__")]
  
  d <- d %>% mutate(Phylum_conf=ifelse(!Phylum_conf%in%c("p__",top10_phylum),"p__others",Phylum_conf),
                    Class_conf=ifelse(!Class_conf%in%c("c__",top10_class),"c__others",Class_conf),
                    Order_conf=ifelse(!Order_conf%in%c("o__",top10_order),"o__others",Order_conf),
                    Genus_conf=ifelse(!Genus_conf%in%c("g__",top10_genus),"g__others",Genus_conf),
                    Family_conf=ifelse(!Family_conf%in%c("f__",top10_family),"f__others",Family_conf))
  d <- d %>%  mutate(Phylum_conf=ifelse(Phylum_conf=="p__","p__unknown",Phylum_conf),
                     Class_conf=ifelse(Class_conf=="c__","c__unknown",Class_conf),
                     Order_conf=ifelse(Order_conf=="o__","o__unknown",Order_conf),
                     Genus_conf=ifelse(Genus_conf=="g__","g__unknown",Genus_conf),
                     Family_conf=ifelse(Family_conf=="f__","f__unknown",Family_conf))
  
  d$reads <- data_tmp$taxa_sums()
  
  df <- d %>%
    make_long(1:5)
  
  TotalCount = nrow(d)
  # Step 2
  dagg <- df%>%
    dplyr::group_by(node)%>%
    tally()
  
  dagg <- dagg%>%
    dplyr::group_by(node)%>%
    dplyr::mutate(pct = n/TotalCount)%>%
    dplyr::mutate(tax_lvl=ifelse(grepl("c__",node),"class",
                                 ifelse(grepl("o__",node),"order",
                                        ifelse(grepl("p__",node),"phylum",
                                               ifelse(grepl("f__",node),"family","genus")))))
  
  
  # Step 3
  df2_16s <- merge(df, dagg, by.x = 'node', by.y = 'node', all.x = TRUE)
  # df2_16s%<>%filter(x%in%c("Phylum_conf","Order_conf","Class_conf"))
  pl <- ggplot(df2_16s, aes(x = x, 
                            next_x = next_x, 
                            node = node, 
                            next_node = next_node, 
                            fill = factor(node),
                            label = paste0(node," n=", n, ' (',  round(pct* 100,1), '%)' ))
  )
  
  pl <- pl +geom_sankey(flow.alpha = 0.5,  color = "gray40", show.legend = TRUE)
  pl <- pl +geom_sankey_label(size = 3, color = "black", fill= "white", hjust = -0.1)
  
  pl <- pl +  theme_bw()
  pl <- pl + theme(legend.position = "none")
  pl <- pl +  theme(axis.title = element_blank()
                    , axis.text.y = element_blank()
                    , axis.ticks = element_blank()  
                    , panel.grid = element_blank())
  
  pl <- pl + labs(title = ifelse(grepl("16s",i),"Sankey diagram - Bacteria", "Sankey diagram - Fungi"))
  pl <- pl + labs(fill = 'Nodes')
  
  pl <- pl + scale_fill_viridis_d(option = "inferno")
  
  ggsave(paste0("./articlez/gen1/resultats/final/taxa_smry/sankey_",i,".png"),pl, device="png", width=15,height = 10)
  
  # Table recap
  
  phylum <- filter(dagg,tax_lvl=="phylum")
  phylum %<>% mutate(node=gsub("p__","",node),
                     abun = paste0(n," (",round(pct*100, digits=2),"%)"))%>%
    dplyr::select(-c(2:4))
  
  class <- filter(dagg,tax_lvl=="class")
  class %<>% mutate(node=gsub("c__","",node),
                    abun = paste0(n," (",round(pct*100, digits=2),"%)"))%>%
    dplyr::select(-c(2:4))
  
  order <- filter(dagg,tax_lvl=="order")
  order %<>% mutate(node=gsub("o__","",node),
                    abun = paste0(n," (",round(pct*100, digits=2),"%)"))%>%
    dplyr::select(-c(2:4))
  
  genus <- filter(dagg,tax_lvl=="genus")
  genus %<>% mutate(node=gsub("g__","",node),
                    abun = paste0(n," (",round(pct*100, digits=2),"%)"))%>%
    dplyr::select(-c(2:4))
  
  family <- filter(dagg,tax_lvl=="family")
  family %<>% mutate(node=gsub("f__","",node),
                     abun = paste0(n," (",round(pct*100, digits=2),"%)"))%>%
    dplyr::select(-c(2:4))
  
  if(nrow(phylum)<12){
    phylum <- rbind(phylum,data.frame(node=rep(NA,12-nrow(phylum)),abun=rep(NA,12-nrow(phylum))))
  }
  
  smry_tab <- cbind(phylum,class,order,family,genus)
  smry_tab %<>% 
    add_row(.after = 10)%>%
    flextable()%>%
    set_header_labels(node...1="Name",abun...2="Count",node...3="Name",abun...4="Count",node...5="Name",abun...6="Count",node...7="Name",abun...8="Count",node...9="Name",abun...10="Count")%>%
    add_header_row(top = TRUE, values = c("Phylum", "Class","Order","Family","Genus"), colwidths = c(2,2,2,2,2))%>%
    align(align = "center", part = "all") %>%
    autofit()%>%
    bold(part='header')
  
  
  otu_count <- data_tmp$tax_table%>%
    nrow()
  otu_richness_range <- colSums(data_tmp$otu_table>0)%>%range()
  
  otu_info[[i]] <- 
    data.frame(`Marker`= ifelse(grepl("16s",i),"16s - Bacteria","ITS - Fungi"),
               `Total number of OTUs`=otu_count,
               `Sample richness range`=paste0(otu_richness_range,collapse = " - "))
  
  
  sect_properties <- officer::prop_section(
    page_size = officer::page_size(
      width = 42/2.54,
      height = 30/2.54,
      orient = "landscape"
    ),
    type = "continuous",
    page_margins = officer::page_mar()
  )
  
  save_as_docx(smry_tab,path=paste0("./articlez/gen1/resultats/final/taxa_smry/taxa_smry_",i,".docx"),pr_section = sect_properties)
  
  smry_tab_reads <- 
    rowSums(data_tmp$otu_table)%>%
    reshape2::melt()%>%
    mutate(OTU=rownames(.))%>%
    left_join(mutate(data_tmp$tax_table,OTU=rownames(data_tmp$tax_table)))
  
  phylum_reads <- smry_tab_reads%>%
    group_by(Phylum_conf)%>%
    summarise(nreads=sum(value))%>%
    mutate(nreads_prop=(nreads/sum(nreads))*100)%>%
    arrange(desc(nreads_prop))%>%
    dplyr::slice(1:10)
  
  
  class_reads <- smry_tab_reads%>%
    group_by(Class_conf)%>%
    summarise(nreads=sum(value))%>%
    mutate(nreads_prop=(nreads/sum(nreads))*100)%>%
    arrange(desc(nreads_prop))%>%
    dplyr::slice(1:10)
  
  order_reads <- smry_tab_reads%>%
    group_by(Order_conf)%>%
    summarise(nreads=sum(value))%>%
    mutate(nreads_prop=(nreads/sum(nreads))*100)%>%
    arrange(desc(nreads_prop))%>%
    dplyr::slice(1:10)
  
  family_reads <- smry_tab_reads%>%
    group_by(Family_conf)%>%
    summarise(nreads=sum(value))%>%
    mutate(nreads_prop=(nreads/sum(nreads))*100)%>%
    arrange(desc(nreads_prop))%>%
    dplyr::slice(1:10)
  
  genus_reads <- smry_tab_reads%>%
    group_by(Genus_conf)%>%
    summarise(nreads=sum(value))%>%
    mutate(nreads_prop=(nreads/sum(nreads))*100)%>%
    arrange(desc(nreads_prop))%>%
    dplyr::slice(1:10)
  
  if(nrow(phylum_reads)<10){
    phylum_reads <- rbind(phylum_reads,data.frame(Phylum_conf=rep(NA,10-nrow(phylum_reads)),nreads=rep(NA,10-nrow(phylum_reads)),nreads_prop=rep(NA,10-nrow(phylum_reads))))
  }
  
  smry_tab_reads_merged <- cbind(phylum_reads,class_reads,order_reads,family_reads,genus_reads)
  smry_tab_reads_merged<-  set_colnames(smry_tab_reads_merged,paste0("col",1:15))
  smry_tab_reads_merged%>%
    mutate(across(c(1,4,7,10,13),.fns=~gsub("[a-z]__","",.x)))%>%
    mutate(across(c(1,4,7,10,13),.fns=~ifelse(.x=="","Unkown",.x)))%>%
    mutate(across(c(1,4,7,10,13),.fns=~paste0(.x,": ")))%>%
    mutate(across(c(3,6,9,12,15),.fns=~round(.x, digits = 3)))%>%
    mutate(across(c(3,6,9,12,15),.fns=~paste0("(",.x,"%)")))%>%
    unite("Phylum",1:3,sep = " ")%>%
    unite("Class",2:4,sep = " ")%>%
    unite("Order",3:5,sep = " ")%>%
    unite("Family",4:6,sep = " ")%>%
    unite("Genus",5:7,sep = " ")%>%
    mutate(across(1:5,~ifelse(grepl("NA",.x),"",.x)))%>%
    flextable()%>%
    align(align = "center", part = "all") %>%
    autofit()%>%
    bold(part='header')%>%
    save_as_docx(path=paste0("./articlez/gen1/resultats/final/taxa_smry/taxa_smry_reads_",i,".docx"),pr_section = sect_properties)
}


otu_info_merged <- rbind(otu_info$meco_mo16s_seqsg_raref,otu_info$meco_moits_seqsg_raref)
colnames(otu_info_merged) <- gsub("\\."," ",colnames(otu_info_merged))
otu_info_merged%>%
  flextable()%>%
  align(align = "center", part = "all")%>%
  save_as_image("./articlez/gen1/resultats/final/taxa_smry/otu_counts.png")
```

### Diversity

```{r}
bxplt_alpha_list <- NULL
flower_sr_list <- NULL
for (i in datasets){
  tmp <- get(i)
  trtmt <- tmp$sample_table$Trtmt
  Nb_fr <- tmp$sample_table$Nb_Fr
  sr <- hill_taxa(t(tmp$otu_table), q = 0, MARGIN = 1, base = exp(1))
  
  bxp_sr <- ggplot(data.frame(sr,trtmt),aes(x=trtmt,y=sr,color=trtmt))+
    geom_boxplot(outlier.shape = NA)+
    geom_jitter()+
    scale_color_manual(values=mo_col)+
    theme_classic2()+
    ylab(paste0("Observed ",ifelse(grepl("16",i),"bacterial","fungal")," OTUs richness"))+
    xlab("")+
    labs(color="Treatment")+
    theme(legend.key.size = unit(1.7, 'cm'),
          legend.title = element_text(size=14),
          legend.text = element_text(size=12))+
    guides(color=guide_legend(ncol=2,
                              title.hjust=0.5))
  
  df_fr_sr <- data.frame(sr,trtmt,Nb_fr)
  lm_eqn <- function(df){
    m <- lm(sr ~ Nb_fr, df_fr_sr);
    eq <- substitute(italic(y) == a + b %.% italic(x)*","~~italic(r)^2~"="~r2, 
                     list(a = format(unname(coef(m)[1]), digits = 2),
                          b = format(unname(coef(m)[2]), digits = 2),
                          r2 = format(summary(m)$r.squared, digits = 3)))
    as.character(as.expression(eq));
  }
  
  flower_sr_list[[i]] <- df_fr_sr%>%
    ggplot(aes(x=Nb_fr,y=sr,color=trtmt))+
    geom_point()+
    scale_color_manual(values=mo_col)+
    theme_classic2()+
    ylab(paste0("Observed ",ifelse(grepl("16",i),"bacterial","fungal")," OTUs richness"))+
    xlab("")+
    labs(color="Treatment")+
    theme(legend.key.size = unit(1.7, 'cm'),
          legend.title = element_text(size=14),
          legend.text = element_text(size=12))+
    guides(color=guide_legend(ncol=2,
                              title.hjust=0.5))+
    geom_smooth(method="lm",color="black")+
    annotate(geom = "text",label=paste0("pval: ",ifelse(gtools::stars.pval(summary(lm(data=df_fr_sr,sr~ Nb_fr))$coefficients[2,4])%in%c("."," "),"NS",                                      gtools::stars.pval(summary(lm(data=df_fr_sr,sr~ Nb_fr))$coefficients[2,4])),
                                        collapse = "\n"),
             x=min(Nb_fr)+(max(Nb_fr)-min(Nb_fr))*0.05,y=max(sr)+(max(sr)-min(sr))*0.3,hjust="left",vjust="Top",size=4)+
    annotate(geom="text",label=lm_eqn(df_fr_sr),x=min(Nb_fr)+(max(Nb_fr)-min(Nb_fr))*0.05,y=max(sr)+(max(sr)-min(sr))*0.25,hjust="left",vjust="Top",size=4,parse=T)
  
  
  bxp_sr <- cld_bxplot_wilcox(data = data.frame(sr,trtmt),
                              x = data.frame(sr,trtmt)$trtmt,
                              y =  data.frame(sr,trtmt)$sr,
                              plot = bxp_sr,
                              nudge = .05)
  
  
  shan <- hill_taxa(t(tmp$otu_table), q = 1, MARGIN = 1, base = exp(1))
  bxp_shan <- ggplot(data.frame(shan,trtmt),aes(x=trtmt,y=shan,color=trtmt))+
    geom_boxplot(outlier.shape = NA)+
    geom_jitter()+
    scale_color_manual(values=mo_col)+
    theme_classic2()+
    ylab(paste0(ifelse(grepl("16",i),"Bacterial","Fungal")," exponential Shannon entropy "))+
    xlab("")+
    theme(legend.position="none")
  
  bxp_shan <- cld_bxplot_wilcox(data = data.frame(shan,trtmt),
                                x = data.frame(shan,trtmt)$trtmt,
                                y =  data.frame(shan,trtmt)$shan,
                                plot = bxp_shan,
                                nudge = .05)
  
  invsimp <- hill_taxa(t(tmp$otu_table), q = 2, MARGIN = 1, base = exp(1))
  bxp_invsimp <- ggplot(data.frame(invsimp,trtmt),aes(x=trtmt,y=invsimp,color=trtmt))+
    geom_boxplot(outlier.shape = NA)+
    geom_jitter()+
    scale_color_manual(values=mo_col)+
    theme_classic2()+
    ylab(paste0(ifelse(grepl("16",i),"Bacterial","Fungal")," Gini-Simpson index"))+
    xlab("")+
    labs(color="Treatment")+
    theme(legend.position="none")
  
  bxp_invsimp <- cld_bxplot_wilcox(data = data.frame(invsimp,trtmt),
                                   x = data.frame(invsimp,trtmt)$trtmt,
                                   y =  data.frame(invsimp,trtmt)$invsimp,
                                   plot = bxp_invsimp,
                                   nudge = .05)
  
  
  bxplt_alpha <- (bxp_sr+theme(legend.position="none")|bxp_shan)/(bxp_invsimp|EcophyCofog::xtract_legend(bxp_sr))
  bxplt_alpha_list[[i]] <- bxplt_alpha
  ggsave(paste0("./articlez/gen1/resultats/final/alphadiv/hill_alpha_",i,".png"),bxplt_alpha,device = "png", width = 8, height = 8)
}
```

```{r}
flower_sr_list$meco_mo16s_seqsg_raref+flower_sr_list$meco_moits_seqsg_raref
```


```{r}

test_bac_rich <- kruskal.test(bxplt_alpha_list$meco_mo16s_seqsg_raref[[1]][[1]]$data$sr,bxplt_alpha_list$meco_mo16s_seqsg_raref[[1]][[1]]$data$trtmt)
test_bac_shan <- kruskal.test(bxplt_alpha_list$meco_mo16s_seqsg_raref[[1]][[2]]$data$shan,bxplt_alpha_list$meco_mo16s_seqsg_raref[[1]][[2]]$data$trtmt)


p1 <- bxplt_alpha_list$meco_mo16s_seqsg_raref[[1]][[1]]
p1 <- p1 + annotate("text",x=3,y=max(p1$data$sr*1.1),label=paste0("KW-Chi²=",round(test_bac_rich$statistic,digits=2)," | Df=",test_bac_rich$parameter," | p-value=",round(test_bac_rich$p.value,digits = 4)))

p2 <- bxplt_alpha_list$meco_mo16s_seqsg_raref[[1]][[2]]
p2 <- p2 + annotate("text",x=3,y=max(p2$data$shan*1.1),label=paste0("KW-Chi²=",round(test_bac_shan$statistic,digits=2)," | Df=",test_bac_shan$parameter," | p-value=",round(test_bac_shan$p.value,digits = 4)))


test_fun_rich <- kruskal.test(bxplt_alpha_list$meco_moits_seqsg_raref[[1]][[1]]$data$sr,bxplt_alpha_list$meco_moits_seqsg_raref[[1]][[1]]$data$trtmt)
test_fun_shan <- kruskal.test(bxplt_alpha_list$meco_moits_seqsg_raref[[1]][[2]]$data$shan,bxplt_alpha_list$meco_moits_seqsg_raref[[1]][[2]]$data$trtmt)


p3 <- bxplt_alpha_list$meco_moits_seqsg_raref[[1]][[1]]
p3 <- p3 + annotate("text",x=3,y=max(p3$data$sr*1.1),label=paste0("KW-Chi²=",round(test_fun_rich$statistic,digits=2)," | Df=",test_fun_rich$parameter," | p-value=",round(test_fun_rich$p.value,digits = 4)))

p4 <- bxplt_alpha_list$meco_moits_seqsg_raref[[1]][[2]]
p4 <- p4 + annotate("text",x=3,y=max(p4$data$shan*1.1),label=paste0("KW-Chi²=",round(test_fun_shan$statistic,digits=2)," | Df=",test_fun_shan$parameter," | p-value=",round(test_fun_shan$p.value,digits = 4)))

comp_alpha <- (p1|p3)/(p2|p4)
ggsave("./articlez/gen1/resultats/final/alphadiv/hill_alpha_comp.png",comp_alpha,width = 10,height = 8)
```

## Betadiv

### Data transformation


Hellinger transformation
```{r}
for(i in datasets){
  tmp <- get(i)
  tmp <- clone(tmp) # break link between tmp and og file
  tmp$tidy_dataset()
  tmp$otu_table <- t(t(labdsv::hellinger(tmp$otu_table)))
  tmp$sample_table %<>% mutate(nb_reads = colSums(tmp$otu_table),
                               nb_motus = colSums(tmp$otu_table>0))
  assign(paste0(i,"_hell"),tmp) 
}
```


CLR transformation
```{r}
for(i in datasets){
  tmp <- get(i)
  tmp <- clone(tmp) # break link between tmp and og file
  tmp$tidy_dataset()
  tmp$otu_table <- t(t(compositions::clr(tmp$otu_table)))
  tmp$sample_table %<>% mutate(nb_reads = colSums(tmp$otu_table),
                               nb_motus = colSums(tmp$otu_table>0))
  assign(paste0(i,"_clr"),tmp) 
}
```


### Hill betadiv

```{r}
beta_comb_list <- NULL
plot_cor_beta <- NULL
omg_pts <- NULL
wilcx_res <- NULL
for(i in datasets){
  tmp_data <- get(i)
  tmp_data <- clone(tmp_data)
  beta_hill <- hill_taxa_parti_pairwise(
    t(tmp_data$otu_table),
    q = 1,
    rel_then_pool = TRUE,
    output = c("data.frame", "matrix"),
    pairs =  "full",
    .progress = TRUE,
    show_warning = TRUE
  )
  
  trtmt_info <- tmp_data$sample_table[,c("sample_id","Trtmt")]
  names(trtmt_info) <- c("site1","trtmt")
  beta <- beta_hill %>% left_join(trtmt_info,beta16s_hill,by="site1")
  names(trtmt_info) <- c("site2","trtmt")
  beta <- beta %>% left_join(trtmt_info,beta,by="site2")
  
  sim_comp <- filter(beta,trtmt.x==trtmt.y)
  sim_comp_sitewise <- sim_comp %>% group_by(site1) %>% summarise(mean_beta = mean(TD_beta))
  names(trtmt_info) <- c("site1","trtmt")
  sim_comp_sitewise %<>% left_join(trtmt_info,by="site1")
  
  beta_bxpplot <- ggplot(sim_comp_sitewise,aes(x=trtmt,y=mean_beta,color=trtmt,shape=trtmt))+
    geom_boxplot(outlier.shape = NA)+
    geom_jitter(width=.25)+
    scale_color_manual(values = mo_col,name="Treatment")+
    scale_shape_manual(values = c(17,17,17,19,19,19),name="Treatment")+
    theme_classic2()+
    ylab("Within treatment average pairwise \u03B2diversity (q=1)")+
    xlab("Treatment")+
    ggtitle(ifelse(grepl("16s",i),"Bacteria","Fungi"))
  
  beta_bxpplot <- cld_bxplot_wilcox(data = sim_comp_sitewise,
                                    x = sim_comp_sitewise$trtmt,
                                    y =  sim_comp_sitewise$mean_beta,
                                    plot = beta_bxpplot,
                                    nudge = .07)
  beta_comb_list[[i]] <- beta_bxpplot
  
  tmp_data$sample_table%<>%
    dplyr::select(all_of(c(trait_list_glob)))%>%
    mutate(site1 = rownames(.))
  
  sim_comp_sitewise%<>%left_join(tmp_data$sample_table)
  for(j in trait_list_glob){
    
    lm_eqn <- function(df){
      m <- lm(mean_beta ~ sim_comp_sitewise[[col]], sim_comp_sitewise);
      eq <- substitute(italic(y) == a + b %.% italic(x)*","~~italic(r)^2~"="~r2, 
                       list(a = format(unname(coef(m)[1]), digits = 2),
                            b = format(unname(coef(m)[2]), digits = 2),
                            r2 = format(summary(m)$r.squared, digits = 3)))
      as.character(as.expression(eq));
    }
    
    col <- colnames(sim_comp_sitewise)[which(colnames(sim_comp_sitewise)==j)]
    plot_cor_beta[[i]][[j]] <- ggplot(sim_comp_sitewise,aes(x=.data[[col]],y=mean_beta))+
      geom_point(mapping=aes(color=trtmt,shape=trtmt))+
      scale_color_manual(values = mo_col,name='Treatments')+
      scale_shape_manual(values = c(17,17,17,19,19,19),name="Treatments")+
      theme_classic2()+
      ggtitle(ifelse(grepl("16s",i),"Bacteria","Fungi"))+
      annotate(geom = "text",label=paste0("pval: ",ifelse(gtools::stars.pval(summary(lm(data=sim_comp_sitewise,mean_beta~ sim_comp_sitewise[[col]]))$coefficients[2,4])%in%c("."," "),"NS",                                      gtools::stars.pval(summary(lm(data=sim_comp_sitewise,mean_beta~ sim_comp_sitewise[[col]]))$coefficients[2,4])),
                                          collapse = "\n"),
               x=min(sim_comp_sitewise[[col]])+(max(sim_comp_sitewise[[col]])-min(sim_comp_sitewise[[col]]))*0.05,y=max(sim_comp_sitewise$mean_beta)+(max(sim_comp_sitewise$mean_beta)-min(sim_comp_sitewise$mean_beta))*0.3,hjust="left",vjust="Top",size=4)+
      annotate(geom="text",label=lm_eqn(sim_comp_sitewise),x=min(sim_comp_sitewise[[col]])+(max(sim_comp_sitewise[[col]])-min(sim_comp_sitewise[[col]]))*0.05,y=max(sim_comp_sitewise$mean_beta)+(max(sim_comp_sitewise$mean_beta)-min(sim_comp_sitewise$mean_beta))*0.25,hjust="left",vjust="Top",size=4,parse=T)+
      ylim(c(min(sim_comp_sitewise$mean_beta)-(max(sim_comp_sitewise$mean_beta)-min(sim_comp_sitewise$mean_beta))*0.35,
             max(sim_comp_sitewise$mean_beta)+(max(sim_comp_sitewise$mean_beta)-min(sim_comp_sitewise$mean_beta))*0.35))+
      theme(axis.title=element_blank(),
            axis.text.y = element_blank())+
      scale_x_continuous(expand = c(0,0))+
      theme(legend.position = ifelse(grepl("16s",i),"n","right"))
    
    lmtmp <-summary(lm(mean_beta ~ sim_comp_sitewise[[col]], sim_comp_sitewise))
    if(lmtmp$coefficients[2,4]<0.05){
      plot_cor_beta[[i]][[j]] <-plot_cor_beta[[i]][[j]]+
        geom_smooth(method="lm",color="black")
    }
    
    
    grp_mean_df <- sim_comp_sitewise%>%
      group_by(trtmt)%>%
      summarize(mean_beta_grp=mean(mean_beta),
                mean_trait=mean(.data[[col]]))
    
    
    
    lims_dens_beta <-c(min(sim_comp_sitewise$mean_beta)-(max(sim_comp_sitewise$mean_beta)-min(sim_comp_sitewise$mean_beta))*0.35,
                       max(sim_comp_sitewise$mean_beta)+(max(sim_comp_sitewise$mean_beta)-min(sim_comp_sitewise$mean_beta))*0.35)
    lims_dens_traits <- c(min(sim_comp_sitewise[[col]])-(max(sim_comp_sitewise[[col]])-min(sim_comp_sitewise[[col]]))*0.01,
                          max(sim_comp_sitewise[[col]])+(max(sim_comp_sitewise[[col]])-min(sim_comp_sitewise[[col]]))*0.01)
    
    # get cld for traits
    grp_mean_df$coord_cld <- rev(seq(to=round(lims_dens_beta[2]),by=0.1,length.out=6))
    cld_vec <- pairwise_letterZ(sim_comp_sitewise,sim_comp_sitewise$trtmt,sim_comp_sitewise[[col]])
    grp_mean_df$cld <- cld_vec[[1]]
    wilcx_res[[i]][[col]] <- cld_vec[[2]]
    
    coord_cld_trait <- 
      sim_comp_sitewise%>%
      group_by(trtmt)%>%
      summarise(densimean_trait=mean(density(.data[[col]])$y),
                densimean_beta=mean(density(mean_beta)$y))
    grp_mean_df%<>%left_join(coord_cld_trait)
    
    # get cld for beta
    cld_vec_beta <- pairwise_letterZ(sim_comp_sitewise,sim_comp_sitewise$trtmt,sim_comp_sitewise$mean_beta)
    grp_mean_df$cld_beta <- cld_vec_beta[[1]]
    wilcx_res[[i]][["beta"]] <- cld_vec_beta[[2]]
    
    # plot densities
    dens_beta <-
      ggplot()+
      geom_density(sim_comp_sitewise,mapping=aes(y=mean_beta,color=trtmt,fill=trtmt),alpha=.4)+
      scale_color_manual(values=mo_col,name="Treatments")+
      scale_fill_manual(values=mo_col,name="Treatments")+
      ggrepel::geom_label_repel(grp_mean_df,mapping=aes(x=mean(densimean_beta),y=mean_beta_grp,label=cld_beta,color=trtmt),size=4,vjust=.5,hjust=.5)+
      theme_bw()+
      ylab(paste0("Within treatment average pairwise \u03B2diversity (q=1)"))+
      scale_x_reverse(expand=c(0,0))+
      theme(legend.position = "n",
            axis.title.x = element_blank())+
      scale_y_continuous(position='left',limits =lims_dens_beta)
    
    dens_trait <-   ggplot()+
      geom_density(sim_comp_sitewise,mapping=aes(x=.data[[col]],color=trtmt,fill=trtmt),alpha=.4)+
      scale_color_manual(values=mo_col,name="Treatments")+
      scale_fill_manual(values=mo_col,name="Treatments")+
      ggrepel::geom_label_repel(grp_mean_df,mapping=aes(x=mean_trait,y=mean(densimean_trait),label=cld,color=trtmt),size=4,vjust=.5,hjust=.5)+
      theme_bw()+
      scale_y_continuous(expand=c(0,0),position="right")+
      theme(legend.position = "n")+
      ylab("density")+
      scale_x_continuous(expand = c(0,0))+
      xlab(trait_df_name$cleannames[match(names(sim_comp_sitewise[col]),trait_df_name$rawnames)])
    
    
    pts <-dens_beta+plot_cor_beta[[i]][[j]]+plot_spacer()+dens_trait+plot_layout(guides = "collect",widths = c(1,5),heights = c(7,1))
    
    omg_pts[[i]][[j]] <- pts
    ggsave(filename = paste0("./articlez/gen1/resultats/final/betadiv/beta_traits_cor/",i,"/plot_",j,".png"),plot = pts,height = 6,width=7)
    
  }
  ggsave(beta_bxpplot,filename=paste0("./articlez/gen1/resultats/final/betadiv/betabxplt/beta_",i,"_hill.png"),device="png",width=6,height=5)
  
}

ggsave(beta_comb_list$meco_mo16s_seqsg_raref+beta_comb_list$meco_moits_seqsg_raref+plot_layout(guides="collect"),filename="./articlez/gen1/resultats/final/betadiv/betabxplt/beta_comb_hill.png",device="png",width=10,height=5)
```

```{r}
p2save <- omg_pts[["meco_mo16s_seqsg_raref"]][["Nb_Fr"]]|omg_pts[["meco_moits_seqsg_raref"]][["Nb_Fr"]]+plot_layout(guides = "collect")

p2save[[2]][[1]] <- p2save[[2]][[1]]+
  theme(axis.title.y = element_blank())

ggsave(filename = "./articlez/gen1/resultats/final/betadiv/betadiv2flowers.png",p2save,width = 12, height = 6)
```


### Vol2vol plots

```{r}
betadisp_plot <- NULL


for(i in datasets){
  tmp <- clone(get(paste0(i,"_hell")))
  tmp <- tmp
  dis2 <- t(tmp$otu_table)%>%vegdist(method = "horn",diag = T,upper = T)
  # names(dis2) <- gsub("^0|_.*_.*$","",names(dis2))
  mod2 <- betadisper(dis2,data_leaf[which(data_leaf$number%in%gsub("^0|_.*_.*$","",rownames(as.matrix(dis2)))),]$Trtmt)
  
  
  dis <- data_leaf[which(data_leaf$number%in%gsub("^0|_.*_.*$","",rownames(as.matrix(dis2)))),Nbcol]%>%vegdist(method="euclidean",diag = T,upper = T)
  mod <- betadisper(dis,data_leaf[which(data_leaf$number%in%gsub("^0|_.*_.*$","",rownames(as.matrix(dis2)))),]$Trtmt)
  
  df_plot <- data.frame(dis2centro_traits=mod$distances,
                        dis2centro_micro=mod2$distances,
                        plants=names(mod2$distances),
                        Trtmt=mod2$group)
  betadisp_plot[[paste0(i,"_hell")]] <- df_plot%>%
    ggplot(aes(x=dis2centro_traits,y=dis2centro_micro))+
    geom_point(aes(color=Trtmt,shape=Trtmt))+
    scale_color_manual(values=mo_col,name="Treatments", labels = levels(mod2$group))+
    scale_shape_manual(values = c(17,17,17,19,19,19), name = "Treatments", labels = levels(mod2$group))+
    theme_classic2()+
    xlab("Traits distance to group centroid")+
    ylab("Microbial distance to group centroid")+
    ggtitle(ifelse(grepl("16s",i),"Bacteria","Fungi"))+
    annotate(geom = "text",label=paste0("pval: ",ifelse(gtools::stars.pval(summary(lm(data=df_plot,dis2centro_micro~dis2centro_traits))$coefficients[2,4])%in%c("."," "),"NS",                                      gtools::stars.pval(summary(lm(data=df_plot,dis2centro_micro~dis2centro_traits))$coefficients[2,4])),
                                        collapse = "\n"),
             x=15,y=max(df_plot$dis2centro_micro)+(max(df_plot$dis2centro_micro)-min(df_plot$dis2centro_micro))*0.3,hjust="left",vjust="Top",size=4)+
    annotate(geom="text",label=lm_eqn(df_plot),x=15,y=max(df_plot$dis2centro_micro)+(max(df_plot$dis2centro_micro)-min(df_plot$dis2centro_micro))*0.25,hjust="left",vjust="Top",size=4,parse=T)
}

ggsave("./articlez/gen1/resultats/final/betadiv/betadisp_cor.png",betadisp_plot$meco_mo16s_seqsg_raref_hell+betadisp_plot$meco_moits_seqsg_raref_hell+plot_layout(guides="collect"),width=15,height =6)
```

### global betadisp cor trait

```{r}
globbetadisp_plot <- NULL

lm_eqn <- function(df){
  m <- lm(value ~ Nb_Fr, df);
  eq <- substitute(italic(y) == a + b %.% italic(x)*","~~italic(r)^2~"="~r2, 
                   list(a = format(unname(coef(m)[1]), digits = 2),
                        b = format(unname(coef(m)[2]), digits = 2),
                        r2 = format(summary(m)$r.squared, digits = 3)))
  as.character(as.expression(eq));
}

for(i in datasets){
  tmp <- clone(get(paste0(i,"_hell")))
  tmp <- tmp
  dis2 <- t(tmp$otu_table)%>%vegdist(method = "horn",diag = T,upper = T)
  mod2 <- betadisper(dis2,rep("dummy_group",nrow(tmp$sample_table)),type="median")
  
  names(mod2$distances) <- gsub("^0|_.*_.*$","",names(mod2$distances))

  
  marker <- ifelse(grepl("16s",i),"Bacteria","Fungi")
  if(marker=="Bacteria"){
    stripz <-ggh4x::strip_themed(text_x = ggh4x::elem_list_text(color="white",face="bold"),
                                 background_x = ggh4x::elem_list_rect(fill="#E09322"))
  }else{
    stripz <-ggh4x::strip_themed(text_x = ggh4x::elem_list_text(color="white",face="bold"),
                                 background_x = ggh4x::elem_list_rect(fill="#3F5E61"))
  }
  
  
  
  dplot <- data.frame(value=mod2$distances,
                      number=names(mod2$distances))%>%
    left_join(mutate(data_leaf,number=as.character(number)))
  
  
  smry_lm <-  summary(lm(data=dplot,value~Nb_Fr))$coefficients[2,4]
  
  plot_label <- paste0("pval: ",
                       ifelse(gtools::stars.pval(smry_lm)%in%c("."," "),
                              "NS",
                              gtools::stars.pval(smry_lm)),
                       collapse = "\n")
  
  nudge_label <- c(max(dplot$value)+(max(dplot$value)-min(dplot$value))*.15,
                   max(dplot$value)+(max(dplot$value)-min(dplot$value))*.1)
  
  limz <- c(min(dplot$value)-(max(dplot$value)-min(dplot$value))*0.2,
            max(dplot$value)+(max(dplot$value)-min(dplot$value))*0.2)
  
  p <- dplot %>%
    mutate(mark=marker)%>%
    ggplot(aes(x=Nb_Fr,y=value))+
    geom_point(aes(color=Trtmt,shape=Trtmt),size=2,alpha=.6)+
    scale_color_manual(values=mo_col,name="Treatments", labels = levels(data_leaf$Trtmt))+
    scale_shape_manual(values = c(17,17,17,19,19,19), name = "Treatments", labels = levels(data_leaf$Trtmt))+
    theme_bw()+
    ggh4x::facet_grid2(~ mark,strip = stripz)+
    annotate(geom = "text",label=plot_label,
             x=15,y=nudge_label[1],hjust="left",vjust="Top",size=3,fontface="bold")+
    annotate(geom="text",label=lm_eqn(dplot),x=15,y=nudge_label[2],hjust="left",vjust="Top",size=3,parse=T)+
    ylim(limz)+
    theme(axis.title=element_blank(),
          axis.text = element_blank(),
          axis.ticks = element_blank())+
    scale_x_continuous(expand = c(0.01,0.01))+
    theme(legend.position = ifelse(grepl("16s",i),"n","right"))
  
  
     if(smry_lm<0.05){
      p <- p+
        geom_smooth(method="lm",color="black")
    }
  
  
   grp_mean_df <- dplot%>%
    group_by(Trtmt)%>%
    summarise(mean_dist_grp=mean(value),
              mean_trait_grp=mean(Nb_Fr))
   
  # get cld for traits
  grp_mean_df$coord_cld <- rev(seq(from=round(limz[1],digits = 3),to=round(limz[2],digits = 3),length.out=6))
  cld_vec <- pairwise_letterZ(dplot,dplot$Trtmt,dplot$Nb_Fr)
  grp_mean_df$cld <- cld_vec[[1]]
  coord_cld_trait <- 
    dplot%>%
    group_by(Trtmt)%>%
    summarise(densimean_trait=mean(density(Nb_Fr)$y),
              densimean_beta=mean(density(value)$y))
  
  grp_mean_df%<>%left_join(coord_cld_trait)
  
  # get cld for beta
  cld_vec_beta <- pairwise_letterZ(dplot,dplot$Trtmt,dplot$value)
  grp_mean_df$cld_beta <- cld_vec_beta[[1]]

  # plot densities
  dens_beta <- ggplot()+
    geom_density(dplot,mapping=aes(y=value,color=Trtmt,fill=Trtmt),alpha=.4)+
    scale_color_manual(values=mo_col,name="Treatments")+
    scale_fill_manual(values=mo_col,name="Treatments")+
    ggrepel::geom_label_repel(grp_mean_df,
                              mapping=aes(x=mean(densimean_beta),
                                          y=mean_dist_grp,
                                          label=cld_beta,
                                          color=Trtmt),size=3,vjust=.5,hjust=.5)+
    theme_bw()+
    ylab("Distance to spatial median")+
    scale_x_reverse(expand=c(0,0))+
    theme(legend.position = "n",
          axis.title.x = element_blank(),
          axis.text.x = element_blank(),
          axis.ticks.x = element_blank())+
    scale_y_continuous(position='left',limits =limz)
  
  if(i=="meco_moits_seqsg_raref"){
    dens_beta <- dens_beta+theme(axis.title.y = element_blank())
  }
  
  
  dens_trait <-   ggplot()+
    geom_density(dplot,mapping=aes(x=Nb_Fr,color=Trtmt,fill=Trtmt),alpha=.4)+
    scale_color_manual(values=mo_col,name="Treatments")+
    scale_fill_manual(values=mo_col,name="Treatments")+
    ggrepel::geom_label_repel(grp_mean_df,mapping=aes(x=mean_trait_grp,y=mean(densimean_trait),label=cld,color=Trtmt),size=3,vjust=.5,hjust=.5)+
    theme_bw()+
    theme(legend.position = "n",
          axis.text.y = element_blank(),
          axis.ticks.y = element_blank(),
          axis.title.y = element_blank())+
    scale_x_continuous(expand = c(0.01,0.01))+
    scale_y_continuous(position="right",expand = c(0,0))+
    xlab("Number of flowers")
  
   globbetadisp_plot[[i]] <- dens_beta+p+
    plot_spacer()+dens_trait+plot_layout(guides = "collect",widths = c(1,5),heights = c(7,1))
  
}

blobbetadisp <- globbetadisp_plot$meco_mo16s_seqsg_raref|globbetadisp_plot$meco_moits_seqsg_raref+plot_layout(guides = "collect")

ggsave("./articlez/gen1/resultats/final/betadiv/globbetadisp_flowers.png",blobbetadisp,width = 9,height = 4)
```

###PCOA

```{r}
#check the hypothesis that dispersion is roughly the same between my groups in multivariate space
dis <- vegdist(t(meco_mo16s_seqsg_raref_hell$otu_table),method="horn") #we create a distance matrix between our individuals
mod <- betadisper(dis,meco_mo16s_seqsg_raref_hell$sample_table$Trtmt) #we calculate the multivarite dispersions i.e. average distance to centroids
anova(mod) #now we check that they are no difference between our groups i.e. our idnividuals have the same dispersion in the multivariate space

#p-value = 0.05119   non significative so our group dispersion is homogeneous between habitats
k = 1000
adon.res1 <- adonis2(t(meco_mo16s_seqsg_raref_hell$otu_table)~meco_mo16s_seqsg_raref_hell$sample_table$drought*meco_mo16s_seqsg_raref_hell$sample_table$substrate, perm = k, na.rm = T, method = "horn",p.adjust.methods="holm" ) 
adon.res1%>%
  flextable()%>%
  save_as_image(path="./articlez/gen1/resultats/final/betadiv/permanova_mubiota_16s.png")
```

```{r}
pair.adon.res1 <- pairwiseAdonis::pairwise.adonis(t(meco_mo16s_seqsg_raref_hell$otu_table),meco_mo16s_seqsg_raref_hell$sample_table$Trtmt,p.adjust.m = "holm", sim.method = "horn", perm=k)
pair.adon.res1%>%
  flextable()%>%
  save_as_image(path="./articlez/gen1/resultats/final/betadiv/pwisepermanova_mubiota_16s.png")
```

```{r}
#check the hypothesis that dispersion is roughly the same between my groups in multivariate space
dis <- vegdist(t(meco_moits_seqsg_raref_hell$otu_table),method="horn") #we create a distance matrix between our individuals
mod <- betadisper(dis,meco_moits_seqsg_raref_hell$sample_table$Trtmt) #we calculate the multivarite dispersions i.e. average distance to centroids
anova(mod) #now we check that they are no difference between our groups i.e. our idnividuals have the same dispersion in the multivariate space

#p-value = 0.1369  non significative so our group dispersion is homogeneous between habitats
k = 1000
adon.res1 <- adonis2(t(meco_moits_seqsg_raref_hell$otu_table)~meco_moits_seqsg_raref_hell$sample_table$drought*meco_moits_seqsg_raref_hell$sample_table$substrate, perm = k, na.rm = T, method = "horn",p.adjust.methods="holm" ) 
adon.res1%>%
  flextable()%>%
  save_as_image(path="./articlez/gen1/resultats/final/betadiv/permanova_mubiota_its.png")
```

```{r}
pair.adon.res1 <- pairwiseAdonis::pairwise.adonis(t(meco_moits_seqsg_raref_hell$otu_table),meco_moits_seqsg_raref_hell$sample_table$Trtmt,p.adjust.m = "holm", sim.method = "horn", perm=k)
pair.adon.res1%>%
  flextable()%>%
  save_as_image(path="./articlez/gen1/resultats/final/betadiv/pwisepermanova_mubiota_its.png")
```

```{r}


methods_dist <- c("horn","bray","jaccard")
transfo_list <- c("_hell","_clr")
pcoas_list <- NULL
pcoas_taxa_list <- NULL
ordi_list <- NULL
for(d in datasets){
  for(m in methods_dist){
    
    dtmp <- get(paste0(d,"_hell"))
    dtmp <- microeco::clone(dtmp) #clone data
    # dtmp$otu_table <- dtmp$otu_table[rowSums(dtmp$otu_table)>1,]#remove singletons
    orditmp <- phyloseq::ordinate(file2meco::meco2phyloseq(dtmp), "PCoA", m) # run pcoa with Morisita-Horn distance
    
    pcoa_sites <- phyloseq::plot_ordination(file2meco::meco2phyloseq(dtmp), orditmp,type = "sites",axes = c(1,2),color = "Trtmt")
    pcoa_taxa <- phyloseq::plot_ordination(file2meco::meco2phyloseq(dtmp), orditmp,type = "taxa",axes = c(1,2),color="Phylum_conf")
    
    orditmp$scores <- orditmp$vectors
    envfitted <- envfit(orditmp,
                        dtmp$sample_table[,c(trait_df_name$rawnames[-which(trait_df_name$rawnames=="C.N_leaf")])],
                        permutations=9999)
    
    ordi_list[[i]][[m]] <- orditmp
    
    expl1 <- orditmp$values$Relative_eig[1]*100
    expl2 <- orditmp$values$Relative_eig[2]*100
    
    en_coord_cont <- as.data.frame(scores(envfitted, "vectors")) * ggvegan:::arrow_mul(scores(envfitted, "vectors"),orditmp$vectors)
    en_coord_cont%<>%
      mutate(rawnames=rownames(.))%>%
      left_join(trait_df_name)%>%
      left_join(data.frame(rawnames=rownames(envfitted$vectors$arrows),
                           r2=envfitted$vectors$r,
                           psign=envfitted$vectors$pvals))
    
    df <- pcoa_sites$data
    gg <- ggplot(data = df, aes(x = Axis.1, y = Axis.2)) + 
      geom_point(data = df, aes(colour = Trtmt,shape=Trtmt), size = 3, alpha = 1) + 
      scale_colour_manual(values = mo_col) + 
      labs(colour = "Treatment")+
      ylab(paste0("Dim2 (",round(expl2,2),"%)"))+
      xlab(paste0("Dim1 (",round(expl1,2),"%)"))+
      scale_shape_manual(values=c(17,17,17,19,19,19),name="Treatment")+
      geom_convexhull(df,mapping=aes(x=Axis.1,y=Axis.2,fill = Trtmt, color = Trtmt),
                      alpha=0.1)+
      scale_fill_manual(values=mo_col,name="Treatment")
    
    centro_df <- df%>%
      group_by(Trtmt)%>%
      summarise(centro_x=mean(Axis.1),
                centro_y=mean(Axis.2))
    
    
    df_taxa <- pcoa_taxa$data
    gg_taxa <- ggplot(data = df_taxa, aes(x = Axis.1, y = Axis.2)) + 
      geom_point(data = df_taxa, aes(colour = Phylum_conf), size = 3, alpha = 1) + 
      ylab(paste0("Dim2 (",round(expl2,2),"%)"))+
      xlab(paste0("Dim1 (",round(expl1,2),"%)"))
    
    pcoas_taxa_list[[d]][[m]] <- gg_taxa
    ggsave(paste0("./articlez/gen1/resultats/final/betadiv/pcoa/taxa/pcoa_",d,"_",t,"_",m,"_",taxa,".svg"),pcoas_taxa_list[[d]][[m]],width = 8,height = 6,create.dir=T)
    
    for(all_traits_or_no in c("alltraits","notalltraits")){
      
      
      if(all_traits_or_no=="notalltraits"){
        en_coord_cont2 <- en_coord_cont%>%
          filter(psign<0.05)
      }else{
        en_coord_cont2 <- en_coord_cont
      }
      
      pcoas_list[[d]][[m]][[all_traits_or_no]] <- gg+
        geom_segment(aes(x = 0, y = 0, xend = Axis.1, yend = Axis.2), 
                     data = en_coord_cont2, linewidth =1, colour = "grey65",arrow = arrow(length = unit(0.2,"cm")),lty=ifelse(en_coord_cont2$psign<0.05,1,2)) +
        geom_point(centro_df,mapping=aes(x=centro_x,y=centro_y,color=Trtmt,shape=Trtmt),size=5)+
        ggrepel::geom_text_repel(data = en_coord_cont2, aes(x = Axis.1, y = Axis.2), 
                                 fontface = "bold",label=en_coord_cont2$cleannames,color=en_coord_cont2$colorz) + 
        
        ggnewscale::new_scale_color()+
        scale_color_manual(values=var_cat_col)+
        geom_vline(xintercept=0,lty=2)+
        geom_hline(yintercept=0,lty=2)+
        theme_classic2()+
        theme(axis.text = element_text(face="bold"),
              axis.title = element_text(face = "bold"))+
        ggtitle(ifelse(grepl("16s",d),"Bacterial communities","Fungal communities"))
      
      ggsave(paste0("./articlez/gen1/resultats/final/betadiv/pcoa/pcoa_",d,"_",t,"_",m,"_",all_traits_or_no,".svg"),pcoas_list[[d]][[m]][[all_traits_or_no]],width = 8,height = 6)
      
    }
  }
}

```

```{r}
scores(ordi_list$meco_mo16s_seqsg_raref$`_hell`$horn)%>%Vie
```


### Dist2centro ~ Nb flowers

```{r}
betadisp_nbflow_plot <- NULL
plot_dis2centro_nbfr <- NULL
lm_eqn <- function(df){
  m <- lm(value ~ Nb_Fr, df);
  eq <- substitute(italic(y) == a + b %.% italic(x)*","~~italic(r)^2~"="~r2, 
                   list(a = format(unname(coef(m)[1]), digits = 2),
                        b = format(unname(coef(m)[2]), digits = 2),
                        r2 = format(summary(m)$r.squared, digits = 3)))
  as.character(as.expression(eq));
}
wilcx_res2 <- NULL
for(d in datasets){
  
#cosmetics
  marker <- ifelse(grepl("16s",d),"Bacteria","Fungi")
  if(marker=="Bacteria"){
    stripz <-ggh4x::strip_themed(text_x = ggh4x::elem_list_text(color="white",face="bold"),
                                 background_x = ggh4x::elem_list_rect(fill="#E09322"))
  }else{
    stripz <-ggh4x::strip_themed(text_x = ggh4x::elem_list_text(color="white",face="bold"),
                                 background_x = ggh4x::elem_list_rect(fill="#3F5E61"))
  }
  
  for(t in transfo_list){
    
    for(m in methods_dist){
      if(m=="horn"){
        dtmp <- get(paste0(d,"_hell"))
      }else{
        dtmp <- get(d)
      }
      for(type in c("cmd_coord","betadisp")){
        
        if(type=="cmd_coord"){
          df_pcoa <- pcoas_list[[d]][[t]][[m]]$alltraits$layers[[1]]$data
          dplot <- df_pcoa %>%
            mutate(number=gsub("^0|_.*_.*$","",rownames(.)))%>%
            # dplyr::rename(Trtmt=trtmt)%>%
            group_by(Trtmt)%>%
            mutate(value=sqrt((PCoA1-mean(PCoA1))^2+(PCoA2-mean(PCoA2))^2))%>%
            left_join(mutate(data_leaf,number=as.character(number)))  
        }else{
          orditmp <- dtmp$otu_table
          orditmp <- vegdist(t(orditmp),method=m)
          betadisp_tmp <- betadisper(vegdist(t(orditmp),method=m),dtmp$sample_table$Trtmt)
          dplot <- betadisp_tmp$distances%>%
            reshape2::melt()%>%
            mutate(number=as.numeric(gsub("^0|_.*_.*$","",rownames(.))))%>%
            left_join(data_leaf[,c("number","Trtmt","Nb_Fr")])
          
        }
        
        smry_lm <-  summary(lm(data=dplot,value~Nb_Fr))$coefficients[2,4]
        
        plot_label <- paste0("pval: ",
                             ifelse(gtools::stars.pval(smry_lm)%in%c("."," "),
                                    "NS",
                                    gtools::stars.pval(smry_lm)),
                             collapse = "\n")
        
        nudge_label <- c(max(dplot$value)+(max(dplot$value)-min(dplot$value))*.15,
                         max(dplot$value)+(max(dplot$value)-min(dplot$value))*.1)
        
        limz <- c(min(dplot$value)-(max(dplot$value)-min(dplot$value))*0.2,
                  max(dplot$value)+(max(dplot$value)-min(dplot$value))*0.2)
        
        p <- dplot %>%
          mutate(mark=marker)%>%
          ggplot(aes(x=Nb_Fr,y=value))+
          geom_point(aes(color=Trtmt,shape=Trtmt),size=4,alpha=.6)+
          scale_color_manual(values=mo_col,name="Treatments", labels = levels(data_leaf$Trtmt))+
          scale_shape_manual(values = c(17,17,17,19,19,19), name = "Treatments", labels = levels(data_leaf$Trtmt))+
          theme_classic2()+
          ggh4x::facet_grid2(~ mark,strip = stripz)+
          annotate(geom = "text",label=plot_label,
                   x=15,y=nudge_label[1],hjust="left",vjust="Top",size=4,fontface="bold")+
          annotate(geom="text",label=lm_eqn(dplot),x=15,y=nudge_label[2],hjust="left",vjust="Top",size=4,parse=T)+
          ylim(limz)+
          theme(axis.title=element_blank(),
                axis.text.y = element_blank())+
          scale_x_continuous(expand = c(0.01,0.01))+
          theme(legend.position = ifelse(grepl("16s",d),"n","right"))
        
        
        if(smry_lm<0.05){
          p <- p+
            geom_smooth(method="lm",color="black")
        }
        
        grp_mean_df <- dplot%>%
          group_by(Trtmt)%>%
          summarise(mean_dist_grp=mean(value),
                    mean_trait_grp=mean(Nb_Fr))
        
        
        # get cld for traits
        grp_mean_df$coord_cld <- rev(seq(from=round(limz[1],digits = 3),to=round(limz[2],digits = 3),length.out=6))
        cld_vec <- pairwise_letterZ(dplot,dplot$Trtmt,dplot$Nb_Fr)
        grp_mean_df$cld <- cld_vec[[1]]
        wilcx_res2[[d]][[t]][[m]][[type]][["Nb_Fr"]] <-cld_vec[[1]]
        coord_cld_trait <- 
          dplot%>%
          group_by(Trtmt)%>%
          summarise(densimean_trait=mean(density(Nb_Fr)$y),
                    densimean_beta=mean(density(value)$y))
        
        grp_mean_df%<>%left_join(coord_cld_trait)
        
        # get cld for beta
        cld_vec_beta <- pairwise_letterZ(dplot,dplot$Trtmt,dplot$value)
        grp_mean_df$cld_beta <- cld_vec_beta[[1]]
        wilcx_res2[[d]][[t]][[m]][[type]][["beta"]] <- cld_vec_beta[[2]]
        
        # plot densities
        ylabel <- ifelse(grepl("coord",type),paste0("Distance to group centroid"),
                         paste0("Betadisper distances"))
        dens_beta <-    ggplot()+
          geom_density(dplot,mapping=aes(y=value,color=Trtmt,fill=Trtmt),alpha=.4)+
          scale_color_manual(values=mo_col,name="Treatments")+
          scale_fill_manual(values=mo_col,name="Treatments")+
          ggrepel::geom_label_repel(grp_mean_df,
                                    mapping=aes(x=mean(densimean_beta),
                                                y=mean_dist_grp,
                                                label=cld_beta,
                                                color=Trtmt),size=4,vjust=.5,hjust=.5)+
          theme_bw()+
          ylab(ylabel)+
          scale_x_reverse(expand=c(0,0))+
          theme(legend.position = "n",
                axis.title.x = element_blank())+
          scale_y_continuous(position='left',limits =limz)
        
        dens_trait <-   ggplot()+
          geom_density(dplot,mapping=aes(x=Nb_Fr,color=Trtmt,fill=Trtmt),alpha=.4)+
          scale_color_manual(values=mo_col,name="Treatments")+
          scale_fill_manual(values=mo_col,name="Treatments")+
          ggrepel::geom_label_repel(grp_mean_df,mapping=aes(x=mean_trait_grp,y=mean(densimean_trait),label=cld,color=Trtmt),size=4,vjust=.5,hjust=.5)+
          theme_bw()+
          theme(legend.position = "n")+
          ylab("density")+
          scale_x_continuous(expand = c(0,0))+
          scale_y_continuous(position="right")+
          xlab("Number of flowers")
        
        
        plot_dis2centro_nbfr[[d]][[t]][[m]][[type]] <-  dens_beta+p+
          plot_spacer()+dens_trait+plot_layout(guides = "collect",widths = c(1,5),heights = c(7,1))
      }
    }
  }
}
p1 <- plot_dis2centro_nbfr$meco_mo16s_seqsg_raref$bray
p2 <- plot_dis2centro_nbfr$meco_moits_seqsg_raref$bray

p3 <- plot_dis2centro_nbfr$meco_mo16s_seqsg_raref$`_hell`$horn$betadisp|plot_dis2centro_nbfr$meco_moits_seqsg_raref$`_hell`$horn$betadisp+plot_layout(guides = "collect")

p4 <- plot_dis2centro_nbfr$meco_mo16s_seqsg_raref$`_hell`$horn$cmd_coord|plot_dis2centro_nbfr$meco_moits_seqsg_raref$`_hell`$horn$cmd_coord+plot_layout(guides = "collect")

p5 <- plot_dis2centro_nbfr$meco_mo16s_seqsg_raref$`_hell`$bray$cmd_coord|plot_dis2centro_nbfr$meco_moits_seqsg_raref$`_hell`$bray$cmd_coord+plot_layout(guides = "collect")

p6 <- plot_dis2centro_nbfr$meco_mo16s_seqsg_raref$`_hell`$bray$betadisp|plot_dis2centro_nbfr$meco_moits_seqsg_raref$`_hell`$bray$betadisp+plot_layout(guides = "collect")

ggsave("./articlez/gen1/resultats/final/betadiv/arg.png",p3,height = 6,width = 12)
```

```{r}
dtmp$otu_table
```



## selected OTUs

```{r}
# load("./articlez/gen1/resultats/final/mixomics/diablo_rds.rds")

load("./articlez/gen1/resultats/final/mixomics/diablo_rds.rds")
```

```{r}
file_titan <- NULL
for(i in datasets){
  files_titan <- list.files(pattern="*",
                            path = paste0("./articlez/gen1/resultats/final/titan/",i,"/"),
                            full.names = T) # read the titan results
  f_max <- which.max(sapply(files_titan, function(x) nrow(readRDS(x)$sppmax))) # get the rds titan file with the highest number of rows un sppmax (highest number of otu)
  fmax_names <- rownames(readRDS(files_titan[f_max])$sppmax) # get rownames (otus)
  
  titan <-
    sapply(files_titan, function(f){
      x <- data.frame(readRDS(f)$sppmax) # get tabular result of singular otus results
      z <- matrix(rep(NA, length(fmax_names)),
                  ncol = 1, dimnames=list(fmax_names, NULL))
      decr <- subset(x, filter==1) # get negative response (decrease of the otu)
      incr <- subset(x, filter==2) # positive ones
      z[match(rownames(decr), rownames(z))] <- -decr$zscore # fill the column for the trait
      z[match(rownames(incr), rownames(z))] <- incr$zscore
      return(z)
    })
  length(fmax_names)==nrow(titan)
  rownames(titan) <- fmax_names # add otus as row names
  file_titan[[i]] <- titan
  }
```


## Diablo

```{r}
diablo_figz <- NULL
info_paper_diablo <- NULL
cutoff2use <- .4
for(i in datasets){
  # for(filteredornot in c("titancut","uncut")){    [[filteredornot]]
  for(t in gsub("_","",transfo_list)){
  ntuned <- diablo_mod[[i]][[t]][["nontuned"]]
  tuning <- diablo_mod[[i]][[t]][["tuning"]]
  final <- diablo_mod[[i]][[t]][["final"]]
  data_tmp <- get(i)
  data_tmp <- clone(data_tmp)
  pARR <- plotArrow(final,
                    ind.names = FALSE,
                    col.per.group = mo_col,
                    legend = TRUE,
                    title = "Arrow Plot")
  
  pIND <- plotIndiv(final, ind.names = FALSE, legend = TRUE, title = 'Data Integration',style='ggplot2',pch=20,ellipse=F,col.per.group = mo_col[c(3,1,5,4,2,6)])
  
  corcirvar <- plotVar(final, 
                       var.names = TRUE, 
                       style = 'graphics',
                       # cutoff=cutoff2use, 
                       legend = TRUE, 
                       pch = c(16, 17),
                       cex = c(1,0.8),
                       col = c('aquamarine3', 'darkorange2'))
  
  corcirvar%<>%
    mutate(names=rownames(.))%>%
    mutate(names=ifelse(grepl("MOTU",names),gsub("M","",names),names))%>%
    # filter(names%in%c(stabY[stabY$value>0.65,]$features,stabX[stabX$value>0.65,]$features))%>%
    dplyr::rename(cleannames=names)%>%
    left_join(trait_df_name)%>%
    mutate(var_cat=ifelse(is.na(var_cat),"otus",as.character(var_cat)))%>%
    mutate(var_cat=as.factor(var_cat))%>%
    mutate(wellrep = ifelse(abs(x)>cutoff2use|abs(y)>cutoff2use,"yes","no"))
  
  corcirplot_diablo <-  ggplot(corcirvar, aes(x = x, y = y)) +
    ggforce::geom_circle(aes(x0=0,y0=0,r=1),color="black")+
    # ggforce::geom_circle(aes(x0=0,y0=0,r=.33),color="black")+
    geom_point(data = filter(corcirvar,var_cat!="otus"&wellrep=="yes"),aes(color = var_cat),size=4,shape=17)+
    ggrepel::geom_text_repel(data =filter(corcirvar,var_cat!="otus"&wellrep=="yes"),aes(label=cleannames,color = var_cat))+
    scale_color_manual(values =c(biomass="burlywood3",
                                 nutrient= "darkcyan",
                                 photosynthesis="darkolivegreen4",
                                 structural="chocolate3",
                                 water="darkslateblue",
                                 reproduction="yellow3",
                                 otus= "darkred"), 
                       name="Variable categories")+ 
    guides(color = guide_legend(override.aes = aes(label = "")))+
    ggnewscale::new_scale_color()+
    geom_point(data = filter(corcirvar,var_cat=="otus"), # add arrows for variables
               aes(alpha=wellrep),size=3,shape=16,color="darkred") +
    scale_alpha_manual(values = c("no"=.3,"yes"=1),label = paste0(c("|cor|>","|cor|<"),cutoff2use), name = "OTUs")+
    xlim(c(-1,1))+
    ylim(c(-1,1))+
    geom_vline(xintercept = 0,linetype="dashed")+
    geom_hline(yintercept = 0,linetype="dashed")+
    theme_classic2()+
    ylab("Component 2")+
    xlab("Component 1")+
    coord_fixed()+
    ggtitle(ifelse(grepl("16s",i),"Bacteria","Fungi"))
  
  
  
  youch <- cimDiablo(final,
                     color = colorRampPalette(c("darkblue","#a30800"))(6),
                     color.Y = mo_col,
                     comp=c(1,2),
                     trim=F)
  
  data <- youch$mat[youch$row.names,youch$col.names]%>%
    as.data.frame()
  
  df_tmp <- youch$mat[youch$row.names,youch$col.names]%>%
    as.data.frame()%>%
    mutate(sample_id=rownames(.))%>%
    reshape2::melt()%>%
    left_join(data_tmp$sample_table[,c("sample_id","Trtmt")])%>%
    mutate(sample_id=str_extract(sample_id,"^[0-9]{2}"))%>%
    mutate(sample_id=fct_relevel(sample_id,str_extract(youch$row.names,"^[0-9]{2}")))%>%
    dplyr::rename(cleannames=variable)%>%
    left_join(trait_df_name)%>%
    mutate(var_cat=ifelse(is.na(var_cat),"otus",as.character(var_cat)))%>%
    mutate(var_cat=as.factor(var_cat))%>%
    mutate(var_cat=fct_relevel(var_cat,"biomass","nutrient","photosynthesis","structural","water","reproduction","otus"))%>%
    filter(cleannames%in%corcirvar[corcirvar$wellrep=="yes","cleannames"])%>%
    mutate(cleannames=as.factor(cleannames))%>%
    mutate(cleannames=fct_relevel(cleannames,c(youch$col.names)[which(youch$col.names%in%corcirvar[corcirvar$wellrep=="yes","cleannames"])]))
  
  
  tax_diablo <- data_tmp$tax_table[rownames(data_tmp$tax_table)%in%paste0("M",df_tmp$cleannames),] # extract taxonomic info about our OTUs
  
  tax_diablo %<>% 
    mutate(class=gsub("c__","",Class_conf))%>% #save class info
    unite(c(1:6),sep=";",col=Taxonomy)%>% # unite taxo info
    mutate(low_tax = gsub("(;[a-z]__)+?$","",Taxonomy))%>%
    mutate(low_tax = str_extract(pattern = "([a-z]__[^;]+?$)",string=low_tax))%>% # get lowest taxnomical info
    mutate(lvl_lowtax = str_extract(pattern = "[a-z]__",string=low_tax))%>% # get associated tax level
    mutate(low_tax = gsub("[a-z]__","",low_tax))%>%
    mutate(lvl_lowtax = ifelse(lvl_lowtax == "g__", "sp.", 
                               ifelse(lvl_lowtax == "f__", "(Family)",
                                      ifelse(lvl_lowtax == 'o__', "(Order)",
                                             ifelse(lvl_lowtax=="p__","(Phylum)",
                                                    ifelse(lvl_lowtax=="k__","(Kingdom)","(Class)"))))))%>%
    mutate(Var2=rownames(.)) #renames levels and get rownames
  
  df_tmp%<>%
    mutate(Var2=paste0("M",cleannames))

  df_tmp%<>%
    mutate(istitantoo=ifelse(Var2%in%rownames(file_titan[[i]]),"Yes","No"))%>%
    left_join(tax_diablo)

  df_tmp%<>%
    mutate(cleannames=as.character(cleannames))%>%
    mutate(cleannames=ifelse(var_cat=="otus",paste0(cleannames," ",low_tax," ",lvl_lowtax),cleannames))%>%
    mutate(cleannames=ifelse(istitantoo=="Yes",paste0("\u25cf ",cleannames),cleannames))

  var_order <-  df_tmp%>%
    group_by(var_cat,cleannames)%>%
    summarise(mean_val=mean(value))%>%
    ungroup()%>%
    arrange(var_cat,mean_val)%>%
    mutate(cleannames=fct_inorder(cleannames))
  
  
  
  value_scale_range <- ceiling(max(abs(range(df_tmp$value))))
  df_tmp%<>%
    arrange(Trtmt,sample_id)%>%
    mutate(sample_id=fct_inorder(sample_id))%>%
    mutate(cleannames=fct_relevel(cleannames,levels(var_order$cleannames)))%>%
    arrange(cleannames)
  
  
  p1 <- df_tmp%>%
    ggplot(aes(y=cleannames,x=sample_id,fill=value))+
    geom_tile(color = "white")+
    theme(axis.text.x = element_blank(), #element_text(angle = 90,vjust = .5,size=7)
          axis.text.y = element_text(size =8),
          axis.text = element_text(color="black"),
          axis.ticks.x = element_blank(),
          panel.background = element_blank(),
          legend.position = "right",
          axis.title = element_blank(),
          plot.margin = unit(c(0, 0, 0, 0), "lines"))+
    scale_fill_gradient2(low = "darkblue",mid = "lightyellow3", high = "#a30800",limits=c(-value_scale_range,value_scale_range))
  
  p2 <- df_tmp%>%
    ggplot(aes(x=1,y=cleannames,fill=var_cat))+
    geom_tile(color="white")+
    scale_fill_manual(values = c(biomass="burlywood3",
                                 nutrient="darkcyan",
                                 photosynthesis="darkolivegreen4",
                                 structural="chocolate3",
                                 water="darkslateblue",
                                 reproduction="yellow3",
                                 otus="darkred"), 
                      name="Variable categories")+
    theme_void()
  
  p3 <- df_tmp%>%
    ggplot(aes(x=sample_id,y=1,fill=Trtmt))+
    geom_tile(color="white")+
    scale_fill_manual(values = mo_col, 
                      name="Treatments")+
    theme_void()
  
  p_diablo <- p1+p2+p3+plot_spacer()+plot_layout(guides = "collect",widths = c(12,.2),heights = c(9,.2))
  
  
  diablo_figz[[i]][[t]][["hmap"]] <- p_diablo
  diablo_figz[[i]][[t]][["circ"]] <- corcirplot_diablo
  
  info_paper_diablo[[i]][[t]][["traits"]] <- corcirvar%>%filter(Block=="Leaf_traits")%>%dplyr::select(cleannames)
  
  info_paper_diablo[[i]][[t]][["ntraits"]] <-   corcirvar%>%filter(Block=="Leaf_traits")%>%nrow()
  
  info_paper_diablo[[i]][[t]][["otus"]] <- corcirvar%>%filter(Block=="OTUs_com")%>%dplyr::select(cleannames)
  
  info_paper_diablo[[i]][[t]][["notus"]] <- corcirvar%>%filter(Block=="OTUs_com")%>%nrow()
  
  info_paper_diablo[[i]][[t]][["interpratedotus"]] <-  dplyr::filter(corcirvar,var_cat=="otus"&wellrep=="yes")[,"cleannames"]
  
  # }
  }
}
```

```{r}
for(i in datasets){
  for(t in gsub("_","",transfo_list)){
    ggsave(paste0("./articlez/gen1/resultats/final/mixomics/diablo","_",i,"_",t,"_hmap.png"),diablo_figz[[i]][[t]][["hmap"]],width = 10,height = 11)
    ggsave(paste0("./articlez/gen1/resultats/final/mixomics/diablo","_",i,"_",t,"_circ.png"),diablo_figz[[i]][[t]][["circ"]],width = 7,height = 7)
  }
}
```

```{r}
notus_wellrep_list <- NULL
for(i in datasets){
  for(t in gsub("_","",transfo_list)){
    diablo_tmp <- diablo_figz[[i]][[t]]
    
    for(cutoffgradient in c(0,.1,.2,.3,.4,.5,.6)){
      
      notus <- diablo_tmp$circ$data%>%
        filter(var_cat=="otus")%>%
        filter(abs(x)>cutoffgradient|abs(y)>cutoffgradient)%>%
        nrow()
      notus_wellrep_list[[i]][[t]] <- c(notus_wellrep_list[[i]][[t]],
                                   notus)
      
    }
  }
}
df <- data.frame(marker=c(rep("Bacteria",14),
                          rep("Fungi",14)),
                 transfo=rep(c(rep("hell",7),
                               rep("clr",7)),
                               2),
                 coordinate_thrshld=rep(c(0,.1,.2,.3,.4,.5,.6),2),
                 notus=c(notus_wellrep_list$meco_mo16s_seqsg_raref$hell,
                         notus_wellrep_list$meco_mo16s_seqsg_raref$clr,
                         notus_wellrep_list$meco_moits_seqsg_raref$hell,
                         notus_wellrep_list$meco_moits_seqsg_raref$clr))

number_of_diabided_otus <- df%>%
  ggplot(aes(x=coordinate_thrshld,y=notus,fill=marker))+
  geom_bar(stat="identity",position = "dodge")+
  theme_classic2()+
  scale_fill_manual(values=c(Bacteria="#E09322",Fungi="#3F5E61"))+
  # scale_y_continuous(limits = c(0,360),expand = c(0,0))+
  scale_x_continuous(breaks = c(0,.1,.2,.3,.4,.5,.6))+
  theme(axis.text = element_text(face="bold"),
        axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1),
        axis.title.x = element_blank(),
        axis.title.y = element_text(face="bold"))+
  ylab("Number of diablo OTUs")+
  facet_wrap(.~transfo)
ggsave("./articlez/gen1/resultats/final/mixomics/idedotus_cutoff.png",number_of_diabided_otus)
```

```{r}
clust_wellrep_list <- NULL
max_height_list <- NULL
hclust_list <- NULL
plot_list <- NULL
for(i in datasets){
  for(t in gsub("_","",transfo_list)){
    diablo_tmp <- diablo_figz[[i]][[t]]
    
    for(cutoffgradient in c(0,.1,.2,.3,.4,.5)){
      
      print(paste0(i," & ",t," & ",cutoffgradient))
      notus <- diablo_tmp$circ$data%>%
        filter(var_cat=="otus")%>%
        filter(abs(x)>cutoffgradient|abs(y)>cutoffgradient)
      
      rownames(notus) <- notus$cleannames
      dist_coord <- notus[,c("x","y")]%>%dist(method="euclidean")
      clust_coord <- hclust(dist_coord)
      
      
      maxK <-ifelse(nrow(notus)<8,nrow(notus)-2,8)
      
      p <-fviz_nbclust(notus[,c("x","y")],FUNcluster = kmeans,k.max=maxK)+ggtitle(paste0("silhouette: ",cutoffgradient))
      p2<-fviz_nbclust(notus[,c("x","y")],FUNcluster = kmeans,k.max=maxK,method = "wss")+ggtitle(paste0("wss: ",cutoffgradient))
      p3<-fviz_nbclust(notus[,c("x","y")],FUNcluster = kmeans,k.max=maxK,method="gap_stat")+ggtitle(paste0("gapstat: ",cutoffgradient))
      
      clust_wellrep_list[[i]][[t]][[paste0("v",gsub("\\.","",cutoffgradient),"_nbclust")]] <- p/p3&theme(axis.title.y = element_blank())
      clust_wellrep_list[[i]][[t]][[paste0("v",gsub("\\.","",cutoffgradient),"_ggdendro")]] <- ggdendro::ggdendrogram(clust_coord)+ggtitle(paste0("ggdendro: ",cutoffgradient))+
        theme(axis.line.y = element_line(),
              axis.text.x = element_blank())
      
      max_height_list[[i]][[t]] <- c(max_height_list[[i]][[t]], max(clust_coord$height))
      hclust_list[[i]][[t]][[paste0("v",gsub("\\.","",cutoffgradient))]] <- clust_coord
      
      bp_tmp <-  NbClust::NbClust(notus[,c("x","y")],distance = "euclidean",method = "kmeans",min.nc = 2,max.nc = maxK)
      
      dfbp <-  bp_tmp$Best.partition%>%
        reshape2::melt()%>%
        mutate(cleannames=rownames(.),
               value=paste0("clust_",value))%>%
        dplyr::rename(cluster=value)
      
      dfbp <- diablo_tmp$circ$data%>%
        filter(var_cat=="otus")%>%
        left_join(dfbp)
      
      
      
      plot_list[[i]][[t]][[paste0("v",gsub("\\.","",cutoffgradient))]] <-   dfbp%>%
        ggplot(aes(x,y,color=cluster))+
        geom_point(alpha=.5)+
        xlim(c(-1,1))+
        ylim(c(-1,1))+
        ggtitle(paste0("threshold: ",cutoffgradient))
    }
  }
}
ggarrange(plotlist=clust_wellrep_list$meco_mo16s_seqsg_raref$hell)
ggarrange(plotlist=clust_wellrep_list$meco_moits_seqsg_raref$hell)


ggarrange(plotlist = plot_list$meco_mo16s_seqsg_raref$hell,legend = "none")
ggarrange(plotlist = plot_list$meco_moits_seqsg_raref$hell,legend = "none")

```

```{r}
for(i in datasets){
  for(t in gsub("_","",transfo_list)){
    ptmp <- diablo_figz[[i]][[t]]$circ$data%>%
      dplyr::filter(var_cat=="otus")%>%
      mutate(dis2origin=sqrt(x^2+y^2))%>%
      arrange(dis2origin)%>%
      dplyr::select(x,y,dis2origin)%>%
      gghistogram(x="dis2origin")+
      ggtitle(ifelse(grepl("16s",i),"Bacteria","Fungi"))
    ggsave(paste0("./articlez/gen1/resultats/final/mixomics/freq_dis2origin_",i,"_",t,".png"),ptmp)
  }
}
```

### Diablo with groups

```{r}
diablofigz_2 <- NULL
pcirc_list <- NULL
chsn_thrshld <- "v0" #represent all selected OTUS
for(i in datasets){
  for(t in gsub("_","",transfo_list)){
  
  ptest <- diablo_figz[[i]][[t]]$circ$data
  tmp <- get(i)
  tmp <- clone(tmp)
  ptest %<>%
    left_join(dplyr::select(plot_list[[i]][[t]][[chsn_thrshld]]$data,cleannames,cluster))
  
  # if(grepl("16s",i)){
  #   ptest%<>%
  #     mutate(otu_group=ifelse(x<0&y<0&wellrep=="yes","bac1",
  #                             ifelse(y>0.3&wellrep=="yes","bac2",
  #                                    ifelse(x>.4&wellrep=="yes","bac3","Not interprated"))))%>%
  #     mutate(otu_group=as.factor(otu_group))%>%
  #     mutate(otu_goup=fct_relevel(otu_group,"bac1","bac2","bac3","Not interprated"))
  # }else{
  #   ptest%<>%
  #     mutate(otu_group=ifelse(x<(-.15)&wellrep=="yes","fung1",
  #                             ifelse(x>(-0.15)&wellrep=="yes","fung2","Not interprated")))%>%
  #     mutate(otu_group=as.factor(otu_group))%>%
  #     mutate(otu_goup=fct_relevel(otu_group,"fung1","fung2","Not interprated"))
  # }
  if(grepl("16s",i)){
    ptest%<>%
      mutate(otu_group=ifelse(!is.na(cluster),
                              paste0("Bac cluster",gsub("clust_"," ",cluster)),
                              cluster))
    
    cluster_palette <- paletteer::paletteer_d("ochRe::winmar")[-6]%>%
      rev()
  }else{
    ptest%<>%
      mutate(otu_group=ifelse(!is.na(cluster),
                              paste0("Fung cluster",gsub("clust_"," ",cluster)),
                              cluster))
                                  
    cluster_palette <- colorzits <- rev(c("#F56455FF", "#2BAA92FF", "#15134BFF", "#9EC8FFFF", "#E9A800FF", "#00af66ff", "#ffebcc"))

  }
  
  pcirc <- ggplot(ptest, aes(x = x, y = y)) +
    ggforce::geom_circle(aes(x0=0,y0=0,r=1),color="black")+
    geom_point(data = filter(ptest,var_cat!="otus"), #&wellrep=="yes"
               aes(color = var_cat),
               size=4,
               shape=17)+
    ggrepel::geom_text_repel(data =filter(ptest,var_cat!="otus"), #&wellrep=="yes"
                             aes(label=cleannames,
                                 color = var_cat),fontface="bold")+
    scale_color_manual(values =c(biomass="burlywood3",
                                 nutrient= "darkcyan",
                                 photosynthesis="darkolivegreen4",
                                 structural="chocolate3",
                                 water="darkslateblue",
                                 reproduction="yellow3"), 
                       name="Variable categories")+ 
    guides(color = guide_legend(override.aes = aes(label = "")))+
    ggnewscale::new_scale_color()+
    geom_point(data = filter(ptest,var_cat=="otus"),
               aes(color=otu_group), #alpha=wellrep,
               size=3,
               shape=16) +
    scale_color_manual(values = cluster_palette,
                       name="Groups",
                       na.translate=F)+
    # scale_alpha_manual(values = c("no"=.3,"yes"=1),label = paste0(c("|cor|>","|cor|<"),cutoff2use), name = "OTUs")+
    xlim(c(-1,1))+
    ylim(c(-1,1))+
    geom_vline(xintercept = 0,linetype="dashed")+
    geom_hline(yintercept = 0,linetype="dashed")+
    theme_classic2()+
    ylab("Component 2")+
    xlab("Component 1")+
    coord_fixed()+
    ggtitle(ifelse(grepl("16s",i),"Bacteria","Fungi"))
  
  pcirc_list[[i]][[t]] <- pcirc
  youch <- cimDiablo(diablo_mod[[i]][[t]]$final,
                     color = colorRampPalette(c("darkblue","#a30800"))(6),
                     color.Y = mo_col,
                     comp=c(1,2),
                     trim=F)
  
  data <- youch$mat[youch$row.names,youch$col.names]%>%
    as.data.frame()
  
  
  
  df_tmp <- youch$mat[youch$row.names,youch$col.names]%>%
    as.data.frame()%>%
    mutate(sample_id=rownames(.))%>%
    reshape2::melt()%>%
    left_join(tmp$sample_table[,c("sample_id","Trtmt")])%>%
    mutate(sample_id=str_extract(sample_id,"^[0-9]{2}"))%>%
    mutate(sample_id=fct_relevel(sample_id,str_extract(youch$row.names,"^[0-9]{2}")))%>%
    dplyr::rename(cleannames=variable)%>%
    left_join(trait_df_name)%>%
    mutate(var_cat=ifelse(is.na(var_cat),"otus",as.character(var_cat)))%>%
    mutate(var_cat=as.factor(var_cat))%>%
    mutate(var_cat=fct_relevel(var_cat,"biomass","nutrient","photosynthesis","structural","water","reproduction","otus"))%>%
    filter(cleannames%in%ptest[ptest$wellrep=="yes","cleannames"])%>%
    mutate(cleannames=as.factor(cleannames))%>%
    mutate(cleannames=fct_relevel(cleannames,c(youch$col.names)[which(youch$col.names%in%ptest[ptest$wellrep=="yes","cleannames"])]))
  
  
  
  order_features <- hclust(vegdist(t(data[,which(colnames(data)%in%df_tmp$cleannames)]),'euclidean'))$order
  
  
  tax_diablo <- tmp$tax_table[rownames(tmp$tax_table)%in%paste0("M",df_tmp$cleannames),] # extract taxonomic info about our OTUs
  
  tax_diablo %<>% 
    mutate(class=gsub("c__","",Class_conf))%>% #save class info
    unite(c(1:6),sep=";",col=Taxonomy)%>% # unite taxo info
    mutate(low_tax = gsub("(;[a-z]__)+?$","",Taxonomy))%>%
    mutate(low_tax = str_extract(pattern = "([a-z]__[^;]+?$)",string=low_tax))%>% # get lowest taxnomical info
    mutate(lvl_lowtax = str_extract(pattern = "[a-z]__",string=low_tax))%>% # get associated tax level
    mutate(low_tax = gsub("[a-z]__","",low_tax))%>%
    mutate(lvl_lowtax = ifelse(lvl_lowtax == "g__", "sp.", 
                               ifelse(lvl_lowtax == "f__", "(Family)",
                                      ifelse(lvl_lowtax == 'o__', "(Order)",
                                             ifelse(lvl_lowtax=="p__","Phylum",
                                                    ifelse(lvl_lowtax=="k__","Kingdom","(Class)"))))))%>%
    mutate(Var2=rownames(.)) #renames levels and get rownames
  
  
  df_tmp%<>%
    mutate(Var2=paste0("M",cleannames))
  
  df_tmp%<>%
    mutate(istitantoo=ifelse(Var2%in%rownames(file_titan[[i]]),"Yes","No"))%>%
    left_join(tax_diablo)
  
  
  df_tmp%<>%left_join(ptest)
  
  df_tmp%<>%
    mutate(cleannames=as.character(cleannames))%>%
    mutate(cleannames=ifelse(var_cat=="otus",paste0(cleannames," ",low_tax," ",lvl_lowtax),cleannames))%>%
    mutate(cleannames=ifelse(istitantoo=="Yes",paste0("\u25cf ",cleannames),cleannames))
  
  var_order <-  df_tmp%>%
    group_by(var_cat,cleannames)%>%
    summarise(mean_val=mean(value))%>%
    ungroup()%>%
    arrange(var_cat,mean_val)%>%
    mutate(cleannames=fct_inorder(cleannames))
  
  
  
  value_scale_range <- ceiling(max(abs(range(df_tmp$value))))
  df_tmp%<>%
    arrange(Trtmt,sample_id)%>%
    mutate(sample_id=fct_inorder(sample_id))%>%
    mutate(cleannames=fct_relevel(cleannames,levels(var_order$cleannames)))%>%
    arrange(cleannames)
  
  df_tmp%<>%
    mutate(var_cat=ifelse(var_cat=='otus',otu_group,as.character(var_cat)))
  
  var_cat_order <- levels(as.factor(pcirc$data$otu_group))
  df_tmp%<>%mutate(var_cat=fct_relevel(var_cat,"biomass","nutrient","photosynthesis","structural","water","reproduction",var_cat_order))
  
  ordercleannames <- 
    unique(df_tmp$cleannames[order(-rowSums(sapply(seq_along(youch$col.names), function(i)
      i * grepl(youch$col.names[i], df_tmp$cleannames))), df_tmp$cleannames, decreasing = TRUE)])
  
  
  df_tmp%<>%
    mutate(cleannames=fct_relevel(cleannames,as.character(ordercleannames)[order_features]))
  
  
  p1 <- df_tmp%>%
    ggplot(aes(y=cleannames,x=sample_id,fill=value))+
    geom_tile(color = "white")+
    theme(axis.text.x = element_blank(), #element_text(angle = 90,vjust = .5,size=7)
          axis.text.y = element_text(size =8),
          axis.text = element_text(color="black"),
          axis.ticks.x = element_blank(),
          panel.background = element_blank(),
          legend.position = "right",
          axis.title = element_blank(),
          plot.margin = unit(c(0, 0, 0, 0), "lines"))+
    scale_fill_gradient2(low = "darkblue",mid = "#F9FAE8", high = "#a30800",limits=c(-value_scale_range,value_scale_range),na.value="white")
  
  
  names(cluster_palette) <- var_cat_order
  p2 <- df_tmp%>%
    ggplot(aes(x=1,y=cleannames,fill=var_cat))+
    geom_tile(color="white")+
    scale_fill_manual(values = c(biomass="burlywood3",
                                 nutrient="darkcyan",
                                 photosynthesis="darkolivegreen4",
                                 structural="chocolate3",
                                 water="darkslateblue",
                                 reproduction="yellow3",
                                 cluster_palette), 
                      name="Variable categories")+
    theme_void()
  
  p3 <- df_tmp%>%
    ggplot(aes(x=sample_id,y=1,fill=Trtmt))+
    geom_tile(color="white")+
    scale_fill_manual(values = mo_col, 
                      name="Treatments")+
    theme_void()
  
  
  p_diablo <- p1+p2+p3+plot_spacer()+plot_layout(guides = "collect",widths = c(12,.5),heights = c(9,.2))
  diablofigz_2[[i]][[t]] <- p_diablo
  
  }
}
```


```{r}
ggsave("./articlez/gen1/resultats/final/mixomics/diablo_16s_hmap_hell.png",diablofigz_2$meco_mo16s_seqsg_raref$hell)

ggsave("./articlez/gen1/resultats/final/mixomics/diablo_its_hmap_hell.png",diablofigz_2$meco_moits_seqsg_raref$hell)

```



```{r}
for(i in datasets){
  for(t in gsub("_","",transfo_list)){
    
    ggsave(paste0("./articlez/gen1/resultats/final/mixomics/diablo_test_",i,"_",t,"_hmap.png"),diablofigz_2[[i]][[t]],width = 10,height = 11)
    
    pcirc_list[[i]][[t]]$layers <- list(pcirc_list[[i]][[t]]$layers[[1]],
                                   pcirc_list[[i]][[t]]$layers[[4]],
                                   pcirc_list[[i]][[t]]$layers[[2]],
                                   pcirc_list[[i]][[t]]$layers[[3]],
                                   pcirc_list[[i]][[t]]$layers[[5]],
                                   pcirc_list[[i]][[t]]$layers[[6]])
    ggsave(paste0("./articlez/gen1/resultats/final/mixomics/diablo_test_",i,"_",t,"_circ.png"),pcirc_list[[i]][[t]],width = 7,height = 8)
  }
}
```


## otu_group DIABLO

### Relative abundances of my groups
```{r}
bxp_list <- NULL
for(t in gsub("_","",transfo_list)){ # for each transformation
  
# data for 16s
d16stmp <- meco_mo16s_seqsg_raref$otu_table%>%
  filter(rownames(.)%in%paste0("M",info_paper_diablo$meco_mo16s_seqsg_raref[[t]]$interpratedotus))%>%
  mutate(across(where(is.numeric),~(.x/sum(.))*100))%>%
  mutate(cleannames=gsub("M","",rownames(.)))%>%
  reshape2::melt()%>%
  dplyr::rename(relab=value)%>%
  left_join(pcirc_list$meco_mo16s_seqsg_raref[[t]]$data)%>%
  filter(relab!=0)%>%
  mutate(otu_group=as.factor(otu_group))

# data for its
ditstmp <- meco_moits_seqsg_raref$otu_table%>%
  filter(rownames(.)%in%paste0("M",info_paper_diablo$meco_moits_seqsg_raref[[t]]$interpratedotus))%>%
  mutate(across(where(is.numeric),~(.x/sum(.))*100))%>%
  mutate(cleannames=gsub("M","",rownames(.)))%>%
  reshape2::melt()%>%
  dplyr::rename(relab=value)%>%
  left_join(pcirc_list$meco_moits_seqsg_raref[[t]]$data)%>%
  filter(relab!=0)%>%
  mutate(otu_group=as.factor(otu_group))

# get colors
colorz16s <- rev(paletteer::paletteer_d("ochRe::winmar")[-6])
colorzits <- rev(c("#F56455FF", "#2BAA92FF", "#15134BFF", "#9EC8FFFF", "#E9A800FF", "#00af66ff", "#ffebcc"))

names(colorz16s) <- levels(d16stmp$otu_group)
names(colorzits) <- levels(ditstmp$otu_group)
colorz_binded <- c(colorz16s,colorzits)

# Bxp 16s
# run t test
stat.test <- d16stmp %>%
  mutate(relab=log10(relab))%>%
  rstatix::t_test(relab ~ otu_group) %>%
  rstatix::add_significance()

# set facet strip style
strips <- ggh4x::strip_themed(background_x =  ggh4x::elem_list_rect(fill=c("#E09322")),
                              text_x = ggh4x::elem_list_text(color="white",face="bold"))

# make bxp
bxp16s <- d16stmp%>%
  mutate(marker="Bacteria")%>%
  mutate(otu_group=factor(otu_group, levels = c(levels(otu_group), levels(ditstmp$otu_group))))%>%
  ggplot(aes(x=otu_group,y=relab))+
  geom_boxplot(aes(fill=otu_group),show.legend = T)+
  scale_fill_manual(values=  colorz_binded,
                    name="Diablo groups",
                    drop=F)+
  xlab("")+
  ylab("Relative abundance (%)")+
  theme_classic2()+
  scale_y_log10()+
  ggh4x::facet_wrap2(~marker,strip = strips)

# add stat
stat.test <- stat.test %>% rstatix::add_xy_position(x = "otu_group")
bxp16s <- bxp16s + stat_pvalue_manual(stat.test,tip.length = 0,bracket.shorten = 0.3,step.increase = .05)

# Bxp its
# run t test
stat.test <- ditstmp %>%
  mutate(relab=log10(relab))%>%
  rstatix::t_test(relab ~ otu_group) %>%
  rstatix::add_significance()

# set facet strip style
strips <- ggh4x::strip_themed(background_x =  ggh4x::elem_list_rect(fill=c(Fungi="#3F5E61")),
                              text_x = ggh4x::elem_list_text(color="white",face="bold"))

# mak bxp
bxpits <- ditstmp%>%
  mutate(marker="Fungi")%>%
  ggplot(aes(x=otu_group,y=relab))+
  geom_boxplot(aes(fill=otu_group))+
  scale_fill_manual(values =colorz_binded,
                    name="Fungal groups")+
  xlab("")+
  ylab("Relative abundance (%)")+
  theme_classic2()+
  scale_y_log10()+
  ggh4x::facet_wrap2(~marker,strip = strips)

# add stat
stat.test <- stat.test %>% rstatix::add_xy_position(x = "otu_group")
bxpits <- bxpits + stat_pvalue_manual(stat.test,tip.length = 0,bracket.shorten = 0.3,step.increase = .05)

bxp_list$p16s[[t]] <- bxp16s
bxp_list$pits[[t]] <- bxpits
}
p2s <- (bxp_list$p16s$hell+(bxp_list$pits$hell+theme(legend.position = "n",axis.text.y = element_blank(),axis.title.y = element_blank(),axis.line.y = element_blank(),axis.ticks.y = element_blank())))+plot_layout(guides = "collect")&theme(axis.text.x = element_blank(),text=element_text(face="bold"))
ggsave("./articlez/gen1/resultats/final/mixomics/diablo_groups_relab.png",p2s,width = 6,height = 3)

```

```{r}
bxp_trtmt_list <- NULL
otucountdiablo_smry <- NULL
for(t in gsub("_","",transfo_list)){

# data for 16s
d16stmp <- meco_mo16s_seqsg_raref$otu_table%>%
  filter(rownames(.)%in%paste0("M",info_paper_diablo$meco_mo16s_seqsg_raref[[t]]$interpratedotus))%>%
  mutate(across(where(is.numeric),~(.x/sum(.))*100))%>%
  mutate(cleannames=gsub("M","",rownames(.)))%>%
  reshape2::melt()%>%
  dplyr::rename(relab=value)%>%
  left_join(pcirc_list$meco_mo16s_seqsg_raref[[t]]$data)%>%
  filter(relab!=0)%>%
  mutate(otu_group=as.factor(otu_group))


d16stmp$marker <-"Bacteria"
d16stmp%<>%
  left_join(dplyr::rename(meco_mo16s_seqsg_raref$sample_table,variable=sample_id))


# data for its
ditstmp <- meco_moits_seqsg_raref$otu_table%>%
  filter(rownames(.)%in%paste0("M",info_paper_diablo$meco_moits_seqsg_raref[[t]]$interpratedotus))%>%
  mutate(across(where(is.numeric),~(.x/sum(.))*100))%>%
  mutate(cleannames=gsub("M","",rownames(.)))%>%
  reshape2::melt()%>%
  dplyr::rename(relab=value)%>%
  left_join(pcirc_list$meco_moits_seqsg_raref[[t]]$data)%>%
  filter(relab!=0)%>%
  mutate(otu_group=as.factor(otu_group))

ditstmp$marker <- "Fungi"
ditstmp%<>%
  left_join(dplyr::rename(meco_moits_seqsg_raref$sample_table,variable=sample_id))
dtmpmerged <- rbind(d16stmp,ditstmp)

# get colors
colorz16s <- rev(paletteer::paletteer_d("ochRe::winmar")[-6])
colorzits <- rev(c("#F56455FF", "#2BAA92FF", "#15134BFF", "#9EC8FFFF", "#E9A800FF", "#00af66ff", "#ffebcc"))

names(colorz16s) <- levels(d16stmp$otu_group)
names(colorzits) <- levels(ditstmp$otu_group)
colorz_binded <- c(colorz16s,colorzits)

# run kruskal test
stat.test <- dtmpmerged %>%
  mutate(relab=log10(relab))%>%
  group_by(marker,Trtmt)%>%
  rstatix::kruskal_test(relab ~ otu_group) %>%
  rstatix::add_significance()

#set up strips
strips <- ggh4x::strip_themed(background_x =  ggh4x::elem_list_rect(fill=c(Bacteria="#E09322",Fungi="#3F5E61")),
                              text_x = ggh4x::elem_list_text(color="white",face="bold"))

# bxp 16s
p16stmp <- dtmpmerged%>%
  filter(marker=="Bacteria")%>%
  ggplot(aes(x=Trtmt,y=relab,fill=otu_group))+
  geom_boxplot(outlier.shape = NA) +
  geom_point(shape=21,position = position_jitterdodge(),alpha=.8)+
  ggh4x::facet_wrap2(~marker,strip = strips)+
  scale_y_log10()+
  theme_classic2()+
  ylab("Relative abundance (%)")+
  xlab("")+
  scale_fill_manual(values= colorz_binded,
                    name="Diablo groups")+
  theme(legend.position = 'n')

#set up strips
strips <- ggh4x::strip_themed(background_x =  ggh4x::elem_list_rect(fill=c(Fungi="#3F5E61")),
                              text_x = ggh4x::elem_list_text(color="white",face="bold"))

# bxp its
pitstmp <- dtmpmerged%>%
  mutate(otu_group=as.factor(otu_group))%>%
  filter(marker=="Fungi")%>%
  ggplot(aes(x=Trtmt,y=relab,fill=otu_group))+
  geom_boxplot(outlier.shape = NA) +
  geom_point(shape=21,position = position_jitterdodge(),alpha=.8)+
  ggh4x::facet_wrap2(~marker,strip = strips)+
  scale_y_log10()+
  theme_classic2()+
  ylab("Relative abundance (%)")+
  xlab("")+
  scale_fill_manual(values= colorz_binded,
                    name="Diablo groups",
                    drop=F)

bxp_trtmt_list[["bac"]][[t]] <- p16stmp
bxp_trtmt_list[["its"]][[t]] <- pitstmp

# set up strips
strips <- ggh4x::strip_themed(background_x =  ggh4x::elem_list_rect(fill=c("#E09322","#3F5E61")),
                              text_x = ggh4x::elem_list_text(color="white",face="bold"))

# make barplot
otucountdiablo_smry[[t]] <- dtmpmerged%>%
  dplyr::select(cleannames,Trtmt,otu_group,marker)%>%
  unique()%>%
  group_by(marker,Trtmt,otu_group)%>%
  summarise(count=n())%>%
  ggplot(aes(x=Trtmt,y=count,fill=otu_group))+
  geom_bar(stat="identity",position = position_dodge())+
  scale_fill_manual(values=colorz_binded,
                    name="Diablo groups",
                    drop=F)+
  ggh4x::facet_wrap2(~marker,strip=strips)+
  theme_classic2()+
  scale_y_continuous(expand = c(0,0), breaks = breaks_pretty())+
  xlab("")+
  ylab("Number OTUs/treatment")+
  theme(axis.text = element_text(face="bold"),
        axis.title = element_text(face="bold"))

# save otu groups
tmp2save <- dtmpmerged[,c("cleannames","otu_group")]%>%
  unique()
write.csv(tmp2save,paste0("./articlez/gen1/resultats/final/otugroup_",t,".csv"),row.names = F)

}
# combine and save
diablo_smry_trtmt <- (bxp_trtmt_list$bac$hell+
    bxp_trtmt_list$its$hell+theme(axis.text.y = element_blank(),axis.title.y = element_blank(),legend.position = "none"))/otucountdiablo_smry$hell+plot_layout(guides="collect")&theme(text = element_text(face="bold"))
ggsave("./articlez/gen1/resultats/final/mixomics/countotugroups.png",diablo_smry_trtmt,width=6,height = 5)
```


```{r}
relab_otus_diablo <- NULL
for(t in gsub("_","",transfo_list)){

# data for 16s
d16stmp <- meco_mo16s_seqsg_raref$otu_table%>%
  mutate(across(where(is.numeric),~(.x/sum(.))*100))%>%
  mutate(cleannames=gsub("M","",rownames(.)))%>%
  reshape2::melt()%>%
  dplyr::rename(relab=value)%>%
  left_join(pcirc_list$meco_mo16s_seqsg_raref[[t]]$data)%>%
  filter(relab!=0)%>%
  mutate(otu_group=ifelse(wellrep=="yes",otu_group,NA))%>%
  mutate(otu_group=as.factor(otu_group))


d16stmp$marker <-"Bacteria"
d16stmp%<>%
  left_join(dplyr::rename(meco_mo16s_seqsg_raref$sample_table,variable=sample_id))


# data for its
ditstmp <- meco_moits_seqsg_raref$otu_table%>%
  mutate(across(where(is.numeric),~(.x/sum(.))*100))%>%
  mutate(cleannames=gsub("M","",rownames(.)))%>%
  reshape2::melt()%>%
  dplyr::rename(relab=value)%>%
  left_join(pcirc_list$meco_moits_seqsg_raref[[t]]$data)%>%
  filter(relab!=0)%>%
  mutate(otu_group=ifelse(wellrep=="yes",otu_group,NA))%>%
  mutate(otu_group=as.factor(otu_group))


ditstmp$marker <-"Fungi"
ditstmp%<>%
  left_join(dplyr::rename(meco_moits_seqsg_raref$sample_table,variable=sample_id))

dtmpmerged <- rbind(d16stmp,ditstmp)

# get colors
colorz16s <- rev(paletteer::paletteer_d("ochRe::winmar")[-6])
colorzits <- rev(c("#F56455FF", "#2BAA92FF", "#15134BFF", "#9EC8FFFF", "#E9A800FF", "#00af66ff", "#ffebcc"))

names(colorz16s) <- levels(d16stmp$otu_group)
names(colorzits) <- levels(ditstmp$otu_group)
colorz_binded <- c(colorz16s,colorzits)

# Add otu groups info
dtmp16s <- meco_mo16s_seqsg_raref$otu_table%>%
  mutate(otus=gsub("M","",rownames(.)))%>%
  dplyr::rename(cleannames=otus)%>%
  left_join(dplyr::select(filter(dtmpmerged,marker=="Bacteria"),cleannames,otu_group))%>%
  dplyr::mutate(otu_group=ifelse(is.na(otu_group),"not diablo",as.character(otu_group)))%>%
  dplyr::mutate(across(dplyr::where(is.numeric),~(.x/sum(.))*100))%>%
  reshape2::melt(id.vars=c("cleannames","otu_group"))%>%
  ungroup()

# Add Trtmt info
dtmp16s <- dtmp16s%>%
  left_join(dplyr::rename(meco_mo16s_seqsg_raref$sample_table,variable=sample_id))%>%
  dplyr::arrange(Trtmt,variable)%>%
  mutate(variable=as.factor(variable))

# Plot bac
relab_group_bac <- dtmp16s%>%
  mutate(variable=fct_inorder(variable))%>%
  dplyr:: select(otu_group,variable,value)%>%
  group_by(variable,otu_group)%>%
  summarise(value=sum(value))%>%
  mutate(marker="Bacteria")%>%
  ggplot(aes(x=variable,y=value,fill=otu_group))+
  geom_bar(position = "stack",stat="identity",width=1)+
  scale_fill_manual(values= c(colorz_binded,
                              'not diablo'="grey30"),
                    name="Diablo groups")+
  theme_classic2()+
  scale_y_continuous(expand=c(0,0))+
  ylab("Relative abundance (%)")+
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.title.x = element_blank(),
        text=element_text(face = "bold"))+
  ggh4x::facet_wrap2(~marker,strip = ggh4x::strip_themed(background_x = ggh4x::elem_list_rect(fill="#E09322"),text_x = ggh4x::elem_list_text(color="white")))

# Plot trtmt
trtmtbac <- dtmp16s%>%
  mutate(variable=fct_inorder(variable))%>%
  # dplyr:: select(otu_group,variable,value)%>%
  group_by(variable,otu_group)%>%
  ggplot(aes(x=variable,y=1,fill=Trtmt))+
  geom_tile()+
  scale_fill_manual(values=mo_col)+
  theme_void()

# Combine
relab_group_bac <- 
  (relab_group_bac/trtmtbac)+plot_layout(heights = c(9,1))

#ITS
# Add otu group info
dtmpits <- meco_moits_seqsg_raref$otu_table%>%
  mutate(otus=gsub("M","",rownames(.)))%>%
  dplyr::rename(cleannames=otus)%>%
  left_join(dplyr::select(filter(dtmpmerged,marker=="Fungi"),cleannames,otu_group))%>%
  dplyr::mutate(otu_group=ifelse(is.na(otu_group),"not diablo",as.character(otu_group)))%>%
  dplyr::mutate(across(dplyr::where(is.numeric),~(.x/sum(.))*100))%>%
  reshape2::melt(id.vars=c("cleannames","otu_group"))%>%
  ungroup()

# add treatment info
dtmpits <- dtmpits%>%
  left_join(dplyr::rename(meco_moits_seqsg_raref$sample_table,variable=sample_id))%>%
  dplyr::arrange(Trtmt,variable)%>%
  mutate(variable=as.factor(variable))

# plot fung
relab_group_fun <- dtmpits%>%
  mutate(variable=fct_inorder(variable))%>%
  dplyr:: select(otu_group,variable,value)%>%
  group_by(variable,otu_group)%>%
  summarise(value=sum(value))%>%
  mutate(marker="Fungi")%>%
  ggplot(aes(x=variable,y=value,fill=otu_group))+
  geom_bar(position = "stack",stat="identity",width=1,color=NA)+
  scale_fill_manual(values= c(colorz_binded,
                              'not diablo'="grey30"),
                    name="Diablo groups",
                    drop = TRUE)+
  theme_classic2()+
  scale_y_continuous(expand=c(0,0))+
  ylab("Relative abundance (%)")+
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text.y = element_blank(),
        text=element_text(face = "bold"))+
  ggh4x::facet_wrap2(~marker,strip = ggh4x::strip_themed(background_x = ggh4x::elem_list_rect(fill="#3F5E61"),text_x =  ggh4x::elem_list_text(color="white")))

# plot trtmt
trtmtfun <- dtmpits%>%
  mutate(variable=fct_inorder(variable))%>%
  # dplyr:: select(otu_group,variable,value)%>%
  group_by(variable,otu_group)%>%
  ggplot(aes(x=variable,y=1,fill=Trtmt))+
  geom_tile()+
  scale_fill_manual(values=mo_col)+
  theme_void()

relab_group_fun <- 
  (relab_group_fun/trtmtfun)+plot_layout(heights = c(9,1))

relab_otus_diablo[[t]] <- (relab_group_bac|relab_group_fun)+plot_layout(guides = "collect")
}
ggsave('./articlez/gen1/resultats/final/mixomics/relab_diab_group.png',relab_otus_diablo$hell,height =5,width = 8)

```

## Upset on OTU groups

```{r}
upset_list <- NULL
uplist <- NULL
for(d in datasets){
  data_tmp <- get(d)
  data_tmp <- clone(data_tmp)
  
  merged_tmp <-  data_tmp$merge_samples(use_group = "Trtmt")
  t1 <- trans_venn$new(dataset = merged_tmp)
  t1$res_names <- t1$res_names[c(3,2,1,6,5,4)]
  
  self <- t1
  colnumber <- self$colnumber
  ratio <- self$ratio
  res_names <- self$res_names
  switch_num <- colnumber - 1
  summary_table <- self$data_summary
  name_joint <- self$name_joint
  samplesum <- self$data_samplesum
  
  sample_levels <- res_names %>% rev
  for(s in c("sliced","notsliced")){
    plot_data <- summary_table %>% rownames_to_column %>% .[order(.$Counts, 
                                                                  decreasing = TRUE), ]
    upset_list[[d]] <- plot_data
    
for(t in gsub("_","",transfo_list)){
    
    get_otu_diab_groups <- pcirc_list[[d]][[t]]$data%>%
      filter(otu_group!="Not interprated")%>%
      mutate(cleannames=paste0("M",cleannames))%>%
      dplyr::select(cleannames,otu_group)%>%
      filter(grepl("OTU",cleannames))
    
    # get colors
    if(d==datasets[1]){
      colorz <- rev(paletteer::paletteer_d("ochRe::winmar")[-6])
      names(colorz) <- levels(as.factor(get_otu_diab_groups$otu_group))
    }else{
      
      colorz <- rev(c("#F56455FF", "#2BAA92FF", "#15134BFF", "#9EC8FFFF", "#E9A800FF", "#00af66ff", "#ffebcc"))
      names(colorz) <- levels(as.factor(get_otu_diab_groups$otu_group))
    }
    
    
    plot_data_complete <- t1$data_details
    
    plot_data_complete%<>%
      mutate(dumb=paste0("row_",rownames(.)))%>%
      reshape2::melt("dumb")%>%
      dplyr::rename(cleannames=value)%>%
      filter(cleannames!="")%>%
      left_join(get_otu_diab_groups)
    
    
    tot_counts <- plot_data$Counts%>%sum
    tot_ab <- plot_data$Abundance%>%sum
    most_ab <- max(plot_data$Abundance)
    plot_data%<>%
      filter(Counts!=0)%>%
      mutate(rowname=factor(rowname,levels=rowname))
    
    plot_data_complete%<>%
      mutate(variable=as.factor(variable))%>%
      mutate(variable=fct_relevel(variable,levels(plot_data$rowname)))%>%
      left_join(dplyr::rename(plot_data,variable=rowname))
    
    if(s=='sliced'){
      print(paste0((sum(plot_data$Counts[1:12])/sum(plot_data$Counts))*100," % of OTUs"))
      print(paste0((sum(plot_data$Abundance[1:12])/sum(plot_data$Abundance))*100," % of Reads"))
      
      plot_data %<>%
        dplyr::slice(1:12)%>%
        mutate(rowname=factor(rowname,levels=rowname))
      
      plot_data$rowname <- droplevels(plot_data$rowname)
      
      plot_data_complete %<>%
        filter(variable%in%plot_data$rowname)
      
      plot_data_complete$variable <- droplevels(plot_data_complete$variable)
    }else{
      plot_data %<>%
        mutate(rowname=factor(rowname,levels=rowname))
    }
    g1_grp <- plot_data_complete%>%
      mutate(otu_group=ifelse(is.na(otu_group),'not diablo',otu_group))%>%
      group_by(variable,otu_group)%>%
      summarise(Counts=n())%>%
      ggplot(aes(x = variable, y = Counts,fill=otu_group)) + 
      theme_classic() + 
      geom_bar(stat = "identity",position="stack",width=.5) + 
      ylab("Intersection size") + theme(axis.title.x = element_blank(), 
                                        axis.text.x = element_blank(),
                                        axis.ticks.x = element_blank()) + 
      theme(axis.text = element_text(size = 9), 
            axis.title = element_text(size = 12))+
      scale_fill_manual(values=c(colorz,
                              'not diablo'="grey70"),
                        name="Diablo groups")+
     theme( axis.text = element_text(face="bold"),
            axis.title = element_text(face="bold"))
  
    matrix_data <- matrix(nrow = colnumber, ncol = nrow(plot_data)) %>% 
      as.data.frame
    rownames(matrix_data) <- sample_levels
    colnames(matrix_data) <- plot_data[, 1]
    for (i in colnames(matrix_data)) {
      tmp <- strsplit(i, name_joint, fixed = TRUE) %>% unlist
      for (j in rownames(matrix_data)) {
        if (j %in% tmp) {
          matrix_data[j, i] <- 1
        }
      }
    }
    sample_long <- matrix_data %>% rownames_to_column %>% reshape2::melt(., 
                                                                         id.vars = "rowname")
    sample_long$variable %<>% factor(., levels = levels(plot_data[, 
                                                                  1]))
    sample_long$rowname %<>% factor(levels = sample_levels)
    sample_ture <- sample_long[!is.na(sample_long$value), ]
    sample_all <- sample_long
    sample_all$value <- 1
    g2 <- ggplot(sample_long, aes(x = variable, y = rowname))
    for (i in seq_len(length(unique(sample_ture$rowname)))) {
      if (i%%2 == 1) {
        g2 <- g2 + geom_rect(ymin = i - 0.5, ymax = i + 
                               0.5, xmin = -Inf, xmax = Inf, fill = "grey", 
                             colour =  "grey")
      }
      else {
        g2 <- g2 + geom_rect(ymin = i - 0.5, ymax = i + 
                               0.5, xmin = -Inf, xmax = Inf, fill = "azure2", 
                             colour = "azure2")
      }
    }
    g2 <- g2 + geom_point(aes(x = variable, y = rowname), data = sample_all, 
                          size = 3, color = "white", inherit.aes = FALSE) + 
      geom_point(aes(x = variable, y = rowname), data = sample_ture, 
                 size = 3, color = "black", 
                 inherit.aes = FALSE) + theme_bw() + theme(legend.position = "none") + 
      theme(axis.title = element_blank(), axis.text.x = element_blank(), 
            axis.ticks = element_blank()) + theme(axis.text = element_text(size = 10)) + 
      theme(panel.border = element_blank()) + theme(panel.grid = element_blank())
    line_data <- matrix_data
    line_data[] <- lapply(line_data, function(x) {
      x <- 1:length(x)
      x
    })
    line_data[is.na(matrix_data)] <- NA
    line_data2 <- data.frame(y = apply(line_data, 2, min, na.rm = TRUE), 
                             yend = apply(line_data, 2, max, na.rm = TRUE), x = 1:ncol(line_data), 
                             xend = 1:ncol(line_data))
    g2 <- g2 + geom_segment(aes(x = x, y = y, xend = xend, yend = yend), 
                            data = line_data2,
                            color="black")+
      theme(axis.text = element_text(face="bold"),
            axis.title = element_text(face="bold"))
    
    g3_data <- data.frame(number = samplesum, rowname = names(samplesum))
    g3_data$rowname %<>% factor(levels = sample_levels)
    g3 <- ggplot(g3_data, aes(x = number, y = rowname))
    for (i in seq_len(length(unique(g3_data$rowname)))) {
      if (i%%2 == 1) {
        g3 <- g3 + geom_rect(ymin = i - 0.5, ymax = i + 
                               0.5, xmin = -Inf, xmax = Inf, fill = "lightgrey", 
                             colour = "lightgrey")
      }
      else {
        g3 <- g3 + geom_rect(ymin = i - 0.5, ymax = i + 
                               0.5, xmin = -Inf, xmax = Inf, fill = "azure2", 
                             colour = "azure2")
      }
    }
    g3 <- g3 + geom_col(color =  mo_col[c(3,2,1,6,5,4)], fill =  mo_col[c(3,2,1,6,5,4)]) + 
      theme_bw() + theme(legend.position = "none") + scale_x_reverse(expand=c(0,10)) + 
      theme(axis.title = element_blank(), axis.text.y = element_blank(), 
            axis.ticks.y = element_blank()) + theme(axis.text = element_text(size = 8)) + 
      theme(panel.border = element_blank()) + theme(panel.grid = element_blank())+
      theme(axis.text = element_text(face="bold"),
            axis.title = element_text(face="bold"))
    
    zz <- data_tmp$otu_table%>%mutate(otus=rownames(.))%>%reshape2::melt()%>%filter(value!=0)%>%group_by(variable)%>%mutate(relab_sample=(value/sum(value))*100)
    
    list_relab <- NULL
    for(j in 1:ncol(t1$data_details)){
      
      tmp_relab <- data_tmp$otu_table%>%
        mutate(otus=rownames(.))%>%
        reshape2::melt(id.var="otus")%>%
        mutate(combination=colnames(t1$data_details)[j])%>%
        mutate(combination_status=ifelse(otus%in%t1$data_details[,j],"isin","notin"))
      
      list_relab  <- rbind(list_relab,tmp_relab)
    }
    
    list_relab%<>%
      left_join(zz)%>%
      filter(value!=0)
    
    plot_data %<>% mutate(cumab=cumsum(Abundance))
    plot_data%<>%mutate(intersect=rowname)
    plot_data%<>%mutate(cumrelab=(cumab/most_ab)*100)
    plot_data%<>%mutate(relab=(Abundance/tot_ab)*100)
    
    bxupset <- list_relab %>%
      filter(combination_status!="notin")%>%
      filter(combination%in%plot_data$rowname)%>%
      mutate(combination=fct_relevel(combination,levels(plot_data$rowname)))%>%
      ggplot(aes(x=combination,y=relab_sample))+
      geom_boxplot(fill="cadetblue4")+
      theme_classic2()+
      theme(axis.text.x = element_blank(),
            axis.title.x = element_blank(),
            axis.ticks.x = element_blank())+
      scale_y_log10(breaks = c(.5,1,5,10,20,40,60))+  #trans_breaks(identity, identity, n = 6)
      ylab("Relative abundance (%)")+ 
      theme(axis.text = element_text(face="bold"),
            axis.title = element_text(face="bold")) 
    
    
    layout <- "
#BBB
ACCC
#DDD
"
  up2save <- g3+theme(axis.title = element_blank())+g1_grp+scale_y_continuous(expand = c(0,0))+g2+theme(axis.title = element_blank())+bxupset+plot_layout(design = layout,heights = c(.3,.2,.3),widths = c(.3,.6))
  uplist[[d]][[t]][[s]]  <- up2save
  ggsave(up2save,filename=paste0("./articlez/gen1/resultats/final/betadiv/upset/bxupset_diablo_",s,"_",d,"_",t,".png"),device="png",width=ifelse(s=="sliced",8,12),height=6)
}
  }
}
```


### RankAb and diablo groups

```{r}
zbra <- clone(meco_mo16s_seqsg_raref)

fur_16s <- zbra$otu_table%>%
  mutate(across(.cols=1:ncol(.),.fns=~(.x/sum(.))*100))%>%
  mutate(otus=gsub("M","",rownames(.)))%>%
  reshape2::melt()%>%
  left_join(dplyr::rename(zbra$sample_table,variable=sample_id))%>%
  left_join(data.frame(sumotus=rowSums(zbra$otu_table),
                       prevotus=rowSums(zbra$otu_table>0),
                       otus=gsub("M","",rownames(as.data.frame(rowSums(zbra$otu_table))))))%>%
  mutate(loged_relab=log1p(value))%>%
  arrange(desc(prevotus))%>%
  mutate(otus=forcats::fct_inorder(otus))%>%
  group_by(Trtmt,otus)%>%
  summarise(mean_relab=mean(value),
            mean_loged_relab=mean(loged_relab))%>%
  ungroup()%>%
  filter(Trtmt%in%c('R-W','R-D'))%>%
  # mutate(water=ifelse(grepl("-D",Trtmt),"Dry","Wet"))%>%
  # group_by(water,otus)%>%
  # summarise(mean_relab=mean(mean_relab),
  # mean_loged_relab=mean(mean_loged_relab))%>%
  # ungroup()%>%
  ggplot(aes(x=otus,y=mean_loged_relab,color=Trtmt,shape=Trtmt))+
  geom_point(alpha=.7)+
  scale_color_manual(values = mo_col[c(3,6)],name="Treatments")+
  theme_classic2()+
  coord_trans(y='log1p')+
  ylab("Relative abundance (%)")+
  xlab("OTUs (ordered by prevalence)")+
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank())

zbra <- clone(meco_moits_seqsg_raref)
fur_its <- zbra$otu_table%>%
  mutate(across(.cols=1:ncol(.),.fns=~(.x/sum(.))*100))%>%
  mutate(otus=gsub("M","",rownames(.)))%>%
  reshape2::melt()%>%
  left_join(dplyr::rename(zbra$sample_table,variable=sample_id))%>%
  left_join(data.frame(sumotus=rowSums(zbra$otu_table),
                       prevotus=rowSums(zbra$otu_table>0),
                       otus=gsub("M","",rownames(as.data.frame(rowSums(zbra$otu_table))))))%>%
  mutate(loged_relab=log1p(value))%>%
  arrange(desc(prevotus))%>%
  mutate(otus=forcats::fct_inorder(otus))%>%
  group_by(Trtmt,otus)%>%
  summarise(mean_relab=mean(value),
            mean_loged_relab=mean(loged_relab))%>%
  ungroup()%>%
  filter(Trtmt%in%c('R-W','R-D'))%>%
  # mutate(water=ifelse(grepl("-D",Trtmt),"Dry","Wet"))%>%
  # group_by(water,otus)%>%
  # summarise(mean_relab=mean(mean_relab),
  #           mean_loged_relab=mean(mean_loged_relab))%>%
  # ungroup()%>%
  ggplot(aes(x=otus,y=mean_loged_relab,color=Trtmt,shape=Trtmt))+
  geom_point(alpha=.7)+
  scale_color_manual(values = mo_col[c(3,6)],name="Treatments")+
  theme_classic2()+
  coord_trans(y='log1p')+
  ylab("Relative abundance (%)")+
  xlab("OTUs (ordered by prevalence)")+
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank())

fur_16s+fur_its+plot_layout(guides="collect")
```

# fct DIABLO OTUS

```{r}
#ITS
# devtools::install_github("brendanf/FUNGuildR")
library(FUNGuildR)
tabtax_its <- meco_moits_seqsg_raref$tax_table%>%
  dplyr::filter(rownames(.)%in%paste0("M", info_paper_diablo$meco_moits_seqsg_raref$hell$otus$cleannames))

paste_noNA <- function(x,sep=", ") {
  gsub(", " ,sep, toString(x[!is.na(x) & x!="" & x!="NA"] ) ) }

sep<-";"
tabtax_its$Taxonomy <- apply( tabtax_its[ , c(1:7) ] , 1 , paste_noNA , sep=sep)

fns_its <- funguild_assign(tabtax_its, db = get_funguild_db(), tax_col = "Taxonomy")
fns_its_cl <- fns_its %>% mutate(confidenceRanking = ifelse(confidenceRanking=="Possible",NA,confidenceRanking))

meco_moits_seqsg_raref$taxfun_table <- fns_its_cl
rownames(meco_moits_seqsg_raref$taxfun_table) <- rownames(tabtax_its)

diab_grp <-
pcirc_list$meco_moits_seqsg_raref$hell$data%>%
  filter(cleannames%in%gsub("M","",rownames(tabtax_its)))%>%
  dplyr::rename(otus=cleannames)%>%
  dplyr::select(otus,otu_group)


# get colors

colorzits <- rev(c("#F56455FF", "#2BAA92FF", "#15134BFF", "#9EC8FFFF", "#E9A800FF", "#00af66ff", "#ffebcc"))
names(colorzits) <- levels(as.factor(diab_grp$otu_group))

pfct_fungi <- meco_moits_seqsg_raref$taxfun_table %>%
  mutate(otus=gsub("M","",rownames(.)))%>%
  reshape2::melt(id.vars="otus",measure.vars=c("trophicMode","guild","notes"))%>%
  ggplot(aes(x=variable,y=otus,label=value))+
  geom_tile(fill="lightgrey")+
  ggfittext::geom_fit_text(reflow = T)+
  scale_x_discrete(expand = c(0,0))+
  scale_y_discrete(expand = c(0,0))+
  theme(axis.text.x = element_text(angle = 90),
        axis.title = element_blank(),
        axis.line =element_blank(),
        panel.background = element_blank())+
  meco_moits_seqsg_raref$taxfun_table %>%
  mutate(Order_conf=ifelse(gsub("o__","",Order_conf)=="","Unkwown",gsub("o__","",Order_conf)))%>%
    mutate(otus=gsub("M","",rownames(.)))%>%
      left_join(diab_grp)%>%
  ggplot(aes(x=1,y=otus,fill=Order_conf))+
  geom_tile()+
  scale_x_discrete(expand = c(0,0))+
  scale_y_discrete(expand = c(0,0))+
  theme(axis.text = element_blank(),
        axis.title = element_blank(),
        axis.line =element_blank(),
        axis.ticks = element_blank(),
        panel.background = element_blank())+
  scale_fill_manual(values=colorRampPalette(brewer.pal(8,'Dark2'))(16))+
  meco_moits_seqsg_raref$taxfun_table %>%
  mutate(Order_conf=ifelse(gsub("o__","",Order_conf)=="","Unkwown",gsub("o__","",Order_conf)))%>%
    mutate(otus=gsub("M","",rownames(.)))%>%
      left_join(diab_grp)%>%
  ggplot(aes(x=1,y=otus,fill=otu_group))+
  geom_tile()+
  scale_x_discrete(expand = c(0,0))+
  scale_y_discrete(expand = c(0,0))+
  theme(axis.text = element_blank(),
        axis.title = element_blank(),
        axis.line =element_blank(),
        axis.ticks = element_blank(),
        panel.background = element_blank())+
    scale_fill_manual(values=colorzits)+
  plot_layout(widths = c(9,1,1),guides = "collect")

ggsave("./articlez/gen1/resultats/final/fct_fungi.png",pfct_fungi,width = 10,height = 8)

#16s
data_tmp <-meco_mo16s_seqsg_raref
data_tmp <- clone(data_tmp)
data_tmp$tax_table%<>%
  dplyr::filter(rownames(.)%in%paste0("M", info_paper_diablo$meco_mo16s_seqsg_raref$hell$otus$cleannames))
data_tmp$tidy_dataset()
func16s <- trans_func$new(data_tmp)
func16s$cal_spe_func(prok_database = "FAPROTAX")
func16sdf <- func16s$res_spe_func[,which(colSums(func16s$res_spe_func)>0)]


diab_grp <-
pcirc_list$meco_mo16s_seqsg_raref$hell$data%>%
  filter(cleannames%in%gsub("M","",rownames(func16sdf)))%>%
  dplyr::rename(otus=cleannames)%>%
  dplyr::select(otus,otu_group)


colorz16s <- rev(paletteer::paletteer_d("ochRe::winmar")[-6])

names(colorz16s) <- levels(as.factor(diab_grp$otu_group))



pfct_16s <- func16sdf[,order(colSums(func16sdf),decreasing = T)]%>%
  mutate(otus=gsub("M","",rownames(.)))%>%
  reshape2::melt(id.vars="otus")%>%
  ggplot(aes(x=variable,y=otus,fill=as.factor(value)))+
  geom_raster()+
  theme_classic()+
  scale_x_discrete(expand = c(0,0))+
  scale_y_discrete(expand = c(0,0))+
  theme(axis.text.x = element_text(angle = 90),
        axis.title = element_blank(),
        axis.line =element_blank(),
        panel.background = element_blank(),
        legend.position = "n")+
  scale_fill_manual(values = c("lightgrey","black"))+
  data_tmp$tax_table%>%
  mutate(Order_conf=ifelse(gsub("o__","",Order_conf)=="","Unkwown",gsub("o__","",Order_conf)))%>%
    mutate(otus=gsub("M","",rownames(.)))%>%
  ggplot(aes(x=1,y=otus,fill=Order_conf))+
  geom_tile()+
  scale_x_discrete(expand = c(0,0))+
  scale_y_discrete(expand = c(0,0))+
  theme(axis.text = element_blank(),
        axis.title = element_blank(),
        axis.line =element_blank(),
        axis.ticks = element_blank(),
        panel.background = element_blank())+
  scale_fill_manual(values=colorRampPalette(brewer.pal(8,'Dark2'))(21))+
  data_tmp$tax_table %>%
  mutate(Order_conf=ifelse(gsub("o__","",Order_conf)=="","Unkwown",gsub("o__","",Order_conf)))%>%
    mutate(otus=gsub("M","",rownames(.)))%>%
      left_join(diab_grp)%>%
  ggplot(aes(x=1,y=otus,fill=otu_group))+
  geom_tile()+
  scale_x_discrete(expand = c(0,0))+
  scale_y_discrete(expand = c(0,0))+
  theme(axis.text = element_blank(),
        axis.title = element_blank(),
        axis.line =element_blank(),
        axis.ticks = element_blank(),
        panel.background = element_blank())+
    scale_fill_manual(values=colorz16s)+
  plot_layout(widths = c(9,1,1),guides = "collect")
ggsave("./articlez/gen1/resultats/final/fct_bact.png",pfct_16s,width = 10,height = 8)

```


### Info on slctd otus

```{r}
for(i in datasets){
  data_tmp <- get(i)
  data_tmp <- clone(data_tmp)
  
  taxo_tmp <- data_tmp$tax_table%>%
    mutate(OTUs=gsub("M","",rownames(.)))%>%
    filter(OTUs%in% info_paper_diablo[[i]]$hell$otus$cleannames)
  
  taxo_tmp$sequences <- gsub("s__","",data_tmp$ori_tax$sequence[which(rownames(data_tmp$ori_tax)%in%paste0("M",info_paper_diablo[[i]]$hell$otus$cleannames))])
  taxo_tmp$reads <- data_tmp$otu_table[which(rownames(data_tmp$otu_table)%in%paste0("M",info_paper_diablo[[i]]$hell$otus$cleannames)),]%>%
    rowSums()
  taxo_tmp$prevalence <- rowSums(data_tmp$otu_table[which(rownames(data_tmp$otu_table)%in%paste0("M",info_paper_diablo[[i]]$hell$otus$cleannames)),]>0)
  
  
  if(grepl("16s",i)){
    fct_df <- func16sdf[,order(colSums(func16sdf),decreasing = T)]%>%
      mutate(OTUs=gsub("M","",rownames(.)))
  }else{
    fct_df <- meco_moits_seqsg_raref$taxfun_table %>%
      mutate(OTUs=gsub("M","",rownames(.)))
  }
  
  taxo_tmp%<>%left_join(fct_df)
  taxo_tmp%<>%left_join(dplyr::rename(dplyr::select(filter(pcirc_list[[i]]$hell$data,grepl("OTU",cleannames)),cleannames,otu_group),OTUs=cleannames))
  
  taxo_tmp%<>%
    relocate(OTUs,otu_group)
  
  write.csv(taxo_tmp,paste0("./articlez/gen1/resultats/final/otu_info_",i,".csv"),row.names = F)
}
```

## CV traits

```{r}
cv_list <- data.frame()
for(var in Nbcol){
  
  
  unitofvar <- trait_df_name$unit[match(names(data_leaf)[var],trait_df_name$rawnames)][[1]]
  current_var <-trait_df_name$cleannames[match(names(data_leaf)[var],trait_df_name$rawnames)]
  
  cv_tmp <- (sd(abs(data_leaf[,var]))/mean(abs(data_leaf[,var])))*100
  cv_tmp <- data.frame(cleannames=current_var,cv=cv_tmp)
  cv_list <- rbind(cv_list,cv_tmp)
}

cv_list <- rbind(cv_list,data.frame(cleannames=c("PCA Dim1",
                                                 "PCA Dim2"),
                                    cv=c(sd(abs(PCA_plant_trait$ind$coord[,1]))/mean(abs(PCA_plant_trait$ind$coord[,1]))*100,
                                    sd(abs(PCA_plant_trait$ind$coord[,2]))/mean(abs(PCA_plant_trait$ind$coord[,2]))*100)))

t0 <- cv_list%>%
  left_join(trait_df_name)%>%
  arrange(cv)%>%
  mutate(cleannames=fct_inorder(cleannames))%>%
  filter(cleannames!="C/N")%>%
  mutate(var_cat=replace_na(var_cat,'PCA dimensions'))


t1 <- titan_info_papers$meco_mo16s_seqsg_raref$hell$dontrm$var_cat$notonly$otupertrait%>%
  reshape2::melt()%>%
  mutate(cleannames=rownames(.),
         marker="Bacteria")%>%
  mutate(cleannames=ifelse(grepl("PCA1",cleannames),"PCA Dim1",
         ifelse(grepl("PCA2",cleannames),"PCA Dim2",cleannames)))
t2 <- titan_info_papers$meco_moits_seqsg_raref$hell$dontrm$var_cat$notonly$otupertrait%>%
  reshape2::melt()%>%
  mutate(cleannames=rownames(.),
         marker="Fungi")%>%
  mutate(cleannames=ifelse(grepl("PCA1",cleannames),"PCA Dim1",
         ifelse(grepl("PCA2",cleannames),"PCA Dim2",cleannames)))
t3 <- rbind(t1,t2)
t3%<>%mutate(cleannames=fct_relevel(cleannames,levels(t0$cleannames)))

plot_cv <- ggplot()+
  geom_bar(data=t0,mapping=aes(x=cleannames,y=cv,fill=var_cat),stat = "identity")+
  scale_fill_manual(values =c(biomass="burlywood3",
                              nutrient= "darkcyan",
                              photosynthesis="darkolivegreen4",
                              structural="chocolate3",
                              water="darkslateblue",
                              reproduction="yellow3",
                              `PCA dimensions`="grey40"),
                    name='Variable categories')+
  theme_classic2()+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+
  ylab("Variable CV (%)")+
  ggnewscale::new_scale_fill()+
  geom_point(data=t3,mapping = aes(x=cleannames,y=value*5,fill=marker,shape=marker),size=4)+
  geom_point()+
  scale_shape_manual(values=c(21,23),name="Group")+
  scale_fill_manual(values =c("#E09322",
                              "#3F5E61"),
                    name="Group")+
  scale_y_continuous(expand = c(0,0),sec.axis = sec_axis(~(.)/5,name="Number of TITAN OTUs"))+
  theme(text=element_text(face="bold"))

ggsave("./articlez/gen1/resultats/final/traits/cv_titan.png",plot_cv,height = 4,width = 6.5)
```


# Save env

```{r}
save.image(file='./articlez/gen1/resultats/final/myEnvironment.RData')
```

## MFD

```{r}
data("fruits_traits", package = "mFD")

dtmp <- meco_mo16s_seqsg_raref_hell
otu_tmp <- dtmp$otu_table #get coms
otu_tmp <- 1*(otu_tmp>0)# convert to p/a
otu_tmp <- otu_tmp[rowSums(otu_tmp)>0.05*ncol(otu_tmp),]
otu_space <- otu_tmp %>%
  reshape2::melt() %>% # melt
  filter(value!=0) %>% # filter when absent
  dplyr::rename(otu_id=Var1,
         sample_id=Var2) %>% 
  left_join(dtmp$sample_table)%>% # add traits info
  dplyr::select(all_of(c("otu_id","sample_id","Trtmt",trait_list_glob))) %>%
  group_by(otu_id)%>%
  summarise(samplez=paste0(unique(sample_id),collapse = "|"), # samplez in which OTU is found
            trtmtz=paste0(unique(Trtmt),collapse = "|"), # trtmtz in which OTU is found
            across(where(is.numeric),~median(.x))) %>% # median of host traits
  as.data.frame()
rownames(otu_space) <- otu_space$otu_id

mfd_coms <- t(labdsv::hellinger(tmp$otu_table))

mfd_traits <- data.frame(trait_name=trait_list_glob)%>%
  mutate(trait_type="Q",
         trait_weight=1,
         fuzzy_name=trait_list_glob)
  
# OTUs env distance
OTUs_dist_env <- mFD::funct.dist(
  sp_tr         = otu_space[,-c(1:3)],
  tr_cat        = mfd_traits,
  metric        = "euclidean",
  scale_euclid  = "scale_center",
  ordinal_var   = "classic",
  weight_type   = "equal",
  stop_if_NA    = TRUE)

# space quality
fspaces_quality_OTus <- mFD::quality.fspaces(
  sp_dist             = OTUs_dist_env,
  maxdim_pcoa         = 19,
  deviation_weighting = c("absolute", "squared"),
  fdist_scaling       = c(TRUE,FALSE),
  fdendro             = "average")

fspaces_metrics_qual <- fspaces_quality_OTus$"quality_fspaces" %>%
  tibble::as_tibble(rownames = "Funct.space") %>%
  tidyr::pivot_longer(cols =! Funct.space, names_to = "quality_metric", values_to = "Quality") %>%
  mutate(Funct.space=gsub("pcoa_","PC",Funct.space))%>%
  mutate(Funct.space=gsub("d","",Funct.space))%>%
  mutate(Funct.space=factor(Funct.space,levels=c(paste0("PC",1:19),"tree_average")))%>%
  ggplot2::ggplot(ggplot2::aes(x = Funct.space, y = Quality, 
                               color = quality_metric, shape = quality_metric)) +
  ggplot2::geom_point()+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+
    geom_hline(yintercept = 0.1,color="red",linetype=2)

fspace_eigen <- 
  data.frame(eigen_rel=fspaces_quality_OTus$details_fspaces$pc_eigenvalues$Relative_eig,
             Dimension=factor(c(paste0("PC",1:19)),levels=c(paste0("PC",1:19))))%>%
  ggplot(aes(y=(eigen_rel)*100,x=Dimension))+
  geom_bar(stat="identity")+
  theme_bw()+
  geom_hline(yintercept = (1/19)*100,color="red",linetype=2)+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1),
        axis.title.x = element_blank())+
  ylab("Relative eigenvalues (%)")

fspaces_quality_plots <- mFD::quality.fspaces.plot(
  fspaces_quality            = fspaces_quality_OTus,
  quality_metric             = "mad",
  fspaces_plot               = c("tree_average", "pcoa_2d", "pcoa_4d", 
                                 "pcoa_6d", "pcoa_8d", "pcoa_10d"),
  name_file                  = NULL,
  range_dist                 = NULL,
  range_dev                  = NULL,
  range_qdev                 = NULL,
  gradient_deviation         = c(neg = "darkblue", nul = "grey80", pos = "darkred"),
  gradient_deviation_quality = c(low = "yellow", high = "red"),
  x_lab                      = "Trait-based distance")

# visualise traits - axes relation



faxes_coord_otus <- fspaces_quality_OTus$"details_fspaces"$"sp_pc_coord"
ftraits_values_otus <- otu_space[,-c(1:3)]

mFD:::check.sp.tr(ftraits_values_otus, stop_if_NA = stop_if_NA)
mFD:::check.sp.faxes.coord(faxes_coord_otus)

faxes_nm <- colnames(faxes_coord_otus)
res <- as.data.frame(matrix(NA, length(tr_nm) * length(faxes_nm), 
                            7, dimnames = list(NULL, c("trait", "axis", "test", 
                                                       "stat", "value", "p.value",
                                                       "slope"))))
flag <- 0
for (i in tr_nm) {
  for (j in faxes_nm) {
    flag <- flag + 1
    trait <- NULL
    axis <- NULL
    data_ij <- data.frame(trait = ftraits_values_otus[, i], axis = faxes_coord_otus[, 
                                                                    j])
    
    test_ij <- "Linear Model"
    lm_ij <- summary(stats::lm(trait ~ axis, data = data_ij))
    res[flag, c("trait", "axis", "test", "stat")] <- c(i, 
                                                       j, test_ij, "r2")
    res[flag, c("value", "p.value")] <- c(round(lm_ij$r.squared, 
                                                3), round(lm_ij$coefficients[2, 4], 4))
    res[flag, c("slope")] <- lm_ij$coefficients[2,1]
    
    color_group <- ifelse(round(lm_ij$coefficients[2, 4], 4)<0.05&round(lm_ij$r.squared,3)>0.5,"darkblue",
                          ifelse(round(lm_ij$coefficients[2, 4], 4)<0.05&round(lm_ij$r.squared,3)<0.5,"lightblue","grey40"))
    x_lab_ij <- NULL
    if (j == faxes_nm[length(faxes_nm)]) {
      x_lab_ij <- i
    }
    y_lab_ij <- NULL
    if (i == tr_nm[1]) {
      y_lab_ij <- j
    }
    gg_ij <- ggplot(data_ij, aes(x=trait, 
                                 y=axis)) + 
      xlab(x_lab_ij) +
      ylab(y_lab_ij) + 
      theme_bw()+
      geom_point(size = 2,
                 color=color_group)+
      theme(axis.text.x = element_blank(),
            axis.text.y = element_blank())
    
    if (j == faxes_nm[length(faxes_nm)]) {
      gg_ij <- gg_ij+
        theme(axis.text.x = element_text())
    }
    if (i == tr_nm[1]) {
      gg_ij <- gg_ij+
        theme(axis.text.y = element_text())
    }
    
    if (j == faxes_nm[1]) {
      col_j <- gg_ij
    }
    else {
      col_j <- col_j/gg_ij
    }
  }
  if (i == tr_nm[1]) {
    otu_tr_faxes <- col_j
  }
  else {
    otu_tr_faxes <- otu_tr_faxes | col_j
  }
  
}

otu_tr_faxes

otu_tr_faxes2 <-
res%>%
  dplyr::mutate(slope_sign=ifelse(slope>0,"pos","neg"))%>%
  dplyr::mutate(value=ifelse(p.value>0.05,NA,value))%>%
  dplyr::mutate(axis=forcats::fct(axis,levels=paste0("PC",19:1)))%>%
  ggplot(aes(x=trait,y=axis,color=slope_sign,size=value))+
  geom_point()+
  theme_minimal()+
  labs(size="R²")+
  scale_color_manual(values = c("neg"="red","pos"="green"))+
  theme(axis.text.x = element_text(angle=45,vjust=1,hjust=1),
        axis.title = element_blank())


big_plot <- mFD::funct.space.plot(
  sp_faxes_coord  = faxes_coord_otus[ , c("PC1", "PC2", "PC3", "PC4")],
  faxes           = c("PC1", "PC2", "PC3", "PC4"),
  name_file       = NULL,
  faxes_nm        = NULL,
  range_faxes     = c(NA, NA),
  color_bg        = "grey95",
  color_pool      = "darkgreen",
  fill_pool       = "white",
  shape_pool      = 21,
  size_pool       = 1,
  plot_ch         = TRUE,
  color_ch        = "black",
  fill_ch         = "white",
  alpha_ch        = 0.5,
  plot_vertices   = TRUE,
  color_vert      = "blueviolet",
  fill_vert       = "blueviolet",
  shape_vert      = 23,
  size_vert       = 1,
  plot_sp_nm      = NULL,
  nm_size         = 3,
  nm_color        = "black",
  nm_fontface     = "plain",
  check_input     = TRUE)

# compute diversity indices

otu_fd_weights <- dtmp$otu_table #get coms
otu_fd_weights <- otu_fd_weights[rowSums(otu_fd_weights>0)>0.05*ncol(otu_fd_weights),]

alpha_fd_indices_otus <- mFD::alpha.fd.multidim(
  sp_faxes_coord   = faxes_coord_otus[ , c("PC1", "PC2", "PC3", "PC4")],
  asb_sp_w         = t(otu_fd_weights),
  ind_vect         = c("fdis", "fmpd", "fnnd", "feve", "fric", "fdiv", "fori", 
                       "fspe", "fide"),
  scaling          = TRUE,
  check_input      = TRUE,
  details_returned = TRUE)

# plot diversity indices

plots_alpha <- mFD::alpha.multidim.plot(
  output_alpha_fd_multidim = alpha_fd_indices_otus,
  plot_asb_nm              = rownames(t(otu_fd_weights))[c(1,61)],
  ind_nm                   = c("fdis", "fide", "fnnd", "feve", "fric", 
                               "fdiv", "fori", "fspe"),
  faxes                    = NULL,
  faxes_nm                 = NULL,
  range_faxes              = c(NA, NA),
  color_bg                 = "grey95",
  shape_sp                 = c(pool = 3, asb1 = 21, asb2 = 21),
  size_sp                  = c(pool = 0.7, asb1 = 1, asb2 = 1),
  color_sp                 = c(pool = "grey50", asb1 = "#1F968BFF", asb2 = "#DCE319FF"),
  color_vert               = c(pool = "grey50", asb1 = "#1F968BFF", asb2 = "#DCE319FF"),
  fill_sp                  = c(pool = NA, asb1 = "#1F968BFF", asb2 = "#DCE319FF"),
  fill_vert                = c(pool = NA, asb1 = "#1F968BFF", asb2 = "#DCE319FF"),
  color_ch                 = c(pool = NA, asb1 = "#1F968BFF", asb2 = "#DCE319FF"),
  fill_ch                  = c(pool = "white", asb1 = "#1F968BFF", asb2 = "#DCE319FF"),
  alpha_ch                 = c(pool = 1, asb1 = 0.3, asb2 = 0.3),
  shape_centroid_fdis      = c(asb1 = 22,  asb2 = 24),
  shape_centroid_fdiv      = c(asb1 = 22,  asb2 = 24),
  shape_centroid_fspe      = 23,
  color_centroid_fspe      = "black",
  size_sp_nm               = 3, 
  color_sp_nm              = "black",
  plot_sp_nm               = NULL,
  fontface_sp_nm           = "plain",
  save_file                = FALSE,
  check_input              = TRUE) 


alpha_fd_indices_otus$functional_diversity_indices%>%
  mutate(sample_id=rownames(.))%>%
  pivot_longer(cols = -sample_id,
               names_to = "metrics",
               values_to = "metrics_values")%>%
  left_join(dtmp$sample_table[,c(1,26)])%>%
  ggplot(aes(x=Trtmt,y=metrics_values,fill=Trtmt))+
  geom_boxplot()+
  theme_bw()+
  scale_fill_manual(values=mo_col)+
  facet_wrap(~metrics,scales = "free")
  
# betadiv

#jaccard
beta_fd_indices_otus <- mFD::beta.fd.multidim(
      sp_faxes_coord   = faxes_coord_otus[ , c("PC1", "PC2", "PC3", "PC4")],
      asb_sp_occ       = t(otu_tmp),
      check_input      = TRUE,
      beta_family      = c("Jaccard"),
      details_returned = TRUE)


beta_plot_otus <- mFD::beta.multidim.plot(
  output_beta_fd_multidim = beta_fd_indices_otus,
  plot_asb_nm             = rownames(t(otu_fd_weights))[c(1,61)],
  beta_family             = c("Jaccard"),
  plot_sp_nm              = c("MOTU_194712","MOTU_05683"),
  faxes                   = paste0("PC", 1:4),
  name_file               = NULL,
  faxes_nm                = NULL,
  range_faxes             = c(NA, NA),
  color_bg                = "grey95",
  shape_sp                = c("pool" = 3.0, asb1 = 22, asb2 = 21),
  size_sp                 = c("pool" = 0.8, asb1 =  1, asb2 =  1),
  color_sp                = c("pool" = "grey50", asb1 = "blue", asb2 = "red"),
  fill_sp                 = c("pool" = NA, asb1 = "white", asb2 = "white"),
  fill_vert               = c("pool" = NA, asb1 = "blue", asb2 = "red"),
  color_ch                = c("pool" = NA, asb1 = "blue", asb2 = "red"),
  fill_ch                 = c("pool" = "white", asb1 = "blue", asb2 = "red"),
  alpha_ch                = c("pool" = 1, asb1 = 0.3, asb2 = 0.3),
  nm_size                 = 3,
  nm_color                = "black",
  nm_fontface             = "plain",
  check_input             = TRUE)


df_betaFD_matrix <-
beta_fd_indices_otus$pairasb_fbd_indices$jac_diss%>%
  as.matrix()%>%
  reshape2::melt()%>%
  dplyr::rename(sample_id=Var1,
         sample_id2=Var2,
         jacc_diss=value)%>%
  left_join(dtmp$sample_table[,c("sample_id","Trtmt")])%>%
  left_join(dplyr::rename(dtmp$sample_table[,c("sample_id","Trtmt")],
                          sample_id2=sample_id,
                          Trtmt_2=Trtmt))%>%
  mutate(jacc_diss=ifelse(jacc_diss==0,NA,jacc_diss),
         Trtmt=fct_relevel(Trtmt,levels(trtmt)),
         Trtmt_2=fct_relevel(Trtmt_2,levels(trtmt)))%>%
  arrange(Trtmt,Trtmt_2)%>%
  mutate(sample_id=fct_inorder(sample_id),
         sample_id2=fct_inorder(sample_id2))


playout <- "
AB
CD
"

df_betaFD_matrix%>%
  ggplot(aes(x=sample_id,y=sample_id2,fill=jacc_diss))+
  geom_tile()+
  scale_fill_gradientn(colors=  paletteer::paletteer_d("ggsci::orange_material"))+
    theme_void()+
  df_betaFD_matrix%>%
  ggplot(aes(x=1,y=sample_id2,fill=Trtmt_2))+
  geom_tile()+
  scale_fill_manual(values = mo_col)+
  theme_void()+
  df_betaFD_matrix%>%
  ggplot(aes(x=sample_id,y=1,fill=Trtmt))+
  geom_tile()+
  scale_fill_manual(values = mo_col)+
  theme_void()+
  plot_spacer()+
  plot_layout(guides = "collect",design=playout,widths = c(5,1,5,1),heights = c(5,1,5,1))

#Sorensen
beta_fd_indices_otus_sorensen <- mFD::beta.fd.multidim(
      sp_faxes_coord   = faxes_coord_otus[ , c("PC1", "PC2", "PC3", "PC4")],
      asb_sp_occ       = t(otu_tmp),
      check_input      = TRUE,
      beta_family      = c("Sorensen"),
      details_returned = TRUE)

df_betaFD_matrix_sorensen <-
beta_fd_indices_otus_sorensen$pairasb_fbd_indices$sor_diss%>%
  as.matrix()%>%
  reshape2::melt()%>%
  dplyr::rename(sample_id=Var1,
         sample_id2=Var2,
         sor_diss=value)%>%
  left_join(dtmp$sample_table[,c("sample_id","Trtmt")])%>%
  left_join(dplyr::rename(dtmp$sample_table[,c("sample_id","Trtmt")],
                          sample_id2=sample_id,
                          Trtmt_2=Trtmt))%>%
  mutate(sor_diss=ifelse(sor_diss==0,NA,sor_diss),
         Trtmt=fct_relevel(Trtmt,levels(trtmt)),
         Trtmt_2=fct_relevel(Trtmt_2,levels(trtmt)))%>%
  arrange(Trtmt,Trtmt_2)%>%
  mutate(sample_id=fct_inorder(sample_id),
         sample_id2=fct_inorder(sample_id2))


df_betaFD_matrix_sorensen%>%
  ggplot(aes(x=sample_id,y=sample_id2,fill=sor_diss))+
  geom_tile()+
  scale_fill_gradientn(colors=  paletteer::paletteer_d("ggsci::orange_material"))+
    theme_void()+
  df_betaFD_matrix_sorensen%>%
  ggplot(aes(x=1,y=sample_id2,fill=Trtmt_2))+
  geom_tile()+
  scale_fill_manual(values = mo_col)+
  theme_void()+
  df_betaFD_matrix_sorensen%>%
  ggplot(aes(x=sample_id,y=1,fill=Trtmt))+
  geom_tile()+
  scale_fill_manual(values = mo_col)+
  theme_void()+
  plot_spacer()+
  plot_layout(guides = "collect",design=playout,widths = c(5,1,5,1),heights = c(5,1,5,1))


# Functional entities 
otu_traits_cat <- data.frame(trait_name=colnames(otu_space[,-c(1:3)]),
                             trait_type="Q")

fe_otus <- mFD::sp.to.fe(
  sp_tr       = otu_space[,-c(1:3)], 
  tr_cat      = otu_traits_cat, 
  fe_nm_type  = "fe_rank", 
  check_input = TRUE) 
```


## Dromics

```{r}
library(DRomics)

dromicsouts <- list()
for(i in datasets){
  tmp <- get(i)
  otu_dromics <- tmp$otu_table
  
  for(t in c("Nb_Fr") ) #trait_list_glob
  
  trait_dromics <- tmp$sample_table[t]
  
  if(all(names(otu_dromics)==rownames(trait_dromics))){
    print("GOOD")
  }else{
    warning("WRONG ORDER BETWEEN LINES AND COLUMNS")
  }
  
data4dromics <- formatdata4DRomics(signalmatrix = log2(otu_dromics+1),
                                   dose = dplyr::pull(trait_dromics,t),
                                   samplenames = rownames(trait_dromics))

dromics_normalized <- microarraydata(data4dromics, norm.method = "cyclicloess")


data4dromics <- formatdata4DRomics(signalmatrix = otu_dromics,
                                   dose = dplyr::pull(trait_dromics,t),
                                   samplenames = rownames(trait_dromics))
dromics_normalized <- RNAseqdata(data4dromics, transfo.method = "vst")


dromics_slctd <- itemselect(dromics_normalized, select.method = "quadratic", FDR = 0.05)

dromics_fit <- drcfit(dromics_slctd)

plotsdromics <-  plot(dromics_fit)

dromicsouts[[i]][["data4dromics"]] <- data4dromics
dromicsouts[[i]][["dromics_normalized"]] <- dromics_normalized
dromicsouts[[i]][["dromics_slctd"]] <- dromics_slctd
dromicsouts[[i]][["dromics_fit"]] <- dromics_fit
dromicsouts[[i]][["plotsdromics"]] <- plotsdromics

}
```


## TITAN

### Hmaps

```{r}
NewVarNames <-  trait_df_name$cleannames
comp_plot <- NULL
my_motus <- NULL
titan_list <- NULL
df_titan_plot <- NULL
titan_info_papers <- NULL

for(i in datasets){
  tmp <- get(i)
  tmp <- clone(tmp)
  
  titan <- file_titan[[i]] 
  
  titan <- titan[apply(titan, 1, function(x) any(!is.na(x))),] # remove otus never increasing or decreasing
  titan[is.na(titan)] <- 0 # set nas to 0s
  colnames(titan) <- gsub("./articlez/gen1/resultats/final/titan", "", colnames(titan))
  colnames(titan) <- gsub(paste0("^\\/res_|_",i,"_\\.RDS"), "\\1", colnames(titan)) # renames columns to get variable name only
  order <- sapply(trait_df_name$rawnames, grepl,colnames(titan))%>%reshape2::melt()%>%filter(value=="TRUE")
  order%<>%left_join(dplyr::rename(trait_df_name,Var2=rawnames))
  colnames(titan)[order$Var1] <- order$cleannames
  
  
  if(any(grepl("C/N",colnames(titan)))){
    titan%<>%
      as.data.frame()%>%
      dplyr::select(!(`C/N`)) }# Remove C/N as we have C and N separately
  
  titan <- titan[,!colSums(titan!=0)==0] #remove var with no OTU associated
  titan <- titan[!rowSums(titan!=0)==0,] #remove otu not associated to any var
  
  tax_tab <- as.data.frame(tmp$tax_table)  # get tax table
  tax_tab <- tax_tab[match(rownames(titan), rownames(tax_tab)),] # keep only otus present in titan results
  
  for(j in 1:ncol(tax_tab)){
    tax_tab[nchar(tax_tab[[j]])<4,j] <- "Unknown" # replace non assignation tax levels (e.g. g__ for genus) by "Unknown"
  }
  tax_tab$OTUs <- rownames(tax_tab) # add an OTUs columns
  
  #get prevalence and relab of these OTUs
  prev_titan_otus <- tmp$otu_table%>%
    filter(rownames(.)%in%tax_tab$OTUs)
  prev_titan_otus <- rowSums(prev_titan_otus>0)%>%
    reshape2::melt()%>%
    mutate(Var2=rownames(.))%>%
    dplyr::rename(prev=value)
  
  relab_titan_otus <- tmp$otu_table%>%
    mutate(across(1:ncol(.),~.x/sum(.)))%>%
    mutate(Var2=rownames(.))
  
  for(rm_unknwn_class in c("rmthem","dontrm")){
    
    p <- grep("Unknown", tax_tab$Class_conf) # remove unknown for class from tax table AND titan results !!!!!
    if(rm_unknwn_class=="rmthem"){
      if(is_empty(p)){
        titan2 <- titan
        taxtab2 <- tax_tab
      }else{
        titan2 <- titan[-p,]
        taxtab2 <- tax_tab[-p,]
      }
    }else{
      titan2 <- titan
      taxtab2 <- tax_tab
    }
    
    t <- rbind(apply(titan2,2, function(x) sum(x<0)),
               apply(titan2,2, function(x) sum(x>0)),
               apply(titan2,2, function(x) sum(x!=0)))
    
    rownames(t) <- c("Number of increasers", "Number of decreasers", "Total")
    
    decr <- apply(titan2, 1, function(x) any(x<0))
    incr <- apply(titan2, 1, function(x) any(x>0))
    if (any(decr & incr)) cat("Beware, some OTUs are incr and decr!!") 
    
    tax_titan <- tmp$tax_table[rownames(tmp$tax_table)%in%rownames(titan2),] # extract taxonomic info about our OTUs
    # tax_titan$OTUs <- rownames(tax_titan)
    
    tax_titan %<>% 
      mutate(class=gsub("c__","",Class_conf))%>% #save class info
      unite(c(1:6),sep=";",col=Taxonomy)%>% # unite taxo info
      mutate(low_tax = gsub("(;[a-z]__)+?$","",Taxonomy))%>%
      mutate(low_tax = str_extract(pattern = "([a-z]__[^;]+?$)",string=low_tax))%>% # get lowest taxnomical info
      mutate(lvl_lowtax = str_extract(pattern = "[a-z]__",string=low_tax))%>% # get associated tax level
      mutate(low_tax = gsub("[a-z]__","",low_tax))%>%
      mutate(lvl_lowtax = ifelse(lvl_lowtax == "g__", "sp.", 
                                 ifelse(lvl_lowtax == "f__", "(Family)",
                                        ifelse(lvl_lowtax == 'o__', "(Order)",
                                               ifelse(lvl_lowtax == 'c__', "(Class)",
                                                      ifelse(lvl_lowtax == 'P__', "(Phylum)","(Kingdom)"))))))%>%
      mutate(Var2=rownames(.)) #renames levels and get rownames
    
    
    tax_titan%<>%
      left_join(prev_titan_otus)%>%
      left_join(relab_titan_otus)
    
    corder <- apply(titan2, 2, function(x) sum(x!=0)) %>%
      order # order columns by number of responsive otus
    
    
    tit_dat <- t(titan2[,corder])
    # tit_dat_clust <- hclust(vegdist(t(tit_dat),method = 'euclidean'))
    # tit_dat <- tit_dat[,tit_dat_clust$order]
    data_heatmap_titan <- reshape2::melt(tit_dat) # create ggplot data
    test <- merge(data_heatmap_titan,tax_titan,by="Var2")
    
    otu_tmp <- tmp$otu_table # get full community
    otu_tmp %<>% 
      filter(rownames(.)%in%test$Var2)%>%
      mutate(prevalence=rowSums(.>0),
             Var2=rownames(.))%>%
      dplyr::select(c(prevalence,Var2))
    
    test%<>%
      merge(otu_tmp,by="Var2")%>%
      mutate(Var2 = paste0(gsub("^M","",Var2)," ", low_tax, " ", lvl_lowtax))%>%
      mutate(Var1=ifelse(grepl("PCA1",Var1),"PCA Comp1",
                         ifelse(grepl("PCA2",Var1),"PCA Comp2",as.character(Var1))))%>%
      arrange(class,prevalence)%>%
      mutate(Var1=fct_inorder(Var1))
    lvl_var <- levels(test$Var1)
    test%<>%
      mutate(Var2=as.factor(Var2))%>%
      mutate(Var2=fct_relevel(Var2,c(unique(as.character(test$Var2)))))
    
    
    count_class <- table(test$class)/ncol(titan2) # divided by the number of variables as frame is melted
    
    test%<>%
      dplyr::rename(cleannames=Var1)%>%
      left_join(trait_df_name)%>%
      mutate(across(starts_with("var_cat"),~replace_na(.,"PCA components")))%>%
      mutate(cleannames=as.factor(cleannames))%>%
      mutate(cleannames=fct_relevel(cleannames,lvl_var))%>%
      dplyr::mutate(class=ifelse(class=="","Unknown",class))%>%
      mutate(class=as.factor(class))%>%
      mutate(class=fct_relevel(class,c(levels(class)[levels(class)!="Unknown"],"Unknown")))
    
    for(t in gsub("_","",transfo_list)){
      for(diablo_var in c("onlydiab","notonly")){
        
        if(diablo_var=="onlydiab"){
          df <- test%>%
            filter(cleannames%in%c(filter(diablo_figz[[i]][[t]]$circ$data,var_cat!="otus")[,"cleannames"],"PCA Comp1","PCA Comp2"))
        }else{
          df <- test
          
        }
        
        df$Var2 <- droplevels(df$Var2)
        
        for(var_order in c("nb_resp","var_cat")){
          
          if(var_order=="var_cat"){
            df2 <- df%>%
              mutate(var_cat=fct_relevel(var_cat,"PCA components"))%>%
              arrange(var_cat,cleannames)%>% #add cleannames pour alphabetical order?
              mutate(cleannames=fct_inorder(cleannames))
          }else{
            df2 <- df
          }
          
          
          
          heatmap_titan <- ggplot(df2, aes(cleannames, Var2, fill= value)) +
            geom_raster()+
            xlab("")+
            ylab("")+
            scale_fill_gradient2(low = "darkblue",
                                 mid = "lightyellow3",
                                 high = "#a30800",
                                 limits=c(floor(min(range(df2$value)-1)),
                                          ceiling(max(range(df2$value)+1))),
                                 breaks=c(floor(min(range(df2$value)-1)),
                                          0,
                                          ceiling(max(range(df$value)+1))),
                                 name = "z-score")+
            theme(axis.text.x = element_text(angle=45,
                                             margin=margin(t=30),
                                             vjust = -1,hjust=0),
                  panel.grid.major = element_blank(),
                  panel.grid.minor = element_blank(),
                  panel.background = element_blank(),
                  plot.margin = unit(c(0, 0, 0, 0), "lines"),
                  axis.text = element_text(face="bold"))+
            scale_x_discrete(position = "top")+
            geom_hline(yintercept = cumsum(count_class)[-length(count_class)]+0.5,
                       color='darkgrey')
          
          heat_var <- levels(df2$cleannames)%>%
            tibble()%>%
            set_colnames("cleannames")%>%
            left_join(trait_df_name)%>%
            mutate(across(starts_with("var_cat"),~replace_na(.,"PCA components")))%>%
            mutate(var_cat=fct_relevel(var_cat,"PCA components"))%>%
            mutate(cleannames=as.factor(cleannames))%>%
            mutate(cleannames=fct_relevel(cleannames,levels(df2$cleannames)))%>%
            ggplot(aes(x=cleannames,y=1,fill=var_cat))+
            geom_raster()+
            scale_fill_manual(values=c("grey40",var_cat_col[c(1,2,3,6,4,5)]),
                              name='')+
            theme_void()+
            theme(legend.position = "bottom",
                  plot.margin = unit(c(0, 0, 0, 0), "lines"))
          
          heatclass <- ggplot(df2, aes(1,Var2,fill=class))+
            geom_tile(width=1,height=1)+
            scale_fill_manual(values=colorRampPalette(brewer.pal(8, "Dark2"))(length(unique(df2$class))))+
            theme_void() +
            scale_x_discrete(expand=c(0,0)) +
            scale_y_discrete(expand=c(0,0)) +
            labs(x = NULL, y = NULL)+
            labs(fill = "Class")+
            geom_hline(yintercept = cumsum(count_class)[-length(count_class)]+0.5,color='darkgrey')+
            theme(plot.margin = unit(c(0, 0, 0, 0), "lines"))
          
          
          # where2cut <- round(seq(1,max(df$prev),length.out=10)/10)*10
          where2cut <- c(min(df2$prev),10,20,40,60,max(df2$prev))
          df2$prev_int <- cut(df2$prev,unique(c(1,where2cut)))
          labels2use <- gsub("\\(|,.*","",levels(df2$prev_int))
          
          heatprev <- df2%>%
            ggplot(aes(x=1,y=Var2,size=prev))+
            geom_point(color="grey40")+
            theme_void()+
            # scale_color_gradient(low="lightgrey",high = "black",name="Prevalence")+
            scale_size_area(name="Prevalence",breaks=where2cut,max_size = 4)
          
          
          heatrelab <- df2%>%
            dplyr::select(matches("Var2|FEpl"))%>%
            reshape2::melt()%>%
            filter(value!=0)%>%
            group_by(Var2)%>%
            mutate(mean_relab=mean(value))%>%
            ggplot(aes(y=Var2))+
            geom_point(aes(x=value*100),color="grey40")+
            geom_point(aes(x=mean_relab*100),fill="darkorange",size=2,shape=23)+
            scale_x_log10(breaks=c(0,0.1,1.0,10.0,40.0),position="top")+
            theme_void()+
            theme(axis.line.x = element_line(),
                  axis.title.x = element_text(),
                  axis.text.x = element_text(angle=45,vjust=1,hjust=1),
                  axis.ticks.x = element_line(),
                  axis.ticks.length.x = unit(0.1,"line"))+
            xlab("Relative abundance (%)")
          
          
          comp_plot[[i]][[t]][[rm_unknwn_class]][[var_order]][[diablo_var]] <-((heatmap_titan+plot_layout(guides = "auto"))/
                                                                            (heat_var+plot_layout(guides = "keep"))+
                                                                            plot_layout(heights = c(9,1))|
                                                                            (heatclass/
                                                                               plot_spacer()+
                                                                               theme_void()+
            theme(plot.margin = unit(c(0, 0, 0, 0), "lines")))+
                                                                            plot_layout(heights = c(9,1))|
                                                                            (heatrelab/
                                                                               plot_spacer()+theme_void()+
            theme(plot.margin = unit(c(0, 0, 0, 0), "lines")))+
                                                                            plot_layout(heights = c(9,1))|
                                                                            (heatprev/
                                                                               plot_spacer()+theme_void()+
            theme(plot.margin = unit(c(0, 0, 0, 0), "lines")))+
                                                                            plot_layout(heights = c(9,1))) +
            plot_layout(guides = "collect",widths = c(15,1,4,1))
          
          
          my_motus[[i]][[t]][[rm_unknwn_class]][[var_order]][[diablo_var]] <- taxtab2
          titan_list[[i]][[t]][[rm_unknwn_class]][[var_order]][[diablo_var]] <- titan2
          df_titan_plot[[i]][[t]][[rm_unknwn_class]][[var_order]][[diablo_var]] <- df2
          titan_info_papers[[i]][[t]][[rm_unknwn_class]][[var_order]][[diablo_var]][["notus"]] <- nrow(titan2)
          titan_info_papers[[i]][[t]][[rm_unknwn_class]][[var_order]][[diablo_var]][["ntraits"]] <- ncol(titan2)
          titan_info_papers[[i]][[t]][[rm_unknwn_class]][[var_order]][[diablo_var]][["incdecr"]] <- as.data.frame(t)
          titan_info_papers[[i]][[t]][[rm_unknwn_class]][[var_order]][[diablo_var]][["whosincrdecr"]] <- data.frame(otus=gsub("M","",names(decr)), resp = ifelse(incr&decr,"both",ifelse(incr&!decr,"incr","decr")))
          titan_info_papers[[i]][[t]][[rm_unknwn_class]][[var_order]][[diablo_var]][["otupertrait"]] <- colSums(titan2!=0)
          titan_info_papers[[i]][[t]][[rm_unknwn_class]][[var_order]][[diablo_var]][["taxnumber"]] <- df2%>%
            group_by(class)%>%
            dplyr::count()%>%
            mutate(n=n/ncol(titan2))
          titan_info_papers[[i]][[t]][[rm_unknwn_class]][[var_order]][[diablo_var]][["otus_per_cat"]] <- titan2%>%
            as.data.frame()%>%
            mutate(otus=rownames(.))%>%
            reshape2::melt()%>%
            dplyr::rename(cleannames=variable)%>%
            left_join(trait_df_name)%>%
            filter(value!=0)%>%
            group_by(var_cat)%>%
            summarize(count=n_distinct(otus))  
        }
      }
    }
  }
}
```

```{r}
for(i in 1:2){
  for(t in 1:2){
    for(j in 1:2){
      for(k in 1:2){
        for(l in 1:2){
          ggsave(paste0("./articlez/gen1/resultats/final/titan/titan_xp_",
                        names(comp_plot)[i],"_",
                        names(comp_plot[[i]])[t],"_",
                        names(comp_plot[[i]][[t]])[j],"_",
                        names(comp_plot[[i]][[t]][[j]])[k],"_",
                        names(comp_plot[[i]][[t]][[j]][[k]])[l],"_",
                        ".png"),comp_plot[[i]][[t]][[j]][[k]][[l]],width = 14, height = 10)
        }
      }
    }
  }
}
```

```{r}
comp_plot_plusdiabl <- NULL

for(i in datasets){
  for(t in gsub("_","",transfo_list)){
  dtest <- comp_plot[[i]][[t]]$dontrm$var_cat$notonly[[1]][[1]][[1]]$data%>%
    mutate(Var2=as.character(Var2))
  
 
  count_class <- table(dtest$class)/ncol(titan_list[[i]][[t]])
  dtest2 <- unique(dplyr::mutate(dplyr::select(filter(diablofigz_2[[i]][[t]][[1]]$data,Block=="OTUs_com"),c("cleannames","otu_group")),Var2=gsub('\u25cf ','',cleannames)))
  dtest%<>%left_join(dplyr::select(dtest2,-cleannames))
  
  dtest%<>%
    mutate(Var2=ifelse(grepl(paste0(info_paper_diablo[[i]][[t]]$interpratedotus,collapse="|"),Var2) ,paste0("\u25cf ",Var2),Var2))
  
  dtest%<>%
    arrange(class,prevalence)%>%
    mutate(Var2=as.factor(Var2))%>%
    mutate(Var2=fct_relevel(Var2,c(unique(as.character(dtest$Var2)))))
  
  for(var_order in c("nb_resp","var_cat")){
    for(keeponlydiab in c("yes","no")){
      for(orderdiab in c("yes","no")){
      if(var_order=="var_cat"){
        df <- dtest%>%
          mutate(cleannames=as.character(cleannames))%>%
          arrange(var_cat,cleannames)%>%
          mutate(cleannames=fct_inorder(cleannames))
      }else{
        corder <- df%>%
          group_by(cleannames)%>%
          mutate(value=ifelse(value!=0,1,0))%>%
          summarise(nb_otus=sum(value))%>%
          arrange(nb_otus)%>%
          mutate(cleannames=fct_inorder(cleannames))%>%
          dplyr::select(cleannames)
        
        df <- dtest%>%
          mutate(cleannames=fct_relevel(cleannames,levels(corder$cleannames)))
      }
      if(keeponlydiab=="yes"){
        df%<>%filter(!is.na(otu_group))
      }else{
        df <- df
      }
      if(orderdiab=="yes"){
        df%<>%
          arrange(otu_group,class,prevalence)%>%
          mutate(Var2=fct_inorder(Var2))%>%
          mutate(Var2=fct_relevel(Var2,rev(levels(Var2))))
      }else{
        df <- df
      }
      heatmap_titan <- ggplot(df, aes(cleannames, Var2, fill= value)) +
        geom_tile()+
        scale_fill_gradient2(low = "darkblue",mid = "lightyellow3", high = "#a30800",
                             limits=c(floor(min(range(df$value)-1)),ceiling(max(range(df$value)+1))),
                             breaks=c(floor(min(range(df$value)-1)),0,ceiling(max(range(df$value)+1)))
                             ,name = "z-score")+
        theme(axis.text.x = element_text(angle=45,margin=margin(t=30),vjust = -1,hjust=0),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              panel.background = element_blank(),
              plot.margin = unit(c(0, 0, 0, 0), "lines"),
              axis.text = element_text(face="bold"),
              axis.title = element_blank())+
        scale_x_discrete(position = "top")+
        geom_hline(yintercept = cumsum(count_class)[-length(count_class)]+0.5,color='darkgrey')
      
      heat_var <- levels(df$cleannames)%>%
        tibble()%>%
        set_colnames("cleannames")%>%
        left_join(trait_df_name)%>%
        mutate(cleannames=as.factor(cleannames))%>%
        mutate(cleannames=fct_relevel(cleannames,levels(df$cleannames)))%>%
        mutate(var_cat=replace_na(var_cat,'PCA components'))%>%
        mutate(var_cat=fct_relevel(var_cat,"PCA components"))%>%
        ggplot(aes(x=cleannames,y=1,fill=var_cat))+
        geom_raster()+
        scale_fill_manual(values=c("grey40",var_cat_col[c(1,2,3,6,4,5)]),name='Variable categories')+
        theme_void()+
        theme(legend.position = "bottom",legend.title.position = 'top',legend.title = element_text(hjust = 0.5))
      
      
      heat_var_diab <- levels(df$cleannames)%>%
        tibble()%>%
        set_colnames("cleannames")%>%
        left_join(trait_df_name)%>%
        mutate(is_diablo_var=ifelse(cleannames%in%info_paper_diablo[[i]]$traits$cleannames,"Yes","No"))%>%
        mutate(cleannames=as.factor(cleannames))%>%
        mutate(cleannames=fct_relevel(cleannames,levels(df$cleannames)))%>%
        ggplot(aes(x=cleannames,y=1,fill=is_diablo_var))+
        geom_raster()+
        scale_fill_manual(values=c(Yes="#B2DF8A",No="grey80"),name='Diablo selected trait')+
        theme_void()+
        theme(legend.position = "bottom",legend.direction = 'vertical',legend.title = element_text(hjust = 0.5))
      
      if(grepl("16s",i)){
        col_class <- brewer.pal(8, "Dark2")}else{
          
          col_class <- c(paletteer::paletteer_d("ggthemes::Classic_10")[c(1:7,9:10)],"grey40")
        }
      
      heatclass <- ggplot(df, aes(1,Var2,fill=class))+
        geom_tile()+
        scale_fill_manual(values=colorRampPalette(col_class)(length(unique(df$class))))+
        theme_void() +
        scale_x_discrete(expand=c(0,0)) +
        scale_y_discrete(expand=c(0,0)) +
        labs(x = NULL, y = NULL)+
        labs(fill = "Class")+
        geom_hline(yintercept = cumsum(count_class)[-length(count_class)]+0.5,color='darkgrey')
      
      cluster_palette <- switch(i,
                                "meco_mo16s_seqsg_raref"=rev(paletteer::paletteer_d("ochRe::winmar")[-6]),
                                "meco_moits_seqsg_raref"=rev(c("#F56455FF", "#2BAA92FF", "#15134BFF", "#9EC8FFFF", "#E9A800FF", "#00af66ff", "#ffebcc"))
                                )
      names(cluster_palette) <- levels(as.factor(dtest$otu_group))
      colorz <- c(cluster_palette,`not diablo OTUs`="grey80")
      df$otu_group <- as.factor(df$otu_group)
      df$otu_group <- droplevels(df$otu_group)
      df%<>%
        mutate(otu_group=as.character(otu_group))%>%
        dplyr::mutate(otu_group=replace_na(otu_group,"not diablo OTUs"))
      
      heatdiablogroup <- ggplot(df, aes(1,Var2,fill=otu_group))+
        geom_tile()+
        scale_fill_manual(values=colorz)+
        theme_void() +
        scale_x_discrete(expand=c(0,0)) +
        scale_y_discrete(expand=c(0,0)) +
        labs(x = NULL, y = NULL)+
        labs(fill = "Diablo group")+
        geom_hline(yintercept = cumsum(count_class)[-length(count_class)]+0.5,color='darkgrey')
      
      
      # where2cut <- round(seq(1,max(df$prev),length.out=10)/10)*10
      where2cut <- c(min(df$prev),10,20,40,60,max(df$prev))
      df$prev_int <- cut(df$prev,unique(c(1,where2cut)))
      labels2use <- gsub("\\(|,.*","",levels(df$prev_int))
      
      heatprev <- df%>%
        dplyr::select(Var2,prev)%>%
        distinct()%>%
        ggplot(aes(x=1,y=Var2,size=prev))+
        geom_point()+
        theme_void()+
        # scale_color_gradient(low="lightgrey",high = "black",name="Prevalence")+
        scale_size_area(name="Prevalence",breaks=where2cut,max_size = 4)
      
      
      heatrelab <- df%>%
        dplyr::select(matches("Var2|FEpl"))%>%
        reshape2::melt()%>%
        filter(value!=0)%>%
        group_by(Var2)%>%
        mutate(mean_relab=mean(value))%>%
        ggplot(aes(y=Var2))+
        geom_point(aes(x=value*100))+
        geom_point(aes(x=mean_relab*100),fill="darkorange",size=2,shape=23)+
        scale_x_log10(breaks=c(0,0.1,1.0,10.0,40.0),position="top")+
        theme_void()+
        theme(axis.line.x = element_line(),
              axis.title.x = element_text(),
              axis.text.x = element_text(angle=45,vjust=1,hjust=0),
              axis.ticks.x = element_line(),
              axis.ticks.length.x = unit(0.1,"line"))+
        xlab("Relative abundance (%)")
      
      
      
      
      comp_plot_plusdiabl[[i]][[t]][[keeponlydiab]][[var_order]][[orderdiab]] <-   
        (((heatmap_titan|
             heatclass+theme(plot.margin = unit(c(.1,.1,.1,.1),"lines"))|
             heatdiablogroup+theme(plot.margin = unit(c(.1,.1,.1,.1),"lines"))|
             heatrelab+theme(plot.margin =unit(c(.1,.1,.1,.1),"lines"))|
             heatprev+theme(plot.margin = unit(c(.1,.1,.1,.1),"lines")))+
            plot_layout(guides = "collect",widths = c(9,.25,.25,2,.5)))/
           ((((heat_var|
                 plot_spacer()+theme(plot.margin = unit(c(.1,.1,.1,.1),"lines"))|
                 plot_spacer()+theme(plot.margin = unit(c(.1,.1,.1,.1),"lines"))|
                 plot_spacer()+theme(plot.margin = unit(c(.1,.1,.1,.1),"lines"))|
                 plot_spacer()+theme(plot.margin = unit(c(.1,.1,.1,.1),"lines")))+
                plot_layout(widths = c(9,.25,.25,2,.5)))/
               ((heat_var_diab| plot_spacer()+theme(plot.margin = unit(c(.1,.1,.1,.1),"lines"))|
                   plot_spacer()+theme(plot.margin = unit(c(.1,.1,.1,.1),"lines"))|
                   plot_spacer()+theme(plot.margin = unit(c(.1,.1,.1,.1),"lines"))|
                   plot_spacer()+theme(plot.margin = unit(c(.1,.1,.1,.1),"lines")))+
                  plot_layout(widths = c(9,.25,.25,2,.5))))+
              plot_layout(guides = "collect")&theme(legend.position = "bottom")))+
        plot_layout(heights = c(9,.3))
      
      
      
      
      ggsave(paste0("./articlez/gen1/resultats/final/titan/titanwithdiab/titanwithdiab_",i,"_",t,"_",var_order,"_",keeponlydiab,"_",orderdiab,"_hmap.png"),comp_plot_plusdiabl[[i]][[t]][[keeponlydiab]][[var_order]][[orderdiab]] ,width = 10,height = ifelse(keeponlydiab=="yes",8,12))
      
      }
    }
  }
  }
}
```



### Relab Vs leaf traits

```{r}
for(i in datasets){
  for(t in gsub("_","",transfo_list)){
    data_tmp <- get(i)
    data_tmp <- clone(data_tmp) # get microtable
    
    pca_val <-  PCA_plant_trait$ind$coord[which(data_leaf$number%in%data_tmp$sample_table$number),1:2]%>%
      as.data.frame()%>%
      dplyr::rename(PCA1=Dim.1,
                    PCA2=Dim.2)%>%
      mutate(sample_id=data_tmp$sample_table$sample_id)
    
    df_plot <- data_tmp$otu_table%>%
      mutate(across(everything(),~.x/sum(.x)))%>% # transform to relative abundance
      t()%>% #twist otu table
      as.data.frame%>%
      mutate(sample_id=rownames(.))%>% # get sample id as columns
      left_join(data_tmp$sample_table)%>% # merge with sample table
      reshape2::melt(id.vars=(ncol(.)-(ncol(data_tmp$sample_table)-1)):ncol(.))%>% # melt otu counts
      filter(variable%in%my_motus[[i]][[t]]$dontrm$var_cat$notonly$OTUs)%>% # remove non TITAN otus
      mutate(variable=gsub("M","",variable))%>% # remove M before OTU
      left_join(pca_val)
    
    
    df_plot2 <- data_tmp$otu_table%>%
      t()%>% #twist otu table
      as.data.frame()%>%
      mutate(sample_id=rownames(.))%>% # get sample id as columns
      left_join(data_tmp$sample_table)%>% # merge with sample table
      reshape2::melt(id.vars=(ncol(.)-(ncol(data_tmp$sample_table)-1)):ncol(.))%>% # melt otu counts
      filter(variable%in%my_motus[[i]][[t]]$dontrm$var_cat$notonly$OTUs)%>% # remove non TITAN otus
      mutate(variable=gsub("M","",variable))%>% # remove M before OTU
      left_join(pca_val)
    
    df_tmp <- df_titan_plot[[i]][[t]]$dontrm$var_cat$notonly%>%
      mutate(variable = str_extract(Var2,"^\\S*"))%>% # get taxonomic info
      dplyr::select(!value)
    
    df_plot %<>% left_join(df_tmp) # merge it
    df_plot2 %<>% left_join(df_tmp)
    
    var_n <- colnames(df_plot)[which((colnames(df_plot)%in%trait_df_name$rawnames))] # get variable names
    trait_df_name<- trait_df_name[match(var_n,trait_df_name$rawnames),] #reorder table
    colnames(df_plot)[which((colnames(df_plot)%in%trait_df_name$rawnames))] <- trait_df_name$cleannames # get cleannames so that titan scores and leaf traits have same name
    colnames(df_plot2)[which((colnames(df_plot2)%in%trait_df_name$rawnames))] <- trait_df_name$cleannames # get cleannames so that titan scores and leaf traits have same name
    
    colnames(df_plot)[grep("PCA1",colnames(df_plot))] <- "PCA dim1"
    colnames(df_plot)[grep("PCA2",colnames(df_plot))] <- "PCA dim2"
    
    colnames(df_plot2)[grep("PCA1",colnames(df_plot2))] <- "PCA dim1"
    colnames(df_plot2)[grep("PCA2",colnames(df_plot2))] <- "PCA dim2"
    
    
    titan_tmp <- titan_list[[i]][[t]]$dontrm$var_cat$notonly # get titan scores
    titan_tmp %<>%
      mutate(across(everything(),~ifelse(.x>0,"Pos",ifelse(.x==0,"No response","Neg")),.names = "{.col}_trs"))%>% # transform scores to "pos", "neg" and "no" response 
      dplyr::select(ends_with(match="trs"))%>% # remove score columns
      mutate(variable=gsub("M","",rownames(.)))
    colnames(titan_tmp)[grep("PCA1",colnames(titan_tmp))] <- "PCA dim1_trs"
    colnames(titan_tmp)[grep("PCA2",colnames(titan_tmp))] <- "PCA dim2_trs"
    
    
    df_plot %<>% left_join(titan_tmp) # merge with df
    df_plot2 %<>% left_join(titan_tmp) # merge with df
    
    for(j in match(gsub("_trs$","",colnames(titan_tmp)),colnames(df_plot))[-ncol(titan_tmp)]){ # for each titan variable
      
      lab <- df_plot%>%group_by(variable)%>%reframe(prev=prevalence)%>% distinct(variable,prev) # get prevalence
      anno <- data.frame(lab = lab$prev,
                         variable =  as.factor(lab$variable))
      
      otu_label <-(paste0(anno$variable,": prev=", anno$lab))
      names(otu_label) <- anno$variable
      
      df_plot %<>%
        filter(value>0)%>%
        mutate(variable=as.factor(variable))
      
      col_df <- df_plot[,c("variable",paste0(names(df_plot)[j],"_trs"))]
      col_df <- unique(col_df)
      col_df<- col_df[match(levels(df_plot$variable),col_df$variable),]
      strips <- ggh4x::strip_themed(background_x = ggh4x::elem_list_rect(fill=ifelse(col_df[,2]=="No response",
                                                                                     "grey50",
                                                                                     ifelse(col_df[,2]=="Neg",
                                                                                            scales::alpha("darkred",.7),
                                                                                            scales::alpha("darkgreen",.7)))),
                                    text_x = ggh4x::elem_list_text(color=c("white"),
                                                                   face="bold",
                                                                   size=6))
      
      plot_tmp <- df_plot%>%
        ggplot(aes(y=value,x=.data[[names(df_plot)[j]]],color=.data[[paste0(names(df_plot)[j],"_trs")]]))+
        geom_point()+
        ylab("Relative Abundance")+
        xlab(colnames(df_plot)[j])+
        geom_smooth(method = "lm")+
        ggh4x::facet_wrap2(~ variable,scales = "free", ncol=ceiling(sqrt(length(unique(df_plot$variable)))),labeller = labeller(variable=otu_label),strip = strips)+
        scale_color_manual(name = "Response according to TITAN",values = c('Neg'='darkred','Pos'='darkgreen','No response'='grey50'))+
        theme(legend.position = "bottom",
              axis.text.x = element_text(angle=90))+
        theme_classic2()
      
      dir <- file.path(paste0("./articlez/gen1/resultats/final/titan/cor_relab/",i,"/",t)) 
      if (!dir.exists(dir)) dir.create(dir)
      
      
      ggsave(paste0("./articlez/gen1/resultats/final/titan/cor_relab/",i,"/",t,"/cor_relab_",gsub("/","",colnames(df_plot)[j]),".png"),plot_tmp,width=17,height = 10)
      
      
      lab <- df_plot2%>%group_by(variable)%>%reframe(prev=prevalence)%>% distinct(variable,prev) # get prevalence
      anno <- data.frame(lab = lab$prev,
                         variable =  as.factor(lab$variable))
      
      otu_label <-(paste0(anno$variable,": prev=", anno$lab))
      names(otu_label) <- anno$variable
      
      df_plot2 %<>%
        filter(value>0)%>%
        mutate(variable=as.factor(variable))
      
      col_df <- df_plot2[,c("variable",paste0(names(df_plot2)[j],"_trs"))]
      col_df <- unique(col_df)
      col_df<- col_df[match(levels(df_plot2$variable),col_df$variable),]
      strips <- ggh4x::strip_themed(background_x = ggh4x::elem_list_rect(fill=ifelse(col_df[,2]=="No response",
                                                                                     "grey50",
                                                                                     ifelse(col_df[,2]=="Neg",
                                                                                            scales::alpha("darkred",.7),
                                                                                            scales::alpha("darkgreen",.7)))),
                                    text_x = ggh4x::elem_list_text(color=c("white"),
                                                                   face="bold",
                                                                   size=6))
      
      plot_tmp_ab <- df_plot2%>%
        ggplot(aes(y=value,x=.data[[names(df_plot2)[j]]],color=.data[[paste0(names(df_plot2)[j],"_trs")]]))+
        geom_point()+
        ylab("Abundance (nb reads)")+
        xlab(colnames(df_plot2)[j])+
        geom_smooth(method = "lm")+
        ggh4x::facet_wrap2(~ variable,scales = "free", ncol=ceiling(sqrt(length(unique(df_plot2$variable)))),labeller = labeller(variable=otu_label),strip = strips)+
        scale_color_manual(name = "Response according to TITAN",values = c('Neg'='darkred','Pos'='darkgreen','No response'='grey50'))+
        theme(legend.position = "bottom",
              axis.text.x = element_text(angle=90))+
        theme_classic2()
      
      dir <- file.path(paste0("./articlez/gen1/resultats/final/titan/cor_ab/",i,"/",t)) 
      if (!dir.exists(dir)) dir.create(dir)
      
      
      ggsave(paste0("./articlez/gen1/resultats/final/titan/cor_ab/",i,"/",t,"/cor_ab_",gsub("/","",colnames(df_plot2)[j]),".png"),plot_tmp_ab,width=17,height = 10)
    }
  }
}
```



## OTUs selection

### Select OTUs

```{r}
slctd_otus_titan <- NULL
diff_titan_diablo <- NULL
slctd_otus_diablo <- NULL
for(i in datasets){
    titan_tmp <- titan_list[[i]]$hell$dontrm$nb_resp$notonly
    titan_otus_tmp <- gsub("M","",rownames(titan_tmp))
    
    diablo_tmp <- info_paper_diablo[[i]]$hell$otus$cleannames
    diablo_otu_tmp <- info_paper_diablo[[i]]$hell$interpratedotus
    
    diff_titan_diablo[[i]] <- setdiff(titan_otus_tmp,diablo_otu_tmp)
    slctd_otus_titan[[i]] <- titan_otus_tmp
    slctd_otus_diablo[[i]] <- diablo_otu_tmp
}
```

### Heatmap

```{r}
heatmaps_list <- NULL
heatmaps_list_logd <- NULL
for(i in datasets){
  data_tmp <- get(i)
  data_tmp <- clone(data_tmp)
  
  for(j in c("titan","diablo")){
    
    slctd_otus <- get(paste0("slctd_otus_",j))[[i]]
    
    tset <- data_tmp$otu_table%>%
      t()%>%
      data.frame()%>%
      mutate(samp = rownames(.))%>%
      reshape2::melt(id.vars = "samp")
    
    prev <- data.frame(preva=rowSums(data_tmp$otu_table>1))%>%
      mutate(variable=rownames(.))
    
    
    tax_dat <- data_tmp$tax_table
    
    tax_dat <- tax_dat[gsub("^M","",rownames(tax_dat))%in%slctd_otus,] # extract taxonomic info about our OTUs
    
    tax_dat %<>% 
      mutate(class=gsub("c__","",Class_conf))%>% #save class info
      unite(c(1:6),sep=";",col=Taxonomy)%>% # unite taxo info
      mutate(low_tax = gsub("(;[a-z]__)+?$","",Taxonomy))%>%
      mutate(low_tax = str_extract(pattern = "([a-z]__[^;]+?$)",string=low_tax))%>% # get lowest taxnomical info
      mutate(lvl_lowtax = str_extract(pattern = "[a-z]__",string=low_tax))%>% # get associated tax level
      mutate(low_tax = gsub("[a-z]__","",low_tax))%>%
      mutate(lvl_lowtax = ifelse(lvl_lowtax == "g__", "sp.", 
                                 ifelse(lvl_lowtax == "f__", "(Family)",
                                        ifelse(lvl_lowtax == 'o__', "(Order)","(Class)"))))
    
    tax_dat%<>%
      mutate(variable=gsub("^M","",rownames(.)))%>%
      mutate(Var2 = paste0(gsub("^M","",variable)," ", low_tax, " ", lvl_lowtax))
    tset%<>%
      left_join(prev)%>%
      mutate(variable=gsub("M","",variable))
    
    tset%<>%
      group_by(samp)%>%
      mutate(relab=(value/sum(value))*100)%>%
      ungroup()%>%
      filter(variable%in%gsub("^M","",slctd_otus))%>%
      left_join(tax_dat)%>%
      arrange(preva)%>%
      mutate(Var2=fct_inorder(Var2))
    
    leaftmp <- data_leaf[str_extract(data_leaf$ID,"[0-9]{2}")%in%str_extract(tset$samp,"[0-9]{2}"),c("ID","Nmg","Nb_leaves","Nb_Fr")]
    leaftmp%<>%mutate(ID=str_extract(leaftmp$ID,"[0-9]{2}"))
    tset%<>%mutate(ID=str_extract(tset$samp,"[0-9]{2}"))
    tset%<>%left_join(leaftmp)
    tset%<>%arrange(Nb_Fr)%>%mutate(samp=fct_inorder(samp))
    
    p1 <-tset%>%
      ggplot()+
      # geom_point(mapping=aes(x=samp,y=Nmg),color="darkcyan",size=1)+
      # geom_line(mapping=aes(x=samp,y=Nmg),group=1,color="darkcyan")+
      # geom_point(mapping=aes(x=samp,y=Nb_leaves),color="burlywood",size=1)+
      # geom_line(mapping=aes(x=samp,y=Nb_leaves),group=1,color="burlywood")+
      geom_point(mapping=aes(x=samp,y=Nb_Fr),color="yellow3",size=1)+
      geom_line(mapping=aes(x=samp,y=Nb_Fr),group=1,color="yellow3")+
      annotate(geom="text",label="Number of flowers", x= 20,y=400,fontface=2,color="yellow3")+
      theme_classic2()+
      theme(axis.text.x = element_blank(),
            axis.ticks.x = element_blank(),
            axis.title = element_blank())+
      scale_y_continuous(breaks = trans_breaks(identity, identity, n = 3))+
      theme(axis.text = element_text(face="bold")) 
    
    # p5 <- tibble(x="mid",y=unlist(p1$data[length(p1$data$samp),c("Nmg","Nb_leaves")]),label=c("Nitrogen","Nb leaves"))%>%
    #   ggplot(aes(x=x,y=y,label=label,color=label))+
    #   geom_text(vjust=-.25,fontface="bold",size=2)+
    #   theme_void()+
    #   ylim(c(layer_scales(p1)$y$range$range[1],layer_scales(p1)$y$range$range[2]))+
    #   theme(legend.position = 'none')+
    #   scale_color_manual(values=c("burlywood","darkcyan"))+
    #   scale_x_discrete(expand = c(0,0))
    
    # p5 <- tibble(x="mid",y=unlist(p1$data[length(p1$data$samp),c("Nb_Fr")]),label=c("Nb Flowers"))%>%
    #   ggplot(aes(x=x,y=y,label=label,color=label))+
    #   geom_text(vjust=-.25,fontface="bold",size=4)+
    #   theme_void()+
    #   ylim(c(layer_scales(p1)$y$range$range[1],layer_scales(p1)$y$range$range[2]))+
    #   theme(legend.position = 'none')+
    #   scale_color_manual(values=c("yellow3"))+
    #   scale_x_discrete(expand = c(0,0))
    
    p2 <- tset%>%
      mutate(relab=ifelse(relab==0,NA,relab))%>%
      ggplot(aes(x=samp,y=Var2,fill=relab))+
      geom_raster()+
      theme(axis.text.x = element_blank(),
            axis.ticks.x = element_blank(),
            axis.title.x = element_blank(),
            axis.title.y = element_blank(),
            panel.background = element_blank(),
            panel.border = element_blank(),
            panel.spacing = element_blank(),
            plot.margin = unit(c(0, 0, 0, 0), "lines"))+
      scale_y_discrete(expand = c(0,0))+
      scale_fill_gradient2(low="azure",mid="lightblue",high="darkblue", name="Relative abundance (%)")+
      theme(axis.text.y = element_text(face="bold")) 
    
    p3 <- data_tmp$sample_table%>%
      mutate(sample_id=fct_relevel(sample_id,levels(tset$samp)))%>%
      ggplot(aes(x= sample_id,y=1,fill=Trtmt))+
      geom_raster()+
      scale_fill_manual(values=mo_col)+
      scale_y_discrete(expand = c(0,0))+
      theme_void()+
      theme(legend.position = "n")
    
    p4 <- tset%>%
      ggplot(aes(x=1,y=variable,fill=class))+
      geom_raster()+
      scale_fill_manual(values=colorRampPalette(brewer.pal(8, "Dark2"))(length(unique(tset$class))))+
      theme(axis.text = element_blank(),
            axis.ticks = element_blank(),
            axis.title = element_blank(),
            panel.background = element_blank())
    
    heatmaps_list[[i]] <- (p1+
                             plot_spacer()+plot_layout(widths = c(12,1)))/
      (p2+p4+plot_layout(widths = c(12,1),guides = "collect"))/
      (p3+plot_spacer()+plot_layout(widths = c(12,1)))+
      plot_layout(heights = c(1,10,1))
    ggsave(filename = paste0("./articlez/gen1/resultats/final/heatmaps/relab_",i,"",j,".png"),plot =  heatmaps_list[[i]],height = 14,width=10)
    
    p6 <-  tset%>%
      ggplot(aes(x=samp,y=Var2,fill=log(relab)))+
      geom_raster()+
      theme(axis.text.x = element_blank(),
            axis.ticks.x = element_blank(),
            axis.title.x = element_blank(),
            axis.title.y = element_blank(),
            panel.background = element_blank(),
            plot.margin = unit(c(0, 0, 0, 0), "lines"))+
      scale_fill_gradient2(low="azure",mid="lightblue",high="darkblue", name="Logged relative abundance")+
      theme(axis.text.y = element_text(face="bold")) 
    
    
    heatmaps_list_logd[[i]]<- (p1+plot_spacer()+plot_layout(widths = c(12,1)))/
      (p6+p4+plot_layout(widths = c(12,1),guides = "collect"))/
      (p3+plot_spacer()+plot_layout(widths = c(12,1)))+
      plot_layout(heights = c(1,10,1))
    
    ggsave(filename = paste0("./articlez/gen1/resultats/final/heatmaps/ab_",i,"",j,".png"),plot =  heatmaps_list_logd[[i]]&theme(text = element_text(face="bold")),height = 14,width = 10)
  }
}
```

### prev\>5% OTUs

```{r}
ubiquitous <- NULL
ana2test <- c("titan","diablo")
p3 <- NULL
for(i in datasets){
  data_tmp <- get(i)
  data_tmp <- clone(data_tmp)
  
  otu_tmp <- data_tmp$otu_table%>%
    mutate(otus=gsub("M","",rownames(.)))%>%
    reshape2::melt(id.var="otus")%>%
    dplyr::rename(sample_id=variable)%>%
    left_join(data_tmp$sample_table)%>%
    group_by(Trtmt,otus)%>%
    dplyr::summarise(prev=sum(value>0),
                     reads=sum(value))%>%
    filter(reads!=0)%>%
    filter(otus%in%gsub("M","",colnames(diablo_mod[[i]]$hell$final$X$OTUs_com)))
  
  p1 <- otu_tmp%>%
    ggplot(aes(x=reorder(otus,-prev,sum),y=prev,fill=Trtmt))+
    geom_bar(stat = "identity", position = "stack",width=1)+
    scale_fill_manual(name="Treatments",values=mo_col)+
    scale_y_continuous(expand=c(0,0))+
    theme_classic2()+
    theme(axis.text.x = element_blank(),
          axis.ticks.x = element_blank())+
    ylab("Number of samples in which OTUs are found")+
    theme(axis.text.x = element_blank(),
          axis.ticks.x = element_blank(),
          axis.title.y = element_text(size=8),
          axis.title.x = element_blank())
  
  p2 <- otu_tmp%>%
    group_by(Trtmt)%>%
    mutate(relab=reads/sum(reads))%>%
    ggplot(aes(x=reorder(otus,-prev,sum),y=Trtmt,fill=100*(relab)))+
    geom_raster()+
    scale_y_discrete(limits=rev)+ 
    theme(axis.text.x = element_blank(),
          axis.ticks.x = element_blank(),
          panel.grid = element_blank(),
          panel.background = element_blank(),
          axis.title.y = element_blank(),
          axis.title.x = element_blank(),
          plot.margin = unit(c(0, 0, 0, 0), "lines"))+
    scale_fill_gradientn(colors=c("#B3E7EF","#71A3C7","#344B70","#0C113D"), name="Relative abundance within treatment (%)",breaks=seq(0,25,5),
                         limits=c(0,25))
  
  
  for(j in ana2test){
    
    slctd_otus <- get(paste0("slctd_otus_",j))[[i]]
    
    
    p3[[j]] <- otu_tmp%>%
      mutate(is_slctd=ifelse(otus%in%gsub("M","",slctd_otus),"yes","no"))%>%
      ggplot(aes(x=reorder(otus,-prev,sum),y=1,fill=is_slctd))+
      geom_raster()+ 
      theme_void()+
      ylab(ifelse(j=="titan","TITAN",ifelse(j=="spls","sPLS","DIABLO")))+
      theme(axis.title.y = element_text())+
      scale_fill_manual(values=c("no"="grey",
                                 "yes"=ifelse(grepl("16s",i),"#E09322","#3F5E61")),
                        name=`if`(grepl("16s",i),"Selected OTU",""),
                        labels=`if`(grepl("16s",i),c("No","Bacteria"),c("Fungi")),
                        breaks = `if`(grepl("16s",i),c("no","yes"),c("yes")))
    
    
    
    
  }
  ubiquitous[[i]] <- p1+p2+p3[["titan"]]+p3[["diablo"]]+theme(legend.position = "n")+plot_layout(heights = c(6,4,.5,.5,.5))
  print(paste0("OTUs present in more than ",ceiling(0.05*ncol(data_tmp$otu_table))," sample: ",round(length(unique(otu_tmp$otus))/nrow(data_tmp$otu_table),digits = 2)))
}
```

```{r}
for(i in datasets){
  ggsave(paste0("./articlez/gen1/resultats/final/ubiquitous_",i,".png"),ubiquitous[[i]],height = 5,width=7)
}


p_16s <- ubiquitous$meco_mo16s_seqsg_raref
p_its <- ubiquitous$meco_moits_seqsg_raref

p_16s[[1]] <- p_16s[[1]]+
  ggtitle("Bacteria")+
  theme(plot.title = element_text(face = "bold"))+
  theme(axis.title.x = element_blank())

p_16s[[3]]+
  theme(legend.spacing = unit(3.0, 'cm')) +
  ## important additional element
  guides(fill = guide_legend(byrow = TRUE))


p_its[[1]] <- p_its[[1]]+
  theme(legend.position = 'n')+
  theme(axis.title.y = element_blank())+
  ggtitle("Fungi")+
  theme(plot.title = element_text(face = "bold"))
p_its[[2]] <- p_its[[2]]+
  theme(legend.position = 'n')+
  theme(axis.text.y = element_blank())
p_its[[3]] <- p_its[[3]]+
  theme(axis.title.y = element_blank())
p_its[[4]] <- p_its[[4]]+
  theme(axis.title.y = element_blank())


(p_16s|p_its)+plot_layout(guides = "collect")&theme(legend.position = "bottom",
                                                    legend.direction = "horizontal",
                                                    legend.title.position = "top",
                                                    legend.title = element_text(hjust = .5))

ggsave("./articlez/gen1/resultats/final/ubiquitous.png",(p_16s|p_its)+plot_layout(guides = "collect")&theme(legend.position = "bottom",
                                                                                                            legend.direction = "horizontal",
                                                                                                            legend.title.position = "top",
                                                                                                            legend.title = element_text(hjust = .5)),height = 6,width=10)

p_16s_x <- (p_16s[[1]]/p_16s[[3]]/p_16s[[4]])+plot_layout(heights = c(1,.05,.05))
p_its_x <- (p_its[[1]]/p_its[[3]]/p_its[[4]])+plot_layout(heights = c(1,.05,.05))
(p_16s_x|p_its_x)+plot_layout(guides = "collect")&theme(legend.position = "bottom",
                                                        legend.direction = "horizontal",
                                                        legend.title.position = "top",
                                                        legend.title = element_text(hjust = .5))
ggsave("./articlez/gen1/resultats/final/ubiquitous_x.png",(p_16s_x|p_its_x)+plot_layout(guides = "collect")&theme(legend.position = "bottom",
                                                                                                                  legend.direction = "horizontal",
                                                                                                                  legend.title.position = "top",
                                                                                                                  legend.title = element_text(hjust = .5)),
       height = 5,width = 10)
```

### prev\<5% OTU

```{r}
comp_plot_relablowprev <- NULL
comp_plot_relabtrtmtlowprev <- NULL
comp_plot_relallowprev <- NULL
comp_plot_relaltrtmtlowprev <- NULL
for(i in datasets){
  data_tmp <- get(i)
  data_tmp <- clone(data_tmp)
  
  prev_thshld <- ceiling(0.05*nrow(data_tmp$sample_table))
  
  
  for(j in c("titan","diablo")){
    
    slctd_otus <- get(paste0("slctd_otus_",j))[[i]]
    
    low_prev <- data_tmp$otu_table%>%
      t()%>%
      as.data.frame()%>%
      summarise(across(1:(ncol(.)), ~sum(.x>0)))%>% # otu prevalence
      set_rownames("prev")%>%
      t()%>%
      as.data.frame()%>%
      mutate(lowprev=ifelse(prev<prev_thshld,"yup",
                            ifelse(gsub("^M","",rownames(.))%in%slctd_otus,"Selected","Not selected")))
    
    df_p <- merge(data_tmp$otu_table,low_prev["lowprev"],by="row.names",all.x=TRUE)[,-1]
    title_plot <- paste0(ifelse(grepl("16s",i),"Bacteria - ","Fungi - "),ifelse(j=="titan","TITAN",
                                                                                "DIABLO"))
    
    comp_plot_relablowprev[[i]][[j]] <- (df_p%>%
                                           group_by(lowprev)%>%
                                           summarise(across(1:(ncol(.)-1),sum))%>%
                                           mutate(across(where(is.numeric),~(./sum(.))*100))%>%
                                           reshape2::melt(id.vars="lowprev")%>%
                                           dplyr::rename(sample_id=variable)%>%
                                           left_join(data_tmp$sample_table)%>%
                                           mutate(lowprev=fct_relevel(lowprev,"yup","Not selected","Selected"))%>%
                                           arrange(Trtmt,sample_id)%>%
                                           mutate(sample_id=fct_inorder(sample_id))%>%
                                           ggplot(aes(x=sample_id,y=value,fill=lowprev))+
                                           geom_bar(stat = "identity", position = "stack",width=1)+
                                           scale_fill_manual(values=c("yup"="grey23",
                                                                      "Not selected"="grey",
                                                                      "Selected"=`if`(grepl("16s",i),"#E09322","#3F5E61")),
                                                             breaks =`if`(grepl("16s",i),
                                                                          c("yup","Not selected","Selected"),
                                                                          "Selected"),
                                                             labels=`if`(grepl("16s",i),
                                                                         c( "Low prevalence",
                                                                            paste0("Prevalence >",prev_thshld," but not selected OTUs"),
                                                                            paste0("Prevalence >",prev_thshld," and selected OTUs")),
                                                                         paste0("Prevalence >",prev_thshld," and selected OTUs")),
                                                             name="")+
                                           theme_classic2()+
                                           theme(axis.text.x = element_blank(),
                                                 axis.ticks.x = element_blank(),
                                                 axis.title.x = element_blank(),
                                                 axis.title.y=element_text(face="bold"),
                                                 axis.text.y=element_text(face="bold"))+
                                           scale_y_continuous(expand=c(0,0))+
                                           ylab("Relative abundance within samples (%)")+
                                           ggtitle(title_plot)+
                                           guides(fill=guide_legend(ncol=2)))/
      (data_tmp$sample_table%>%
         arrange(Trtmt,sample_id)%>%
         mutate(sample_id=fct_inorder(sample_id))%>%
         ggplot(aes(x=sample_id,y=1,fill=Trtmt))+
         geom_raster()+
         scale_fill_manual(values=mo_col,guide = "none")+
         scale_x_discrete(expand = c(0,0))+
         theme_void()+
         theme(legend.position = "n"))+
      plot_layout(heights = c(9,.9))
    
    if(grepl("its",i)){comp_plot_relablowprev[[i]][[j]][[1]] <- comp_plot_relablowprev[[i]][[j]][[1]]+theme(axis.text.y = element_blank(),
                                                                                                            axis.title.y = element_blank())}
    
    comp_plot_relallowprev[[i]][[j]] <-  (df_p%>%
                                            group_by(lowprev)%>%
                                            summarise(across(1:(ncol(.)-1),~sum(.>0)))%>%
                                            mutate(across(where(is.numeric),~(./sum(.))*100))%>%
                                            reshape2::melt(id.vars="lowprev")%>%
                                            dplyr::rename(sample_id=variable)%>%
                                            left_join(data_tmp$sample_table)%>%
                                            mutate(lowprev=fct_relevel(lowprev,"yup","Not selected","Selected"))%>%
                                            arrange(Trtmt,sample_id)%>%
                                            mutate(sample_id=fct_inorder(sample_id))%>%
                                            ggplot(aes(x=sample_id,y=value,fill=lowprev))+
                                            geom_bar(stat = "identity", position = "stack",width=1)+
                                            scale_fill_manual(values=c("yup"="grey23",
                                                                       "Not selected"="grey",
                                                                       "Selected"=`if`(grepl("16s",i),"#E09322","#3F5E61")),
                                                              breaks =`if`(grepl("16s",i),
                                                                           c("yup","Not selected","Selected"),
                                                                           "Selected"),
                                                              labels=`if`(grepl("16s",i),
                                                                          c( "Low prevalence",
                                                                             paste0("Prevalence >",prev_thshld," but not selected OTUs"),
                                                                             paste0("Prevalence >",prev_thshld," and selected OTUs")),
                                                                          paste0("Prevalence >",prev_thshld," and selected OTUs")),
                                                              name="")+
                                            theme_classic2()+
                                           theme(axis.text.x = element_blank(),
                                                 axis.ticks.x = element_blank(),
                                                 axis.title.x = element_blank(),
                                                 axis.title.y=element_text(face="bold"),
                                                 axis.text.y=element_text(face="bold"))+
                                            scale_y_continuous(expand=c(0,0))+
                                            ylab("Proportion of OTU within samples (%)")+
                                            ggtitle(title_plot)+
                                            guides(fill=guide_legend(ncol=2)))/
      (data_tmp$sample_table%>%
         arrange(Trtmt,sample_id)%>%
         mutate(sample_id=fct_inorder(sample_id))%>%
         ggplot(aes(x=sample_id,y=1,fill=Trtmt))+
         geom_raster()+
         scale_fill_manual(values=mo_col,guide = "none")+
         scale_x_discrete(expand = c(0,0))+
         theme_void()+
         theme(legend.position = "n"))+
      plot_layout(heights = c(9,.9))
    
    if(grepl("its",i)){comp_plot_relallowprev[[i]][[j]][[1]] <- comp_plot_relallowprev[[i]][[j]][[1]]+theme(axis.text.y = element_blank(),
                                                                                                            axis.title.y = element_blank())}
    
    comp_plot_relaltrtmtlowprev[[i]][[j]] <-  (df_p%>%
                                                 group_by(lowprev)%>%
                                                 summarise(across(1:(ncol(.)-1),.fns=~sum(.>0)))%>%
                                                 mutate(across(where(is.numeric),~(./sum(.))*100))%>%
                                                 reshape2::melt(id.vars="lowprev")%>%
                                                 dplyr::rename(sample_id=variable)%>%
                                                 left_join(data_tmp$sample_table)%>%
                                                 group_by(Trtmt,lowprev)%>%
                                                 summarise(mean_relab=mean(value))%>%
                                                 mutate(lowprev=fct_relevel(lowprev,"yup","Not selected","Selected"))%>%
                                                 ggplot(aes(x=Trtmt,y=mean_relab,fill=lowprev))+
                                                 geom_bar(stat = "identity", position = "stack")+
                                                 scale_fill_manual(values=c("yup"="grey23",
                                                                            "Not selected"="grey",
                                                                            "Selected"=`if`(grepl("16s",i),"#E09322","#3F5E61")),
                                                                   breaks =`if`(grepl("16s",i),
                                                                                c("yup","Not selected","Selected"),
                                                                                "Selected"),
                                                                   labels=`if`(grepl("16s",i),
                                                                               c( "Low prevalence",
                                                                                  paste0("Prevalence >",prev_thshld," but not selected OTUs"),
                                                                                  paste0("Prevalence >",prev_thshld," and selected OTUs")),
                                                                               paste0("Prevalence >",prev_thshld," and selected OTUs")),
                                                                   name="")+
                                                 theme_classic2()+                                    
                                                  theme(axis.text.x = element_blank(),
                                                 axis.ticks.x = element_blank(),
                                                 axis.title.x = element_blank(),
                                                 axis.title.y=element_text(face="bold"),
                                                 axis.text.y=element_text(face="bold"))+
                                                 scale_y_continuous(expand=c(0,0))+
                                                 ylab("Proportion of OTU Richness within samples (%)")+
                                                 guides(fill=guide_legend(ncol=2)))
    
    if(grepl("its",i)){comp_plot_relaltrtmtlowprev[[i]][[j]] <- comp_plot_relaltrtmtlowprev[[i]][[j]]+theme(axis.text.y = element_blank(),
                                                                                                            axis.title.y = element_blank())}
    if(grepl("titan",j)){comp_plot_relaltrtmtlowprev[[i]][[j]] <- comp_plot_relaltrtmtlowprev[[i]][[j]]+theme(axis.text.x = element_blank(),
                                                                                                              axis.title.x = element_blank(),
                                                                                                              axis.ticks.x = element_blank())}
    
    comp_plot_relabtrtmtlowprev[[i]][[j]] <-  (df_p%>%
                                                 group_by(lowprev)%>%
                                                 summarise(across(1:(ncol(.)-1),.fns=sum))%>%
                                                 mutate(across(where(is.numeric),~(./sum(.))*100))%>%
                                                 reshape2::melt(id.vars="lowprev")%>%
                                                 dplyr::rename(sample_id=variable)%>%
                                                 left_join(data_tmp$sample_table)%>%
                                                 group_by(Trtmt,lowprev)%>%
                                                 summarise(mean_relab=mean(value))%>%
                                                 mutate(lowprev=fct_relevel(lowprev,"yup","Not selected","Selected"))%>%
                                                 ggplot(aes(x=Trtmt,y=mean_relab,fill=lowprev))+
                                                 geom_bar(stat = "identity", position = "stack")+
                                                 scale_fill_manual(values=c("yup"="grey23",
                                                                            "Not selected"="grey",
                                                                            "Selected"=`if`(grepl("16s",i),"#E09322","#3F5E61")),
                                                                   breaks=`if`(grepl("16s",i),
                                                                               c("yup","Not selected","Selected"),
                                                                               "Selected"),
                                                                   labels=`if`(grepl("16s",i),
                                                                               c( "Low prevalence",
                                                                                  paste0("Prevalence >",prev_thshld," but not selected OTUs"),
                                                                                  paste0("Prevalence >",prev_thshld," and selected OTUs")),
                                                                               paste0("Prevalence >",prev_thshld," and selected OTUs")),
                                                                   name="")+
                                                 theme_classic2()+
                                                  theme(axis.text.x = element_blank(),
                                                 axis.ticks.x = element_blank(),
                                                 axis.title.x = element_blank(),
                                                 axis.title.y=element_text(face="bold"),
                                                 axis.text.y=element_text(face="bold"))+
                                                 scale_y_continuous(expand=c(0,0))+
                                                 ylab("Mean relative abundance within samples (%)")+
                                                 ggtitle(title_plot)+
                                                 guides(fill=guide_legend(ncol=2)))
    
    if(grepl("its",i)){comp_plot_relabtrtmtlowprev[[i]][[j]] <- comp_plot_relabtrtmtlowprev[[i]][[j]]+theme(axis.text.y = element_blank(),
                                                                                                            axis.title.y = element_blank())}
    if(grepl("titan",j)){comp_plot_relabtrtmtlowprev[[i]][[j]] <- comp_plot_relabtrtmtlowprev[[i]][[j]]+theme(axis.text.x = element_blank(),
                                                                                                              axis.ticks.x = element_blank())}
  }
}
```

```{r}
comp_relab_slctd <- ((comp_plot_relablowprev$meco_mo16s_seqsg_raref$titan[[1]]|comp_plot_relablowprev$meco_moits_seqsg_raref$titan[[1]])/
                       (comp_plot_relablowprev$meco_mo16s_seqsg_raref$diablo|comp_plot_relablowprev$meco_moits_seqsg_raref$diablo))+
  plot_layout(guides = "collect")&theme(legend.position = "bottom")

ggsave("./articlez/gen1/resultats/final/comp_relab_slctd.png",comp_relab_slctd,width = 9,height = 10)


comp_relal_slctd <- ((comp_plot_relallowprev$meco_mo16s_seqsg_raref$titan[[1]]|comp_plot_relallowprev$meco_moits_seqsg_raref$titan[[1]])/
                       (comp_plot_relallowprev$meco_mo16s_seqsg_raref$diablo|comp_plot_relallowprev$meco_moits_seqsg_raref$diablo))+
  plot_layout(guides = "collect")&theme(legend.position = "bottom")

ggsave("./articlez/gen1/resultats/final/comp_relal_slctd.png",comp_relal_slctd,width = 9,height = 10)




comp_relabtrtmt_slctd <- ((comp_plot_relabtrtmtlowprev$meco_mo16s_seqsg_raref$titan|comp_plot_relabtrtmtlowprev$meco_moits_seqsg_raref$titan)/
                            (comp_plot_relabtrtmtlowprev$meco_mo16s_seqsg_raref$diablo|comp_plot_relabtrtmtlowprev$meco_moits_seqsg_raref$diablo))+
  plot_layout(guides = "collect")&theme(legend.position = "bottom")

ggsave("./articlez/gen1/resultats/final/comp_relabtrtmt_slctd.png",comp_relabtrtmt_slctd,width = 9,height = 10)


comp_relaltrtmt_slctd <- ((comp_plot_relaltrtmtlowprev$meco_mo16s_seqsg_raref$titan|comp_plot_relaltrtmtlowprev$meco_moits_seqsg_raref$titan)/
                            (comp_plot_relaltrtmtlowprev$meco_mo16s_seqsg_raref$diablo|comp_plot_relaltrtmtlowprev$meco_moits_seqsg_raref$diablo))+
  plot_layout(guides = "collect")&theme(legend.position = "bottom")


ggsave("./articlez/gen1/resultats/final/comp_relaltrtmt_slctd.png",comp_relaltrtmt_slctd,width = 9,height = 10)

((comp_plot_relablowprev$meco_mo16s_seqsg_raref$titan[[1]]|comp_plot_relablowprev$meco_moits_seqsg_raref$titan[[1]])/
   (comp_plot_relablowprev$meco_mo16s_seqsg_raref$diablo|comp_plot_relablowprev$meco_moits_seqsg_raref$diablo))+
  plot_layout(guides = "collect")&theme(legend.position = "bottom")

```

### Titan/Diablo relab summary

```{r}
summary_slct_otus <- data.frame()
df_plot_smry_trtmt_slct <- data.frame()
for(i in datasets){
  for(j in c("titan","diablo")){
    for(k in c("relablowprev","relallowprev")){
      p_tmp <- get(paste0("comp_plot_",k))
      p_tmp <- p_tmp[[i]][[j]]
      out_tmp <- p_tmp[[1]]$data%>%
        filter(lowprev=="Selected")%>%
        group_by(Trtmt)%>%
        summarise(mean_val=mean(value))%>%
        mutate(analysis=j,
               marker=ifelse(grepl("16s",i),"bacteria","fungi"),
               response=ifelse(grepl("al",k),"alpha","relab"))
      summary_slct_otus<- rbind(summary_slct_otus,out_tmp)
      
      out_tmp2 <-  p_tmp[[1]]$data%>%
        filter(lowprev=="Selected")%>%
        mutate(analysis=j,
               marker=ifelse(grepl("16s",i),"bacteria","fungi"),
               response=ifelse(grepl("al",k),"alpha","relab"))
      
      
      df_plot_smry_trtmt_slct <- rbind(df_plot_smry_trtmt_slct,out_tmp2)
    }
  }
  
}
summary_slct_otus%>%
  flextable()%>%
  save_as_docx(path="./articlez/gen1/resultats/final/slct_smry_trtmt.docx")


plot_smry_slctd <-  ((df_plot_smry_trtmt_slct%>%
                        filter(response=="alpha")%>%
                        mutate(marker=as.factor(marker))%>%
                        ggplot(aes(x=Trtmt,y=value,fill=marker))+
                        geom_boxplot()+
                        facet_wrap(~analysis)+
                        ylab("Proportion of the sample OTUs (%)")+
                        ylim(c(0,100))+
                        scale_fill_manual(values =c(bacteria="#E09322",fungi="#3F5E61"),
                                          name=c("Group"))+
                        theme_classic2()+
                        xlab(""))/
                       (df_plot_smry_trtmt_slct%>%
                          filter(response=="relab")%>%
                          mutate(marker=as.factor(marker))%>%
                          ggplot(aes(x=Trtmt,y=value,fill=marker))+
                          geom_boxplot()+
                          facet_wrap(~analysis)+
                          ylab("Relative abundance within samples (%)")+
                          ylim(c(0,100))+
                          scale_fill_manual(values =c(bacteria="#E09322",fungi="#3F5E61"),
                                            name=c("Group"))+
                          theme_classic2()+
                          xlab("")))+
  plot_layout(guides="collect")

ggsave("./articlez/gen1/resultats/final/smry_slct.png",plot_smry_slctd,width = 8,height = 7)
```

```{r}
source("./Scripts/functions/cld_bxplot_wilcox_2fac.R")

smry_trtmt_slct_list <- NULL
for(resp in c("alpha","relab")){
  for(ana2test in c("titan","diablo")){
    
    ylab_tmp <- ifelse(resp=="alpha","Proportion of the sample OTUs (%)","Proportion of the sample reads (%)")
    
    plot_tmp <- df_plot_smry_trtmt_slct%>%
      filter(response==resp,
             analysis==ana2test)%>%
      mutate(marker=as.factor(marker),
             analysis=toupper(analysis))%>%
      ggplot(aes(x=Trtmt,y=value,fill=marker))+
      geom_boxplot()+
      facet_wrap(~analysis)+
      ylab(ylab_tmp)+
      ylim(c(0,100))+
      scale_fill_manual(values =c(bacteria="#E09322",fungi="#3F5E61"),
                        name=c("Group"))+
      theme_classic2()+
      xlab("")+
      theme(strip.text = element_text(face="bold"))
    
    
    smry_trtmt_slct_list[[resp]][[ana2test]] <- cld_bxplot_wilcox_2fac(plot_tmp$data,
                                                                       "Trtmt",
                                                                       "value",
                                                                       "marker",
                                                                       plot = plot_tmp,nudge = .05)
    
    ggsave(paste0("./articlez/gen1/resultats/final/titan_diablo_smry_",resp,"_",ana2test,".png"),smry_trtmt_slct_list[[resp]][[ana2test]][["daplot"]],width=6,height =4)
  }
}

average_smry_plot <- df_plot_smry_trtmt_slct%>%
  group_by(marker,analysis,response)%>%
  summarise(mean_value=mean(value))%>%
  mutate(analysis=toupper(analysis))%>%
  mutate(response=ifelse(response=="alpha","Number of OTUs","Number of reads"))%>%
  ggplot(aes(x=response,y=mean_value,fill=marker))+
  geom_bar(stat = "identity",position = position_dodge())+
  facet_wrap(~analysis)+
  scale_fill_manual(values =c(bacteria="#E09322",fungi="#3F5E61"),
                    name=c("Group"))+
  theme_classic2()+
  ylab("Mean proportion (%)")+
  xlab("")+
  theme(strip.text = element_text(face="bold"),axis.text.x = element_text(size=6))+
  scale_y_continuous(expand=c(0,0),limits=c(0,100))
ggsave("./articlez/gen1/resultats/final/titan_diablo_avrg_smry.png",average_smry_plot,width=5,height = 3)

df_plot_smry_trtmt_slct%>%
  group_by(marker,analysis,response)%>%
  summarise(mean_value=mean(value))
```

```{r}
prev_analysis <- NULL
relab_analysis <- NULL
for(i in datasets){
  titotu <- rownames(titan_list[[i]]$dontrm$nb_resp$notonly)
  diablotu <- paste0("M",info_paper_diablo[[i]]$otus$cleannames)
  
  data_tmp <- clone(get(i))
  prev_titotu <- data_tmp$otu_table%>%
    filter(rownames(.)%in%titotu)
  prev_diablotu<- data_tmp$otu_table%>%
    filter(rownames(.)%in%diablotu)
  
  prev_titotu <- rowSums(prev_titotu>0)%>%
    reshape2::melt()%>%
    mutate(otu=gsub("M","",rownames(.)),
           analysis="TITAN")
  prev_diablotu <- rowSums(prev_diablotu>0)%>%
    reshape2::melt()%>%
    mutate(otu=gsub("M","",rownames(.)),
           analysis="DIABLO")
  
  prev_analysis[[i]] <- mutate(rbind(prev_titotu,prev_diablotu),marker=ifelse(grepl("16s",i),"Bacteria","Fungi"))
  
  rel_titotu <- data_tmp$otu_table%>%
    filter(rownames(.)%in%titotu)%>%
    mutate(across(where(is.numeric),~.x/sum(.)))%>%
    reshape2::melt()%>%
    mutate(otu=gsub("M","",rownames(.)),
           analysis="TITAN")%>%
    filter(value!=0)
  rel_diablotu <- data_tmp$otu_table%>%
    filter(rownames(.)%in%diablotu)%>%
    mutate(across(where(is.numeric),~.x/sum(.)))%>%
    reshape2::melt()%>%
    mutate(otu=gsub("M","",rownames(.)),
           analysis="DIABLO")%>%
    filter(value!=0)
  
  relab_analysis[[i]] <- mutate(rbind(rel_titotu,rel_diablotu),marker=ifelse(grepl("16s",i),"Bacteria","Fungi"))
  
}

#26677FFF #635C72FF #89374FFF #C46F3EFF #B79E4BFF

strips <- ggh4x::strip_themed(text_x = ggh4x::elem_list_text(face="bold",color="white"),
                              background_x = ggh4x::elem_list_rect(fill=c(Bacteria="#E09322",Fungi="#3F5E61")))

bxp <- rbind(prev_analysis$meco_mo16s_seqsg_raref,prev_analysis$meco_moits_seqsg_raref)%>%
  ggplot(aes(x=analysis,y=value))+
  geom_boxplot(aes(fill=analysis))+
  ggh4x::facet_grid2(~marker,strip = strips)+
  theme_classic2()+
  scale_fill_manual(values =c("#635C72FF","#89374FFF"),
                    name=c("Analysis"))+
  ylab("Prevalence")+
  theme(text = element_text(face="bold"),
        axis.title.x = element_blank())

stat.test <- rbind(prev_analysis$meco_mo16s_seqsg_raref,prev_analysis$meco_moits_seqsg_raref) %>%
  mutate(marker=as.factor(marker),
         analysis=as.factor(analysis))%>%
  group_by(marker) %>%
  rstatix::t_test(.,value ~ analysis,p.adjust.method = "holm") %>%
  rstatix::add_significance()

stat.test <- stat.test %>% rstatix::add_xy_position(.,x = "analysis")
bxp <- bxp + stat_pvalue_manual(stat.test,tip.length = 0,bracket.shorten = 0.3)

ggsave("./articlez/gen1/resultats/final/titan_diablo_prev_smry.png",bxp,width=4,height = 2.5)

rbind(prev_analysis$meco_mo16s_seqsg_raref,prev_analysis$meco_moits_seqsg_raref)%>%
  group_by(marker,analysis)%>%
  summarise(avrg_prev=mean(value))


bxp <- rbind(relab_analysis$meco_mo16s_seqsg_raref,relab_analysis$meco_moits_seqsg_raref)%>%
  ggplot(aes(x=analysis,y=value*100))+
  geom_boxplot(aes(fill=analysis)) +      
  scale_y_log10(breaks = c(0,.5,2.5,5,10,20,50,100))+
  ggh4x::facet_grid2(~marker,strip = strips)+  
  theme_classic2()+
  scale_fill_manual(values =c("#635C72FF","#89374FFF"),
                    name=c("Analysis"))+
  ylab("Relative abundance (%)")+
  xlab("")+
  theme(text = element_text(face="bold"),
        axis.title.x = element_blank())


stat.test <- rbind(relab_analysis$meco_mo16s_seqsg_raref,relab_analysis$meco_moits_seqsg_raref) %>%
  mutate(value=log10(value*100))%>%
  mutate(analysis=as.factor(analysis))%>%
  group_by(marker) %>%
  rstatix::t_test(data=.,formula=value ~ analysis) %>%
  rstatix::add_significance()

stat.test <- stat.test %>% rstatix::add_xy_position(x = "analysis")
bxp <- bxp + stat_pvalue_manual(stat.test,tip.length = 0,bracket.shorten = 0.3)

ggsave("./articlez/gen1/resultats/final/titan_diablo_relal_smry.png",bxp,width=4.5,height = 2.5)

rbind(relab_analysis$meco_mo16s_seqsg_raref,relab_analysis$meco_moits_seqsg_raref)%>%
  group_by(marker,analysis)%>%
  summarise(avrg_relal=mean(value)*100)




bac_diablotitotu <- intersect(gsub("M","",rownames(titan_list$meco_mo16s_seqsg_raref$dontrm$var_cat$notonly)),info_paper_diablo$meco_mo16s_seqsg_raref$interpratedotus)

bac_onlydiab <- info_paper_diablo$meco_mo16s_seqsg_raref$interpratedotus[!info_paper_diablo$meco_mo16s_seqsg_raref$interpratedotus%in%gsub("M","",rownames(titan_list$meco_mo16s_seqsg_raref$dontrm$var_cat$notonly))]

bac_onlytit <- gsub("M","",rownames(titan_list$meco_mo16s_seqsg_raref$dontrm$var_cat$notonly))[!gsub("M","",rownames(titan_list$meco_mo16s_seqsg_raref$dontrm$var_cat$notonly))%in%info_paper_diablo$meco_mo16s_seqsg_raref$interpratedotus]


fun_diablotitotu <- intersect(gsub("M","",rownames(titan_list$meco_moits_seqsg_raref$dontrm$var_cat$notonly)),info_paper_diablo$meco_moits_seqsg_raref$interpratedotus)

fun_onlydiab <- info_paper_diablo$meco_moits_seqsg_raref$interpratedotus[!info_paper_diablo$meco_moits_seqsg_raref$interpratedotus%in%gsub("M","",rownames(titan_list$meco_moits_seqsg_raref$dontrm$var_cat$notonly))]

fun_onlytit <- gsub("M","",rownames(titan_list$meco_moits_seqsg_raref$dontrm$var_cat$notonly))[!gsub("M","",rownames(titan_list$meco_moits_seqsg_raref$dontrm$var_cat$notonly))%in%info_paper_diablo$meco_moits_seqsg_raref$interpratedotus]


test_df <- data.frame(marker=c(rep("Bacteria",sum(length(bac_diablotitotu),length(bac_onlydiab),length(bac_onlytit))),
                               rep("Fungi",sum(length(fun_diablotitotu),length(fun_onlydiab),length(fun_onlytit)))),
                      intersect=c(rep("both",length(bac_diablotitotu)),
                                  rep("diablo",length(bac_onlydiab)),
                                  rep ("titan",length(bac_onlytit)),
                                  rep("both",length(fun_diablotitotu)),
                                  rep( "diablo",length(fun_onlydiab)),
                                  rep("titan",length(fun_onlytit))),
                      otu_name=c(bac_diablotitotu,bac_onlydiab,bac_onlytit,fun_diablotitotu,fun_onlydiab,fun_onlytit))


test_df2 <- test_df%>%
  group_by(marker,intersect)%>%
  summarise(counts=n())%>%
  mutate(intersect=paste0(stringr::str_sub(marker,1,1),"_",intersect))%>%
  arrange(desc(counts))%>%
  mutate(intersect=fct_inorder(intersect))


strips <- ggh4x::strip_themed(background_x = ggh4x::elem_list_rect(fill=c("#E09322")),
                              text_x = ggh4x::elem_list_text(color='white',face="bold"))


stripsf <- ggh4x::strip_themed(background_x = ggh4x::elem_list_rect(fill=c("#3F5E61")),
                               text_x = ggh4x::elem_list_text(color='white',face="bold"))


p_inter_analysis <- 
  test_df2%>%
  dplyr::filter(marker=="Bacteria")%>%
  ggplot( aes(x = intersect, y = counts,fill=intersect)) + 
  theme_classic2() +
  geom_col(width=.5) + 
  ylab("Number of OTUs") +
  theme(axis.title.x = element_blank(),
        axis.ticks.x = element_blank()) + 
  theme(axis.text = element_text(size = 8), 
        axis.title = element_text(size = 10))+
  scale_x_discrete(labels=toupper(gsub("._","",levels(test_df2$intersect))))+
  ggh4x::facet_wrap2(~marker,scales = "free_x",strip = strips)+
  scale_fill_manual(values=c(B_diablo="#635C72FF",
                             F_diablo="#635C72FF",
                             B_titan="#89374FFF",
                             F_titan="#89374FFF",
                             B_both="#0C2340FF",
                             F_both="#0C2340FF"),
                    name="Detected by:",
                    labels=c("DIABLO","TITAN","Both"))+
  scale_y_continuous(limits = c(0,1.1*max(test_df2[which(test_df2$marker=="Bacteria"),"counts"])),
                     expand = c(0,0))+
  test_df2%>%
  dplyr::filter(marker=="Fungi")%>%
  ggplot( aes(x = intersect, y = counts,fill=intersect)) + 
  theme_classic2() +
  geom_col(width=.5) + 
  ylab("") +
  theme(axis.title.x = element_blank(),
        axis.ticks.x = element_blank()) + 
  theme(axis.text = element_text(size = 8), 
        axis.title = element_text(size = 10))+
  scale_x_discrete(labels=toupper(gsub("._","",levels(test_df2$intersect)[4:6])))+
  ggh4x::facet_wrap2(~marker,scales = "free_x",strip = stripsf)+
  scale_fill_manual(values=c(B_diablo="#635C72FF",
                             F_diablo="#635C72FF",
                             B_titan="#89374FFF",
                             F_titan="#89374FFF",
                             B_both="#0C2340FF",
                             F_both="#0C2340FF"))+
  scale_y_continuous(limits = c(0,1.1*max(test_df2[which(test_df2$marker=="Fungi"),"counts"])),
                     expand = c(0,0))+
  theme(legend.position = "n")+
  plot_layout(guides = "collect")&
  theme(text=element_text(face="bold"),
        axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,size=6))

ggsave("./articlez/gen1/resultats/final/inter_method.png",p_inter_analysis,width = 5,height = 3)
```

### Relab responders

```{r}
relab_responders_cat <- NULL
store_df_tmp <- NULL
for(i in datasets){
  data_tmp <- get(i)
  data_tmp <- clone(data_tmp)
  tmp_motus <- slctd_otus_titan[[i]]
  titan_tmp <- titan_list[[i]]$hell$rmthem$var_cat$notonly
  
  
  otu_cat <- titan_tmp%>%
    as.data.frame()%>%
    mutate(otus=gsub("M","",rownames(.)))%>%
    reshape2::melt()%>%
    dplyr::rename(cleannames=variable)%>%
    left_join(trait_df_name)%>%
    filter(value!=0)%>%
    mutate(sing=ifelse(value<0,"Neg","Pos"))
  
  
  df_tmp <- data_tmp$otu_table%>%
    mutate_all(~.x/sum(.x))%>%
    as.data.frame()%>%
    mutate(otus=gsub("M","",rownames(.)))%>%
    mutate(istitan=ifelse(otus%in%tmp_motus,"titan","not titan"))%>%
    filter(istitan=="titan")%>%
    reshape2::melt(id.var=c("otus","istitan"))%>%
    filter(value!=0)
  
  
  df_tmp2 <- left_join(df_tmp,otu_cat[,-c(3)])%>%
    mutate(value=as.numeric(value))
  
  store_df_tmp[[i]] <- df_tmp2
  
  
  plot1 <- df_tmp2%>%
    filter(!grepl("PCA",cleannames))%>%
    mutate(var_cat=)%>%
    ggplot(aes(x=var_cat,y=(value),color=var_cat,shape=sing))+
    geom_point(position=position_jitterdodge(jitter.width = .3),alpha=0.5)+
    geom_boxplot(outlier.shape = NA,alpha=0)+
    scale_color_manual(name="Variable categories",values=var_cat_col[c(1:3,6,4,5)])+
    scale_shape_manual(values = c(16,17),name="Type of association",label=c("Negative",'Positive'))+
    theme_classic2()+
    ylab("Relative abundance %")+
    xlab("")+
    scale_y_log10(breaks = trans_breaks(identity, identity, n = 5))+
    ggtitle(ifelse(grepl("16s",i),"Bacteria","Fungi"))
  
  relab_responders_cat[[i]] <-  cld_bxplot_wilcox(data = df_tmp2,
                                                  x = df_tmp2$var_cat,
                                                  y =  df_tmp2$value,
                                                  plot = plot1,
                                                  nudge = .05)
  
  ggsave(paste0("./articlez/gen1/resultats/final/relab_responders/resp_",i,".png"),relab_responders_cat[[i]],width = 9,height = 6)
}
ggsave(paste0("./articlez/gen1/resultats/final/relab_responders/resp.png"),relab_responders_cat$meco_mo16s_seqsg_raref+relab_responders_cat$meco_moits_seqsg_raref+plot_layout(guides = "collect"),width = 14,height = 6)
```

### Fct slctd OTUs

```{r}
#ITS
# devtools::install_github("brendanf/FUNGuildR")
library(FUNGuildR)
tabtax_its <- meco_moits_seqsg_raref$tax_table%>%
  dplyr::filter(rownames(.)%in%rownames(titan_list$meco_moits_seqsg_raref$dontrm$var_cat$notonly))

paste_noNA <- function(x,sep=", ") {
  gsub(", " ,sep, toString(x[!is.na(x) & x!="" & x!="NA"] ) ) }

sep<-";"
tabtax_its$Taxonomy <- apply( tabtax_its[ , c(1:7) ] , 1 , paste_noNA , sep=sep)

fns_its <- funguild_assign(tabtax_its, db = get_funguild_db(), tax_col = "Taxonomy")
fns_its_cl <- fns_its %>% mutate(confidenceRanking = ifelse(confidenceRanking=="Possible",NA,confidenceRanking))

meco_moits_seqsg_raref$taxfun_table <- fns_its_cl
rownames(meco_moits_seqsg_raref$taxfun_table) <- rownames(tabtax_its)

pfct_fungi <- meco_moits_seqsg_raref$taxfun_table %>%
  mutate(otus=gsub("M","",rownames(.)))%>%
  reshape2::melt(id.vars="otus",measure.vars=c("trophicMode","guild","notes"))%>%
  ggplot(aes(x=variable,y=otus,label=value))+
  geom_tile(fill="lightgrey")+
  ggfittext::geom_fit_text(reflow = T)+
  scale_x_discrete(expand = c(0,0))+
  scale_y_discrete(expand = c(0,0))+
  theme(axis.text.x = element_text(angle = 90),
        axis.title = element_blank(),
        axis.line =element_blank(),
        panel.background = element_blank())+
  meco_moits_seqsg_raref$taxfun_table %>%
  mutate(Order_conf=ifelse(gsub("o__","",Order_conf)=="","Unkwown",gsub("o__","",Order_conf)))%>%
  ggplot(aes(x=1,y=rownames(.),fill=Order_conf))+
  geom_tile()+
  scale_x_discrete(expand = c(0,0))+
  scale_y_discrete(expand = c(0,0))+
  theme(axis.text = element_blank(),
        axis.title = element_blank(),
        axis.line =element_blank(),
        axis.ticks = element_blank(),
        panel.background = element_blank())+
  scale_fill_manual(values=colorRampPalette(brewer.pal(8,'Dark2'))(16))+
  plot_layout(widths = c(9,1))

ggsave("./articlez/gen1/resultats/final/fct_fungi.png",pfct_fungi,width = 10,height = 8)

#16s
data_tmp <-meco_mo16s_seqsg_raref
data_tmp <- clone(data_tmp)
data_tmp$tax_table%<>%
  dplyr::filter(rownames(.)%in%rownames(titan_list$meco_mo16s_seqsg_raref))
data_tmp$tidy_dataset()
func16s <- trans_func$new(data_tmp)
func16s$cal_spe_func(prok_database = "FAPROTAX")
func16sdf <- func16s$res_spe_func[,which(colSums(func16s$res_spe_func)>0)]

pfct_16s <- func16sdf[,order(colSums(func16sdf),decreasing = T)]%>%
  mutate(otus=gsub("M","",rownames(.)))%>%
  reshape2::melt(id.vars="otus")%>%
  ggplot(aes(x=variable,y=otus,fill=as.factor(value)))+
  geom_raster()+
  theme_classic()+
  scale_x_discrete(expand = c(0,0))+
  scale_y_discrete(expand = c(0,0))+
  theme(axis.text.x = element_text(angle = 90),
        axis.title = element_blank(),
        axis.line =element_blank(),
        panel.background = element_blank(),
        legend.position = "n")+
  scale_fill_manual(values = c("lightgrey","black"))+
  data_tmp$tax_table%>%
  mutate(Order_conf=ifelse(gsub("o__","",Order_conf)=="","Unkwown",gsub("o__","",Order_conf)))%>%
  ggplot(aes(x=1,y=rownames(.),fill=Order_conf))+
  geom_tile()+
  scale_x_discrete(expand = c(0,0))+
  scale_y_discrete(expand = c(0,0))+
  theme(axis.text = element_blank(),
        axis.title = element_blank(),
        axis.line =element_blank(),
        axis.ticks = element_blank(),
        panel.background = element_blank())+
  scale_fill_manual(values=colorRampPalette(brewer.pal(8,'Dark2'))(21))+
  plot_layout(widths = c(9,1))

ggsave("./articlez/gen1/resultats/final/fct_bact.png",pfct_16s,width = 10,height = 8)

```


### Info on slctd otus

```{r}
for(i in datasets){
  data_tmp <- get(i)
  data_tmp <- clone(data_tmp)
  
  taxo_tmp <- data_tmp$tax_table%>%
    mutate(OTUs=gsub("M","",rownames(.)))%>%
    filter(OTUs%in%c(slctd_otus_diablo[[i]],slctd_otus_titan[[i]]))%>%
    mutate(only_in_titan = ifelse(!OTUs%in%gsub("M","",slctd_otus_titan[[i]]),"Yes","No"))
  
  taxo_tmp$sequences <- gsub("s__","",data_tmp$ori_tax$sequence[which(rownames(data_tmp$ori_tax)%in%paste0("M",c(slctd_otus_diablo[[i]],slctd_otus_titan[[i]])))])
  taxo_tmp$reads <- data_tmp$otu_table[which(rownames(data_tmp$otu_table)%in%paste0("M",c(slctd_otus_diablo[[i]],slctd_otus_titan[[i]]))),]%>%
    rowSums()
  taxo_tmp$prevalence <- rowSums(data_tmp$otu_table[which(rownames(data_tmp$otu_table)%in%paste0("M",c(slctd_otus_diablo[[i]],slctd_otus_titan[[i]]))),]>0)
  
  
  if(grepl("16s",i)){
    fct_df <- func16sdf[,order(colSums(func16sdf),decreasing = T)]%>%
      mutate(OTUs=gsub("M","",rownames(.)))
  }else{
    fct_df <- meco_moits_seqsg_raref$taxfun_table %>%
      mutate(OTUs=gsub("M","",rownames(.)))
  }
  
  taxo_tmp%<>%left_join(fct_df)
  
  
  write.csv2(taxo_tmp,paste0("./articlez/gen1/resultats/final/otu_info_",i,".csv"))
}
```

# Save env

```{r}
save.image(file='./articlez/gen1/resultats/final/myEnvironment.RData')
```

```{r}

```
